{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " - Updated batch filtering to include a differential missingness filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import os\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import subprocess as sp\n",
    "import matplotlib.pyplot as plt\n",
    "from IPython.display import display\n",
    "\n",
    "import sys\n",
    "sys.path.append('../../code/gwas_analysis')\n",
    "from gwas_plotting import qqplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Full_IID</th>\n",
       "      <th>Genotype_Sex</th>\n",
       "      <th>ID_Reformat</th>\n",
       "      <th>SampleType</th>\n",
       "      <th>sex</th>\n",
       "      <th>Disease</th>\n",
       "      <th>ISTH_SandR_Key</th>\n",
       "      <th>KGH_Survivor_Key</th>\n",
       "      <th>KGH_AcuteLassa_Key</th>\n",
       "      <th>KGH_AcuteEbola_Key</th>\n",
       "      <th>...</th>\n",
       "      <th>Batch</th>\n",
       "      <th>Country</th>\n",
       "      <th>Sex_Discrepant</th>\n",
       "      <th>Sex_Discrepant_Notes</th>\n",
       "      <th>Status</th>\n",
       "      <th>Clean_HLA_ID</th>\n",
       "      <th>Epoch</th>\n",
       "      <th>Array</th>\n",
       "      <th>CollectionEpoch</th>\n",
       "      <th>Age</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Full_IID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>July2014NG_SM-6C4M9</th>\n",
       "      <td>July2014NG_SM-6C4M9</td>\n",
       "      <td>1</td>\n",
       "      <td>SM-6C4M9</td>\n",
       "      <td>SuspectedLassaNG</td>\n",
       "      <td>1</td>\n",
       "      <td>Lassa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>July2014NG</td>\n",
       "      <td>NG</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Passes QC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>Omni_25M_B</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>July2014NG_SM-4XBS9</th>\n",
       "      <td>July2014NG_SM-4XBS9</td>\n",
       "      <td>2</td>\n",
       "      <td>SM-4XBS9</td>\n",
       "      <td>Pre2016ControlNG</td>\n",
       "      <td>2</td>\n",
       "      <td>Lassa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>July2014NG</td>\n",
       "      <td>NG</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Passes QC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>Omni_25M_B</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>July2014NG_SM-6C4MA</th>\n",
       "      <td>July2014NG_SM-6C4MA</td>\n",
       "      <td>1</td>\n",
       "      <td>SM-6C4MA</td>\n",
       "      <td>SuspectedLassaNG</td>\n",
       "      <td>1</td>\n",
       "      <td>Lassa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>July2014NG</td>\n",
       "      <td>NG</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Passes QC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>Omni_25M_B</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>32.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>July2014NG_SM-6C4MB</th>\n",
       "      <td>July2014NG_SM-6C4MB</td>\n",
       "      <td>2</td>\n",
       "      <td>SM-6C4MB</td>\n",
       "      <td>SuspectedLassaNG</td>\n",
       "      <td>2</td>\n",
       "      <td>Lassa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>July2014NG</td>\n",
       "      <td>NG</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Passes QC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>Omni_25M_B</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>65.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>July2014NG_SM-4XBSM</th>\n",
       "      <td>July2014NG_SM-4XBSM</td>\n",
       "      <td>2</td>\n",
       "      <td>SM-4XBSM</td>\n",
       "      <td>Pre2016ControlNG</td>\n",
       "      <td>2</td>\n",
       "      <td>Lassa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>July2014NG</td>\n",
       "      <td>NG</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Passes QC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>Omni_25M_B</td>\n",
       "      <td>Pre2016</td>\n",
       "      <td>47.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Full_IID Genotype_Sex ID_Reformat  \\\n",
       "Full_IID                                                            \n",
       "July2014NG_SM-6C4M9  July2014NG_SM-6C4M9            1    SM-6C4M9   \n",
       "July2014NG_SM-4XBS9  July2014NG_SM-4XBS9            2    SM-4XBS9   \n",
       "July2014NG_SM-6C4MA  July2014NG_SM-6C4MA            1    SM-6C4MA   \n",
       "July2014NG_SM-6C4MB  July2014NG_SM-6C4MB            2    SM-6C4MB   \n",
       "July2014NG_SM-4XBSM  July2014NG_SM-4XBSM            2    SM-4XBSM   \n",
       "\n",
       "                           SampleType sex Disease ISTH_SandR_Key  \\\n",
       "Full_IID                                                           \n",
       "July2014NG_SM-6C4M9  SuspectedLassaNG   1   Lassa            NaN   \n",
       "July2014NG_SM-4XBS9  Pre2016ControlNG   2   Lassa            NaN   \n",
       "July2014NG_SM-6C4MA  SuspectedLassaNG   1   Lassa            NaN   \n",
       "July2014NG_SM-6C4MB  SuspectedLassaNG   2   Lassa            NaN   \n",
       "July2014NG_SM-4XBSM  Pre2016ControlNG   2   Lassa            NaN   \n",
       "\n",
       "                    KGH_Survivor_Key KGH_AcuteLassa_Key KGH_AcuteEbola_Key  \\\n",
       "Full_IID                                                                     \n",
       "July2014NG_SM-6C4M9              NaN                NaN                NaN   \n",
       "July2014NG_SM-4XBS9              NaN                NaN                NaN   \n",
       "July2014NG_SM-6C4MA              NaN                NaN                NaN   \n",
       "July2014NG_SM-6C4MB              NaN                NaN                NaN   \n",
       "July2014NG_SM-4XBSM              NaN                NaN                NaN   \n",
       "\n",
       "                     ...       Batch Country Sex_Discrepant  \\\n",
       "Full_IID             ...                                      \n",
       "July2014NG_SM-6C4M9  ...  July2014NG      NG          False   \n",
       "July2014NG_SM-4XBS9  ...  July2014NG      NG          False   \n",
       "July2014NG_SM-6C4MA  ...  July2014NG      NG          False   \n",
       "July2014NG_SM-6C4MB  ...  July2014NG      NG          False   \n",
       "July2014NG_SM-4XBSM  ...  July2014NG      NG          False   \n",
       "\n",
       "                     Sex_Discrepant_Notes     Status Clean_HLA_ID    Epoch  \\\n",
       "Full_IID                                                                     \n",
       "July2014NG_SM-6C4M9                   NaN  Passes QC          NaN  Pre2016   \n",
       "July2014NG_SM-4XBS9                   NaN  Passes QC          NaN  Pre2016   \n",
       "July2014NG_SM-6C4MA                   NaN  Passes QC          NaN  Pre2016   \n",
       "July2014NG_SM-6C4MB                   NaN  Passes QC          NaN  Pre2016   \n",
       "July2014NG_SM-4XBSM                   NaN  Passes QC          NaN  Pre2016   \n",
       "\n",
       "                          Array CollectionEpoch   Age  \n",
       "Full_IID                                               \n",
       "July2014NG_SM-6C4M9  Omni_25M_B         Pre2016  30.0  \n",
       "July2014NG_SM-4XBS9  Omni_25M_B         Pre2016  26.0  \n",
       "July2014NG_SM-6C4MA  Omni_25M_B         Pre2016  32.0  \n",
       "July2014NG_SM-6C4MB  Omni_25M_B         Pre2016  65.0  \n",
       "July2014NG_SM-4XBSM  Omni_25M_B         Pre2016  47.0  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phenofn = 'gs://popgen-gwas/data/metadata/Annotated_GenotypedData/preANDpost2016_Merged_Complete_Metadata_WithAge_20210818.tsv'\n",
    "metadata = pd.read_csv(phenofn, sep='\\t')\n",
    "metadata.index = metadata['Full_IID']\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_fn = 'gs://popgen-gwas/data/metadata/Annotated_GenotypedData/OmniH3Merged_PrePost2016_AggregatedLASVPhenotypes_20210829.tsv'\n",
    "groups = pd.read_csv(group_fn, sep='\\t')\n",
    "groups.index = groups['#IID']\n",
    "ind = groups.index[groups['Post2016_NG_LASV_qPCR2xPosOrSeqPos'] | groups['Post2016_NG_LASV_SeqPos'] | groups['Pre2016_NG_LASV_SeqPos'] | groups['Pre2016_SL_LASV_AgORSeqPos'] | groups['Post2016_SL_LASV_AgPos']]\n",
    "metadata['LASV_Case'] = 0\n",
    "metadata.loc[ind, 'LASV_Case'] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = '../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223'\n",
    "dupfiltfn_fam = '../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210720.Geno_1e-1.filtBatchvars20210720.DupsToExclude.txt'\n",
    "dupstofilt = pd.read_csv(dupfiltfn_fam, sep='\\t')['1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2     115716\n",
       "1     106801\n",
       "3      95664\n",
       "6      91532\n",
       "4      90673\n",
       "5      87044\n",
       "7      76390\n",
       "8      75365\n",
       "10     71934\n",
       "11     67723\n",
       "12     66940\n",
       "9      62578\n",
       "13     48887\n",
       "16     45412\n",
       "14     45049\n",
       "15     43575\n",
       "18     41243\n",
       "17     37369\n",
       "20     32661\n",
       "23     26316\n",
       "19     26119\n",
       "22     18854\n",
       "21     18330\n",
       "24        58\n",
       "Name: 0, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(a+'.bim', sep='\\t', header=None)[0].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prepare the VCF file for imputation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "b = '../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "plink2 --bfile ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223 --recode vcf-iid id-delim=\"|\" --out ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223 --keep-allele-order  --geno 0.1 --hwe 1e-12\n",
      "PLINK v2.00a3LM 64-bit Intel (1 Jul 2021)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.log.\n",
      "Options in effect:\n",
      "  --bfile ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223\n",
      "  --export vcf-iid id-delim=|\n",
      "  --geno 0.1\n",
      "  --hwe 1e-12\n",
      "  --keep-allele-order\n",
      "  --out ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223\n",
      "\n",
      "Start time: Wed Feb 23 15:34:02 2022\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "Note: --keep-allele-order no longer has any effect.\n",
      "64422 MiB RAM detected; reserving 32211 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "5555 samples (2945 females, 2610 males; 5302 founders) loaded from\n",
      "../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.fam.\n",
      "1392233 variants loaded from\n",
      "../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.bim.\n",
      "1 binary phenotype loaded (2841 cases, 2714 controls).\n",
      "Calculating allele frequencies... 14182328323742475156616570758084899498done.\n",
      "--geno: 0 variants removed due to missing genotype data.\n",
      "Computing chrX Hardy-Weinberg p-values... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--hwe: 4150 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "1388083 variants remaining after main filters.\n",
      "--export vcf to\n",
      "../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Feb 23 15:36:12 2022\n",
      "bcftools annotate -Ou --rename-chrs ../../data/tmp/misc/plink2ensembl.txt ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.vcf > ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.bcf\n",
      "bcftools sort -Ob ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.bcf > ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.bcf\n",
      "Writing to /tmp/bcftools-sort.lOl3z5\n",
      "Merging 21 temporary files\n",
      "Cleaning\n",
      "Done\n",
      "bcftools index ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.bcf\n",
      "bcftools +guess-ploidy -g b37 ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.bcf > ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.SexPloidy.txt\n",
      "Warning: PL tag not found in header, switching to GL\n",
      "Warning: GL tag not found in header, switching to GT\n",
      "bcftools +fixploidy --threads 2 ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.bcf -Oz -o ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.fixPloidy.vcf.gz -- -s ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.SexPloidy.txt\n",
      "bcftools +fixref ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.fixPloidy.vcf.gz -- -f ../../data/tmp/misc/human_g1k_v37.fasta\n",
      "# SC, guessed strand convention\n",
      "SC\tTOP-compatible\t0\n",
      "SC\tBOT-compatible\t0\n",
      "# ST, substitution types\n",
      "ST\tA>C\t58893\t4.2%\n",
      "ST\tA>G\t249551\t18.0%\n",
      "ST\tA>T\t6154\t0.4%\n",
      "ST\tC>A\t68300\t4.9%\n",
      "ST\tC>G\t7326\t0.5%\n",
      "ST\tC>T\t303707\t21.9%\n",
      "ST\tG>A\t302930\t21.8%\n",
      "ST\tG>C\t7373\t0.5%\n",
      "ST\tG>T\t68128\t4.9%\n",
      "ST\tT>A\t6177\t0.4%\n",
      "ST\tT>C\t250656\t18.1%\n",
      "ST\tT>G\t58888\t4.2%\n",
      "# NS, Number of sites:\n",
      "NS\ttotal        \t1388083\n",
      "NS\tref match    \t1388083\t100.0%\n",
      "NS\tref mismatch \t0\t0.0%\n",
      "NS\tskipped      \t0\n",
      "NS\tnon-ACGT     \t0\n",
      "NS\tnon-SNP      \t0\n",
      "NS\tnon-biallelic\t0\n"
     ]
    }
   ],
   "source": [
    "cmd = 'plink2 --bfile {b} --recode vcf-iid id-delim=\"|\" --out {b} --keep-allele-order  --geno 0.1 --hwe 1e-12'.format(b=b)\n",
    "print(cmd)\n",
    "!{cmd}\n",
    "\n",
    "#b= '{b}.maf01'.format(b=b)\n",
    "\n",
    "## Convert 23-->X 24-->Y\n",
    "vcf_fn = b+'.vcf'\n",
    "vcf_fixchr_fn = b+'.renamesexchr.bcf'\n",
    "plink2ensembl = '../../data/tmp/misc/plink2ensembl.txt'\n",
    "cmd = 'bcftools annotate -Ou --rename-chrs %s %s > %s' % (plink2ensembl, vcf_fn, vcf_fixchr_fn)\n",
    "print(cmd)\n",
    "!{cmd}\n",
    "\n",
    "vcf_fixchr_sort_fn = b+'.renamesexchr.sorted.bcf'\n",
    "cmd = 'bcftools sort -Ob %s > %s' % (vcf_fixchr_fn, vcf_fixchr_sort_fn)\n",
    "print(cmd)\n",
    "!{cmd}\n",
    "\n",
    "cmd = 'bcftools index %s' % vcf_fixchr_sort_fn\n",
    "print(cmd)\n",
    "!{cmd}\n",
    "\n",
    "sampfn = vcf_fixchr_sort_fn.replace('.bcf', '.SexPloidy.txt')\n",
    "cmd = 'bcftools +guess-ploidy -g b37 %s > %s' % (vcf_fixchr_sort_fn, sampfn)\n",
    "print(cmd)\n",
    "! {cmd}\n",
    "\n",
    "vcf_fixploidy_fn = vcf_fixchr_sort_fn.replace('.bcf', '.fixPloidy.vcf.gz')\n",
    "cmd = 'bcftools +fixploidy --threads 2 %s -Oz -o %s -- -s %s' % (vcf_fixchr_sort_fn, vcf_fixploidy_fn, sampfn)\n",
    "print(cmd)\n",
    "!{cmd}\n",
    "\n",
    "## Check everything is fixed\n",
    "ref_fn = '../../data/tmp/misc/human_g1k_v37.fasta'\n",
    "cmd = 'bcftools +fixref %s -- -f %s' % (vcf_fixploidy_fn, ref_fn)\n",
    "print(cmd)\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "! bcftools index ../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.renamesexchr.sorted.fixPloidy.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "outbase = '../../data/tmp/plink/MergePrePost/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223'\n",
    "gcsdir = 'gs://popgen-gwas/data/genotype/merge/Unimputed/'\n",
    "for ext in ['bed', 'bim', 'fam', 'renamesexchr.sorted.fixPloidy.vcf.gz', 'renamesexchr.sorted.fixPloidy.vcf.gz.csi']:\n",
    "    cmd = 'gsutil -m cp {a}.{ext} {gcs}'.format(a=outbase, ext=ext, gcs=gcsdir)\n",
    "    print(cmd)\n",
    "    !{cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
