{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "import sys\n",
    "sys.path.append('../../code/gwas_analysis')\n",
    "from gwas_plotting import manhattan, manhattan_fixwidth, qqplot\n",
    "\n",
    "from matplotlib import rcParams, font_manager\n",
    "from scipy.stats import chi2\n",
    "\n",
    "rcParams['pdf.fonttype'] = 42\n",
    "rcParams['ps.fonttype'] = 42\n",
    "rcParams['font.family'] = 'sans-serif'\n",
    "rcParams['font.sans-serif'] = ['Arial']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import statsmodels.stats.multitest as sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A', 'B', 'C', 'DPA1', 'DPB1', 'DQA1', 'DQB1', 'DRB1']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = ['A', 'B', 'C', 'DPA1', 'DPB1', 'DQA1', 'DQB1', 'DRB1']\n",
    "genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile={infn}          --vcfFileIndex={infn}.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile={outfn}         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/{rdir}/{p}/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/{rdir}/{p}/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n"
     ]
    }
   ],
   "source": [
    "cmd_base = \"source activate RSAIGE\\n\"\n",
    "cmd_base += \"Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R        \\\n",
    "        --vcfFile={infn}  \\\n",
    "        --vcfFileIndex={infn}.csi    \\\n",
    "        --vcfField=DS   \\\n",
    "        --AlleleOrder=ref-first \\\n",
    "        --SAIGEOutputFile={outfn} \\\n",
    "        --chrom=6       \\\n",
    "        --minMAF=0 \\\n",
    "        --minMAC=20 \\\n",
    "        --GMMATmodelFile=/home/sraju/{rdir}/{p}/results/raw_gwas/SAIGE.rda \\\n",
    "        --varianceRatioFile=/home/sraju/{rdir}/{p}/results/raw_gwas/SAIGE.varianceRatio.txt   \\\n",
    "        --LOCO=TRUE \\\n",
    "        --is_Firth_beta=TRUE       \\\n",
    "        --pCutoffforFirth=0.01       \\\n",
    "        --is_output_moreDetails=TRUE\"\n",
    "print(cmd_base)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncores = 7\n",
    "resdirmap = {'MergeEpoch_NG_Died_vs_Survived':'outcome_H3covar_20220315',\n",
    "         'MergeEpoch_SL_Died_vs_Survived':'outcome_H3covar_20220315',\n",
    "          'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl':'results_20220311',\n",
    "             'MergeEpoch_SL_AgORSeqPos_Vs_PopControl':'results_20220311'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcfcols = ['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT']\n",
    "\n",
    "header = \"\"\"##fileformat=VCFv4.3\n",
    "##fileDate=20230823\n",
    "##source=PLINKv2.00\n",
    "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
    "##contig=<ID=17,length=81195210>\n",
    "##pbwtVersion=3.1-v3.1-2-gbf6ebe2+htslib-1.3.2-199-gec1d68e-dirty\n",
    "##pbwtCommand=pbwt -log dacceb7706d8cd40f3d5f3ecff0dd56e.pbwt/0/7/0.17.bcf.log -\n",
    "##INFO=<ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes\">\n",
    "##INFO=<ID=AN,Number=1,Type=Integer,Description=\"Total number of alleles in call\n",
    "##INFO=<ID=RefPanelAF,Number=A,Type=Float,Description=\"Allele frequency in imput\n",
    "##INFO=<ID=TYPED,Number=0,Type=Flag,Description=\"Site was genotyped prior to imp\n",
    "##bcftools_mergeVersion=1.9+htslib-1.9\n",
    "##bcftools_mergeCommand=merge -l dacceb7706d8cd40f3d5f3ecff0dd56e.pbwt/merged/8/\n",
    "##bcftools_annotateVersion=1.9+htslib-1.9\n",
    "##bcftools_annotateCommand=annotate -Ou -x INFO/DR2; Date=Thu Feb 24 18:41:11 20\n",
    "##INFO=<ID=INFO,Number=1,Type=Float,Description=\"IMPUTE2 info score\">\n",
    "##bcftools_pluginVersion=1.9+htslib-1.9\n",
    "##bcftools_pluginCommand=plugin impute-info -Ou; Date=Thu Feb 24 18:41:11 2022\n",
    "##bcftools_annotateCommand=annotate -Ou -c CHROM,POS,ID,REF,ALT -a resources/dbs\n",
    "##FORMAT=<ID=DS,Number=A,Type=Float,Description=\"Estimated Alternate Allele Dosage\"\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:32:49) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "12 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.501   2.283   4.611 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:32:55) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "2 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.466   2.315   4.626 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../A_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:33:01) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "22 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.520   2.360   4.714 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:33:06) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "17 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.469   2.364   4.664 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../A_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "B\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:33:12) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "10 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.379   2.372   4.581 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:33:18) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "2 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.461   2.260   4.560 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../B_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:33:24) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "20 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.750   2.143   4.727 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:33:30) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "21 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.507   2.280   4.623 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../B_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "C\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:33:35) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "11 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.611   2.212   4.654 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:33:41) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "3 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.457   2.269   4.560 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../C_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:33:47) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "19 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.713   2.140   4.687 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:33:53) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "16 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.611   2.175   4.629 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../C_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "DPA1\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:33:58) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "4 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.517   2.245   4.595 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:34:04) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "4 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.302   2.439   4.578 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:34:10) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "4 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.554   2.272   4.653 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:34:15) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "4 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.461   2.311   4.612 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "DPB1\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:34:21) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "7 markers were tested.\n",
      "Firth approx was applied to 1 markers. 1 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.394   2.350   4.583 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:34:26) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "1 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.420   2.320   4.571 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:34:32) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "13 markers were tested.\n",
      "Firth approx was applied to 1 markers. 1 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.600   2.285   4.716 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:34:38) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "12 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.534   2.292   4.653 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DPB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "DQA1\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:34:44) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "8 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.441   2.329   4.597 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:34:50) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "5 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.526   2.218   4.577 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:34:55) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "10 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.555   2.280   4.666 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:35:01) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "10 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.600   2.112   4.656 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQA1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "DQB1\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:35:07) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "8 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.499   2.294   4.619 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:35:12) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "4 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.515   2.240   4.584 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:35:18) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "14 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.590   2.269   4.693 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:35:24) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "11 markers were tested.\n",
      "Firth approx was applied to 1 markers. 1 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.529   2.270   4.637 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DQB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "DRB1\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9938294 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 303\n",
      "Setting position of samples in VCF files....\n",
      "m_N 303\n",
      "(2023-08-24 21:35:30) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "11 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.440   2.242   4.567 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9943305 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 108\n",
      "Setting position of samples in VCF files....\n",
      "m_N 108\n",
      "(2023-08-24 21:35:35) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "2 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.552   2.280   4.664 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9749193 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 1598\n",
      "Setting position of samples in VCF files....\n",
      "m_N 1598\n",
      "(2023-08-24 21:35:41) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "17 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.569   2.317   4.717 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n",
      "source activate RSAIGE\n",
      "Rscript /home/sraju/SAIGE_Git/extdata/step2_SPAtests.R                --vcfFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz          --vcfFileIndex=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi            --vcfField=DS           --AlleleOrder=ref-first         --SAIGEOutputFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv         --chrom=6               --minMAF=0         --minMAC=20         --GMMATmodelFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda         --varianceRatioFile=/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt           --LOCO=TRUE         --is_Firth_beta=TRUE               --pCutoffforFirth=0.01               --is_output_moreDetails=TRUE\n",
      "Loading required package: RhpcBLASctl\n",
      "R version 3.6.1 (2019-07-05)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 16.04.7 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /home/sraju/miniconda3/envs/RSAIGE/lib/R/lib/libRblas.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      "[1] data.table_1.14.8   optparse_1.7.3      RhpcBLASctl_0.23-42\n",
      "[4] SAIGE_1.2.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      "[1] compiler_3.6.1     Matrix_1.6-1       Rcpp_1.0.11        getopt_1.20.3     \n",
      "[5] grid_3.6.1         RcppParallel_5.1.7 lattice_0.20-38   \n",
      "$vcfFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz\"\n",
      "\n",
      "$vcfFileIndex\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.vcf.gz.csi\"\n",
      "\n",
      "$vcfField\n",
      "[1] \"DS\"\n",
      "\n",
      "$savFile\n",
      "[1] \"\"\n",
      "\n",
      "$savFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFile\n",
      "[1] \"\"\n",
      "\n",
      "$bgenFileIndex\n",
      "[1] \"\"\n",
      "\n",
      "$sampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$bedFile\n",
      "[1] \"\"\n",
      "\n",
      "$bimFile\n",
      "[1] \"\"\n",
      "\n",
      "$famFile\n",
      "[1] \"\"\n",
      "\n",
      "$AlleleOrder\n",
      "[1] \"ref-first\"\n",
      "\n",
      "$idstoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$rangestoIncludeFile\n",
      "[1] \"\"\n",
      "\n",
      "$chrom\n",
      "[1] \"6\"\n",
      "\n",
      "$is_imputed_data\n",
      "[1] FALSE\n",
      "\n",
      "$minMAF\n",
      "[1] 0\n",
      "\n",
      "$minMAC\n",
      "[1] 20\n",
      "\n",
      "$minGroupMAC_in_BurdenTest\n",
      "[1] 5\n",
      "\n",
      "$minInfo\n",
      "[1] 0\n",
      "\n",
      "$maxMissing\n",
      "[1] 0.15\n",
      "\n",
      "$impute_method\n",
      "[1] \"best_guess\"\n",
      "\n",
      "$LOCO\n",
      "[1] TRUE\n",
      "\n",
      "$GMMATmodelFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.rda\"\n",
      "\n",
      "$varianceRatioFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/SAIGE.varianceRatio.txt\"\n",
      "\n",
      "$SAIGEOutputFile\n",
      "[1] \"/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv\"\n",
      "\n",
      "$markers_per_chunk\n",
      "[1] 10000\n",
      "\n",
      "$groups_per_chunk\n",
      "[1] 100\n",
      "\n",
      "$is_output_moreDetails\n",
      "[1] TRUE\n",
      "\n",
      "$is_overwrite_output\n",
      "[1] TRUE\n",
      "\n",
      "$maxMAF_in_groupTest\n",
      "[1] \"0.0001,0.001,0.01\"\n",
      "\n",
      "$maxMAC_in_groupTest\n",
      "[1] \"0\"\n",
      "\n",
      "$annotation_in_groupTest\n",
      "[1] \"lof,missense;lof,missense;lof;synonymous\"\n",
      "\n",
      "$groupFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMFile\n",
      "[1] \"\"\n",
      "\n",
      "$sparseGRMSampleIDFile\n",
      "[1] \"\"\n",
      "\n",
      "$relatednessCutoff\n",
      "[1] 0\n",
      "\n",
      "$MACCutoff_to_CollapseUltraRare\n",
      "[1] 10\n",
      "\n",
      "$cateVarRatioMinMACVecExclude\n",
      "[1] \"10,20.5\"\n",
      "\n",
      "$cateVarRatioMaxMACVecInclude\n",
      "[1] \"20.5\"\n",
      "\n",
      "$weights.beta\n",
      "[1] \"1,25\"\n",
      "\n",
      "$r.corr\n",
      "[1] 0\n",
      "\n",
      "$markers_per_chunk_in_groupTest\n",
      "[1] 100\n",
      "\n",
      "$condition\n",
      "[1] \"\"\n",
      "\n",
      "$SPAcutoff\n",
      "[1] 2\n",
      "\n",
      "$dosage_zerod_cutoff\n",
      "[1] 0.2\n",
      "\n",
      "$dosage_zerod_MAC_cutoff\n",
      "[1] 10\n",
      "\n",
      "$is_single_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_no_weight_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_output_markerList_in_groupTest\n",
      "[1] FALSE\n",
      "\n",
      "$is_Firth_beta\n",
      "[1] TRUE\n",
      "\n",
      "$pCutoffforFirth\n",
      "[1] 0.01\n",
      "\n",
      "$is_fastTest\n",
      "[1] FALSE\n",
      "\n",
      "$max_MAC_for_ER\n",
      "[1] 4\n",
      "\n",
      "$subSampleFile\n",
      "[1] \"\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "[1] \"opt$r.corr\"\n",
      "[1] 0\n",
      "dosage_zerod_cutoff  0.2 \n",
      "Any dosages <=  0.2  for genetic variants with MAC <=  10  are set to be 0.\n",
      "single-variant association test will be performed\n",
      "Leave chromosome  6  out will be applied\n",
      "chromosome  1  model results are removed to save memory\n",
      "chromosome  2  model results are removed to save memory\n",
      "chromosome  3  model results are removed to save memory\n",
      "chromosome  4  model results are removed to save memory\n",
      "chromosome  5  model results are removed to save memory\n",
      "chromosome  7  model results are removed to save memory\n",
      "chromosome  8  model results are removed to save memory\n",
      "chromosome  9  model results are removed to save memory\n",
      "chromosome  10  model results are removed to save memory\n",
      "chromosome  11  model results are removed to save memory\n",
      "chromosome  12  model results are removed to save memory\n",
      "chromosome  13  model results are removed to save memory\n",
      "chromosome  14  model results are removed to save memory\n",
      "chromosome  15  model results are removed to save memory\n",
      "chromosome  16  model results are removed to save memory\n",
      "chromosome  17  model results are removed to save memory\n",
      "chromosome  18  model results are removed to save memory\n",
      "chromosome  19  model results are removed to save memory\n",
      "chromosome  20  model results are removed to save memory\n",
      "chromosome  21  model results are removed to save memory\n",
      "chromosome  22  model results are removed to save memory\n",
      "P-values of genetic variants with MAC <=  4  will be calculated via effecient resampling.\n",
      "variance Ratio null is  0.9550702 \n",
      "Please note the argument vcfFileIndex will not be used in future versions because the vcf index file must has the name 'vcfFile'.csi\n",
      "dosageFile type is  vcf \n",
      "Open VCF done\n",
      "To read the field DS\n",
      "Number of meta lines in the vcf file (lines starting with ##): 20\n",
      "Number of samples in the vcf file: 921\n",
      "Setting position of samples in VCF files....\n",
      "m_N 921\n",
      "(2023-08-24 21:35:47) ---- Analyzing Chunk 1 :  chrom InitialChunk ---- \n",
      "Reach the end of the vcf file\n",
      "19 markers were tested.\n",
      "Firth approx was applied to 0 markers. 0 sucessfully converged.\n",
      "write to output\n",
      "   user  system elapsed \n",
      "  6.534   2.251   4.621 \n",
      "isVcfEnd  TRUE \n",
      "[1] \"Analysis done! The results have been saved to '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/raw_data/../DRB1_hla_dosage_forSAIGE.results.tsv'.\"\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for gene in genes:\n",
    "    print(gene)\n",
    "    curr_df = pd.read_csv('/home/sraju/HIBAG_sandbox/outputs/' + gene + '_dosages.txt', sep = '\\t', index_col = 0)\n",
    "    for p in ['MergeEpoch_NG_Died_vs_Survived', 'MergeEpoch_SL_Died_vs_Survived',\n",
    "          'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl', 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl']:\n",
    "    \n",
    "        famfn = '/home/sraju/{rdir}/{p}/raw_data/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.deduped.{p}.maf_1e-2.hwe_1e-6.geno_5e-2.FixSexChr.fam'.format(p=p, rdir=resdirmap[p])\n",
    "        samporder = pd.read_csv(famfn, sep='\\t', header=None)[1]    \n",
    "\n",
    "        hlainfo = curr_df.loc[:, samporder]\n",
    "\n",
    "        hlainfo['#CHROM'] = '6'\n",
    "        hlainfo['POS'] = '1'\n",
    "        hlainfo['ID'] = list(curr_df.index) # the allele name\n",
    "        hlainfo['REF'] = 'A'\n",
    "        hlainfo['ALT'] = 'G'\n",
    "        hlainfo['QUAL'] = '.'\n",
    "        hlainfo['FILTER'] = 'PASS'\n",
    "        hlainfo['INFO'] = '.'\n",
    "        hlainfo['FORMAT'] = 'DS'\n",
    "    \n",
    "        hlainfo = hlainfo.loc[:, vcfcols +list(samporder)]    \n",
    "        hlafn = os.path.join(os.path.dirname(famfn), '../' + gene + '_hla_dosage_forSAIGE.vcf').format(p=p)\n",
    "    \n",
    "        with open(hlafn, 'w') as F:\n",
    "            F.write(header)\n",
    "    \n",
    "        hlainfo.to_csv(hlafn, sep='\\t', index=False, mode='a')\n",
    "        outfn =  hlafn.replace('.vcf', '.results.tsv')\n",
    "\n",
    "        cmd = 'bgzip -c {fn} > {fn}.gz'.format(fn=hlafn)\n",
    "        #print(cmd)\n",
    "        !{cmd}\n",
    "    \n",
    "        cmd = 'tabix --csi -p vcf %s.gz' % hlafn\n",
    "        #print(cmd)\n",
    "        !{cmd}    \n",
    "    \n",
    "    \n",
    "        cmd = cmd_base.format(infn=hlafn+'.gz', outfn=outfn, rdir=resdirmap[p], p=p)\n",
    "        print(cmd)\n",
    "        !{cmd}\n",
    "        print('\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "res_all = {}\n",
    "for p in ['MergeEpoch_NG_Died_vs_Survived', 'MergeEpoch_SL_Died_vs_Survived',\n",
    "          'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl', 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl']:\n",
    "    res_pheno = {}\n",
    "    for gene in genes:\n",
    "        curr_df = pd.read_csv('/home/sraju/HIBAG_sandbox/outputs/' + gene + '_dosages.txt', sep = '\\t', index_col = 0)\n",
    "        famfn = '/home/sraju/{rdir}/{p}/raw_data/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220223.deduped.{p}.maf_1e-2.hwe_1e-6.geno_5e-2.FixSexChr.fam'.format(p=p, rdir=resdirmap[p])    \n",
    "        hlafn = os.path.join(os.path.dirname(famfn), '../' + gene + '_hla_dosage_forSAIGE.vcf').format(p=p)\n",
    "        outfn =  hlafn.replace('.vcf', '.results.tsv')\n",
    "\n",
    "        res = pd.read_csv(outfn, sep='\\t')\n",
    "        res['MarkerID'] = res['MarkerID'] + gene\n",
    "        res.index = res['MarkerID']\n",
    "        res['gene'] = gene\n",
    "        res['N'] = res['N_case'] + res['N_ctrl']\n",
    "        res_pheno[gene] = res\n",
    "    res_all[p] = res_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>MarkerID</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>AC_Allele2</th>\n",
       "      <th>AF_Allele2</th>\n",
       "      <th>MissingRate</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>...</th>\n",
       "      <th>AF_case</th>\n",
       "      <th>AF_ctrl</th>\n",
       "      <th>N_case</th>\n",
       "      <th>N_ctrl</th>\n",
       "      <th>N_case_hom</th>\n",
       "      <th>N_case_het</th>\n",
       "      <th>N_ctrl_hom</th>\n",
       "      <th>N_ctrl_het</th>\n",
       "      <th>gene</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MarkerID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23:01A</th>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>23:01A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>35.7732</td>\n",
       "      <td>0.165616</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.485616</td>\n",
       "      <td>0.474548</td>\n",
       "      <td>...</td>\n",
       "      <td>0.164866</td>\n",
       "      <td>0.167000</td>\n",
       "      <td>70</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>A</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>02:01A</th>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>02:01A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>37.7398</td>\n",
       "      <td>0.174721</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.139989</td>\n",
       "      <td>0.426329</td>\n",
       "      <td>...</td>\n",
       "      <td>0.148094</td>\n",
       "      <td>0.223771</td>\n",
       "      <td>70</td>\n",
       "      <td>38</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>A</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR  POS MarkerID Allele1 Allele2  AC_Allele2  AF_Allele2  \\\n",
       "MarkerID                                                              \n",
       "23:01A      6    1   23:01A       A       G     35.7732    0.165616   \n",
       "02:01A      6    1   02:01A       A       G     37.7398    0.174721   \n",
       "\n",
       "          MissingRate      BETA        SE  ...   AF_case   AF_ctrl  N_case  \\\n",
       "MarkerID                                   ...                               \n",
       "23:01A              0 -0.485616  0.474548  ...  0.164866  0.167000      70   \n",
       "02:01A              0 -0.139989  0.426329  ...  0.148094  0.223771      70   \n",
       "\n",
       "          N_ctrl  N_case_hom  N_case_het  N_ctrl_hom  N_ctrl_het  gene    N  \n",
       "MarkerID                                                                     \n",
       "23:01A        38           1          23           1          11     A  108  \n",
       "02:01A        38           5          11           1          15     A  108  \n",
       "\n",
       "[2 rows x 25 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res_all['MergeEpoch_SL_Died_vs_Survived']['A']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make a function which outputs the pval histogram, qval table, min qval table\n",
    "def make_hla_assoc_outputs(res, output_directory):\n",
    "    \n",
    "    # pval histogram\n",
    "    all_res = pd.concat(res.values(), ignore_index=True) \n",
    "    all_res.to_csv(output_directory + \"combined_HLAs.tsv\", sep = '\\t', index = False)\n",
    "    (fig,ax) = plt.subplots(1,1)\n",
    "    plt.hist(all_res['p.value'])\n",
    "    ax.set_xlabel('PVAL')\n",
    "    ax.set_title('PVAL Histogram')\n",
    "    plt.savefig(output_directory + \"pvalhist.png\",\n",
    "               bbox_inches = 'tight')\n",
    "    \n",
    "    # qval table\n",
    "    qval_input = all_res[['MarkerID', 'gene', 'p.value']].dropna(axis=0)\n",
    "    qval_input['qval'] = sm.fdrcorrection(qval_input['p.value'], alpha = 0.05)[1]\n",
    "    qval_input = qval_input.reset_index()\n",
    "    qval_input.to_csv(output_directory + \"qvals.tsv\", sep = '\\t')\n",
    "    \n",
    "    # min qval table\n",
    "    print(qval_input)\n",
    "    qval_final = qval_input.loc[qval_input.groupby('gene')['qval'].idxmin()][['gene', 'qval', 'MarkerID']]\n",
    "    print(qval_final)\n",
    "    qval_final.to_csv(output_directory + \"qval_min.tsv\", sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    index   MarkerID  gene   p.value      qval\n",
      "0       0     34:02A     A  0.731428  0.940138\n",
      "1       1     36:01A     A  0.367203  0.788415\n",
      "2       2     23:01A     A  0.094868  0.788415\n",
      "3       3     02:02A     A  0.956819  0.980150\n",
      "4       4     30:02A     A  0.563731  0.921427\n",
      "..    ...        ...   ...       ...       ...\n",
      "66     66  08:04DRB1  DRB1  0.782302  0.940138\n",
      "67     67  01:02DRB1  DRB1  0.974338  0.980150\n",
      "68     68  07:01DRB1  DRB1  0.932001  0.980150\n",
      "69     69  11:01DRB1  DRB1  0.032952  0.741010\n",
      "70     70  12:01DRB1  DRB1  0.624844  0.924249\n",
      "\n",
      "[71 rows x 5 columns]\n",
      "    gene      qval   MarkerID\n",
      "1      A  0.788415     36:01A\n",
      "13     B  0.741010     57:03B\n",
      "22     C  0.788415     04:01C\n",
      "34  DPA1  0.788415  02:01DPA1\n",
      "40  DPB1  0.663026  13:01DPB1\n",
      "44  DQA1  0.788415  01:02DQA1\n",
      "53  DQB1  0.788415  05:02DQB1\n",
      "69  DRB1  0.741010  11:01DRB1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    index   MarkerID  gene   p.value      qval\n",
      "0       0     23:01A     A  0.306155  0.640143\n",
      "1       1     02:01A     A  0.742640  0.853971\n",
      "2       2     35:01B     B  0.798847  0.853971\n",
      "3       3     53:01B     B  0.080190  0.519183\n",
      "4       4     04:01C     C  0.276768  0.636568\n",
      "5       5     17:01C     C  0.090293  0.519183\n",
      "6       6     16:01C     C  0.480072  0.788690\n",
      "7       7  01:03DPA1  DPA1  0.550498  0.844096\n",
      "8       8  02:01DPA1  DPA1  0.962620  0.962620\n",
      "9       9  03:01DPA1  DPA1  0.136209  0.622513\n",
      "10     10  02:02DPA1  DPA1  0.478733  0.788690\n",
      "11     11  01:01DPB1  DPB1  0.790833  0.853971\n",
      "12     12  01:02DQA1  DQA1  0.658926  0.853971\n",
      "13     13  04:01DQA1  DQA1  0.816842  0.853971\n",
      "14     14  01:01DQA1  DQA1  0.226208  0.636568\n",
      "15     15  05:05DQA1  DQA1  0.073178  0.519183\n",
      "16     16  03:03DQA1  DQA1  0.259180  0.636568\n",
      "17     17  02:02DQB1  DQB1  0.184697  0.622513\n",
      "18     18  04:02DQB1  DQB1  0.669156  0.853971\n",
      "19     19  03:19DQB1  DQB1  0.021013  0.483304\n",
      "20     20  05:01DQB1  DQB1  0.676470  0.853971\n",
      "21     21  13:02DRB1  DRB1  0.189461  0.622513\n",
      "22     22  03:02DRB1  DRB1  0.445749  0.788690\n",
      "    gene      qval   MarkerID\n",
      "0      A  0.640143     23:01A\n",
      "3      B  0.519183     53:01B\n",
      "5      C  0.519183     17:01C\n",
      "9   DPA1  0.622513  03:01DPA1\n",
      "11  DPB1  0.853971  01:01DPB1\n",
      "15  DQA1  0.519183  05:05DQA1\n",
      "19  DQB1  0.483304  03:19DQB1\n",
      "21  DRB1  0.622513  13:02DRB1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     index   MarkerID  gene   p.value      qval\n",
      "0        0     34:02A     A  0.346949  0.778998\n",
      "1        1     36:01A     A  0.282472  0.770469\n",
      "2        2     68:01A     A  0.775235  0.922583\n",
      "3        3     23:01A     A  0.791390  0.931895\n",
      "4        4     02:02A     A  0.423846  0.829009\n",
      "..     ...        ...   ...       ...       ...\n",
      "114    114  01:02DRB1  DRB1  0.123464  0.745153\n",
      "115    115  16:02DRB1  DRB1  0.923632  0.981358\n",
      "116    116  07:01DRB1  DRB1  0.521110  0.891936\n",
      "117    117  11:01DRB1  DRB1  0.072449  0.712771\n",
      "118    118  12:01DRB1  DRB1  0.725511  0.922583\n",
      "\n",
      "[119 rows x 5 columns]\n",
      "     gene      qval    MarkerID\n",
      "5       A  0.712771      30:02A\n",
      "31      B  0.491231      39:10B\n",
      "42      C  0.635436      14:02C\n",
      "61   DPA1  0.712771   01:03DPA1\n",
      "69   DPB1  0.491231  104:01DPB1\n",
      "83   DQA1  0.712771   05:01DQA1\n",
      "93   DQB1  0.712771   02:01DQB1\n",
      "102  DRB1  0.712771   13:03DRB1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     index   MarkerID  gene   p.value      qval\n",
      "0        0     34:02A     A  0.464732  0.886674\n",
      "1        1     68:01A     A  0.758947  0.959589\n",
      "2        2     23:01A     A  0.239162  0.771123\n",
      "3        3     02:02A     A  0.066871  0.596957\n",
      "4        4     30:02A     A  0.138451  0.751860\n",
      "..     ...        ...   ...       ...       ...\n",
      "105    105  01:02DRB1  DRB1  0.365223  0.837776\n",
      "106    106  16:02DRB1  DRB1  0.220806  0.771123\n",
      "107    107  07:01DRB1  DRB1  0.886367  0.993592\n",
      "108    108  11:01DRB1  DRB1  0.516072  0.930622\n",
      "109    109  12:01DRB1  DRB1  0.634788  0.938529\n",
      "\n",
      "[110 rows x 5 columns]\n",
      "    gene      qval   MarkerID\n",
      "3      A  0.596957     02:02A\n",
      "17     B  0.596957     58:01B\n",
      "49     C  0.596957     03:02C\n",
      "55  DPA1  0.596957  02:01DPA1\n",
      "59  DPB1  0.751860  04:01DPB1\n",
      "71  DQA1  0.596957  01:05DQA1\n",
      "85  DQB1  0.596957  04:02DQB1\n",
      "97  DRB1  0.596957  08:06DRB1\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAh8AAAHFCAYAAABSEJsFAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAlGklEQVR4nO3de3TU9Z3/8deQhCEJSRQqkJiYQEUJUBUIdxERjBdA0LZiQcolVly8oKxVOFggtiVeWA5WK640BFQSQYXIAirZrVyFyrW64soioEGMlEsuBo2QfH5/eDL7GxOEpDPvMOH5OGf+mM98Z76f+ZzIPP1+5+JxzjkBAAAYadLQEwAAAOcX4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAzC0cOFCeTwe3yU8PFyJiYkaN26cvvjiC0nSQw89JI/Ho//5n/857eNMmzZNHo9HO3bs8I0dOXJEXq9XHo9H27Ztq/V+Y8eOVfPmzes87wMHDsjj8Wj27Nm13j579mx5PB4dOHDAb18pKSl12s+hQ4c0c+ZM7dq1q85zBBA6iA+gAeTk5Gjz5s0qKCjQb37zG+Xl5alfv34qLy9XRkaGJGnBggW13reqqkovvfSSrrrqKnXt2tU3/vLLL+u7776TJGVnZwf/SZzB7373Oy1fvrxO9zl06JAyMzOJD6CRIz6ABtC5c2f16tVLAwYM0IwZM/TII49o//79ys/PV+fOndWjRw+9/PLLOnXqVI37rlmzRgcPHvRFSrUFCxaoVatW6t69u/Ly8vTNN99YPZ1a/fSnP1WXLl0adA515Zxr8HUDzgfEB3AO6NWrlyTps88+kyRlZGSoqKhIb731Vo1tc3Jy5PV6NWrUKN/Y3/72N/33f/+3Ro8erd/85jcqKSnRG2+8YTP506jttMtrr72mnj17Ki4uTlFRUWrXrp3Gjx8vSVq7dq26d+8uSRo3bpzv1NTMmTN991+xYoV69+6tqKgoxcTE6Prrr9fmzZtr7PvNN9/UFVdcIa/Xq3bt2umZZ57RzJkz5fF4/LbzeDy677779MILLyg1NVVer1eLFi2SJGVmZqpnz55q0aKFYmNj1bVrV2VnZ+uHv8WZkpKiIUOGaOXKlerSpYsiIyOVmpqqlStXSvr+VFtqaqqio6PVo0eP054SA84nxAdwDti7d68k6aKLLpIk/epXv1JUVFSNUy/Hjx/Xm2++qVtvvVUXXnihb7z6NMv48eN1xx13KCoqKiinXqqqqnTq1Kkal6qqqjPed/PmzRoxYoTatWunV199VatWrdL06dN9R3e6du2qnJwcSdJjjz2mzZs3a/PmzbrrrrskSbm5uRo2bJhiY2OVl5en7OxsHT9+XNdee602btzo28/bb7+t2267TS1bttSSJUv01FNPKS8vzxcVP5Sfn6958+Zp+vTpeuedd9SvXz9J37/PZcKECVq6dKmWLVum2267Tffff79+//vf13iMv//975o6daoeffRRLVu2THFxcbrttts0Y8YM/eUvf9GsWbO0ePFilZSUaMiQIRxdARwAMzk5OU6S27Jlizt58qQrKytzK1eudBdddJGLiYlxRUVFvm3HjBnjIiIi3FdffeUbe/bZZ50kV1BQ4BsrLy93sbGxrlevXn739Xg8bu/evX77HzNmjIuOjq7zvPfv3+8knfGyf/9+v30lJyf7rs+ePdtJcsXFxafdz9atW50kl5OT4zdeWVnpEhIS3M9+9jNXWVnpGy8rK3OtWrVyffr08Y11797dJSUluYqKCr/tWrZs6X74T54kFxcX544dO/ajz7+ystKdPHnSPf74465ly5auqqrKd1tycrKLjIx0Bw8e9I3t2rXLSXLx8fGuvLzcN56fn+8kuRUrVvzo/oDGjiMfQAPo1auXIiIiFBMToyFDhqhNmzZ666231Lp1a982GRkZOnnypF5++WXfWE5OjpKTkzVw4EDf2NKlS1VaWuo7fSF9fwTEOec7khAokyZN0tatW2tcJk2adMb7Vp9Suf3227V06VLfp3vOxieffKJDhw5p9OjRatLk//7Zat68uX7+859ry5YtOnHihMrLy7Vt2zYNHz5cTZs29dtu6NChtT72dddd53cUqdpf//pXDRo0SHFxcQoLC1NERISmT5+uo0eP6vDhw37bXnXVVbr44ot911NTUyVJ1157raKiomqMV59eA85XxAfQAF566SVt3bpVO3fu1KFDh/TBBx+ob9++ftv069dPl112mS8gPvjgA+3YscP3fohq2dnZatasmW688UYVFxeruLhYV1xxhVJSUrRw4UJVVlYGbN6JiYlKS0urcUlMTDzjfa+55hrl5+fr1KlT+vWvf63ExER17txZeXl5Z7zv0aNHJUnx8fE1bktISFBVVZWOHz+u48ePyznnF3HVahs73WO+//77Sk9PlyTNnz9fmzZt0tatWzVt2jRJqnHapEWLFn7Xq8PndOPffvttrXMBzhfhDT0B4HyUmpqqtLS0M243fvx4TZkyRe+//75yc3PVpEkTjR071nf7nj17fO93uOSSS2p9jHfeeUc333xzQOb9zxo2bJiGDRumiooKbdmyRVlZWRo5cqRSUlLUu3fv096vZcuWkqQvv/yyxm2HDh1SkyZNdOGFF8o5J4/Ho6+++qrGdkVFRbU+9g/fhCpJr776qiIiIrRy5Uo1a9bMN56fn3+mpwjgLHDkAziHjRkzRuHh4fr3f/93LV68WAMHDlRycrLv9uo3lc6fP1/vvvuu32X16tWKiIg47feFNCSv16v+/fvrySeflCTt3LnTNy7VPLJw+eWX6+KLL1Zubq7fp03Ky8v1xhtv+D4BEx0drbS0NOXn5/u+80SSvv76a9+nT85G9RfAhYWF+ca++eYbv1NgAOqPIx/AOaxNmza6+eablZOTI+ec33d7nDp1Si+99JJSU1N9nwj5oaFDh2rFihX6xz/+4fskTWVlpV5//fUa20ZHR+umm24KzhORNH36dB08eFADBw5UYmKiiouL9cwzzygiIkL9+/eX9P13g0RGRmrx4sVKTU1V8+bNlZCQoISEBD311FMaNWqUhgwZogkTJqiiokJPP/20iouL9cQTT/j28/jjj2vw4MG64YYbNGnSJFVWVurpp59W8+bNdezYsbOa6+DBgzVnzhyNHDlSd999t44eParZs2f74gjAP4cjH8A5LiMjQ845tWjRQsOHD/eNr1q1SkVFRZowYcJp73v33XfXeNPqt99+q1/+8pc1Lv/yL/8SzKehnj17qqioSI8++qjS09N19913KzIyUn/961/VqVMnSfJ9vPjo0aNKT09X9+7d9eKLL0qSRo4cqfz8fB09elQjRozQuHHjFBsbq3fffVdXX321bz833nij3njjDd92kydP1q233qphw4bpggsuOKu5XnfddVqwYIE+/PBDDR06VNOmTdMvfvELTZkyJeDrApyPPM794BtzAKCROXnypO8TKWvWrGno6QDnPU67AGh0MjIydP311ys+Pl5FRUV64YUX9PHHH+uZZ55p6KkBEPEBoBEqKyvTww8/rH/84x+KiIhQ165dtXr1ag0aNKihpwZAnHYBAADGeMMpAAAwRXwAAABTxAcAADB1zr3htKqqSocOHVJMTEytX3sMAADOPc45lZWVKSEhwe8HIGtzzsXHoUOHlJSU1NDTAAAA9VBYWHjGH5s85+IjJiZG0veTj42NbeDZAACAs1FaWqqkpCTf6/iPOefio/pUS2xsLPEBAECIOZu3TPCGUwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACm6hwf69ev19ChQ5WQkCCPx6P8/Hy/251zmjlzphISEhQZGalrr71WH330UaDmCwAAQlyd46O8vFxXXnmlnnvuuVpvf+qppzRnzhw999xz2rp1q9q0aaPrr79eZWVl//RkAQBA6KvzD8vddNNNuummm2q9zTmnuXPnatq0abrtttskSYsWLVLr1q2Vm5urCRMm/HOzBQAAIS+g7/nYv3+/ioqKlJ6e7hvzer3q37+/3nvvvUDuCgAAhKg6H/n4MUVFRZKk1q1b+423bt1an332Wa33qaioUEVFhe96aWlpIKcEAADOMQGNj2oej8fvunOuxli1rKwsZWZmBmMaaEApU1Y19BTq7MATgxt6CgBwXgjoaZc2bdpI+r8jINUOHz5c42hItalTp6qkpMR3KSwsDOSUAADAOSag8dG2bVu1adNGBQUFvrHvvvtO69atU58+fWq9j9frVWxsrN8FAAA0XnU+7fL1119r7969vuv79+/Xrl271KJFC11yySV68MEHNWvWLLVv317t27fXrFmzFBUVpZEjRwZ04gAAIDTVOT62bdumAQMG+K5PnjxZkjRmzBgtXLhQjzzyiL755htNnDhRx48fV8+ePbVmzRrFxMQEbtYAACBkeZxzrqEn8f8rLS1VXFycSkpKOAUTwnjDKQCcX+ry+s1vuwAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABT4Q09AeBckTJlVUNPoc4OPDG4oacAAHXGkQ8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmAh4fp06d0mOPPaa2bdsqMjJS7dq10+OPP66qqqpA7woAAISg8EA/4JNPPqkXXnhBixYtUqdOnbRt2zaNGzdOcXFxmjRpUqB3BwAAQkzA42Pz5s0aNmyYBg8eLElKSUlRXl6etm3bFuhdAQCAEBTw0y5XX321/uu//kt79uyRJP3973/Xxo0bdfPNN9e6fUVFhUpLS/0uAACg8Qr4kY9HH31UJSUl6tChg8LCwlRZWak//vGP+tWvflXr9llZWcrMzAz0NE4rZcoqs30FyoEnBjf0FACEoFD89y4U8W903QX8yMeSJUv0yiuvKDc3Vzt27NCiRYs0e/ZsLVq0qNbtp06dqpKSEt+lsLAw0FMCAADnkIAf+fjtb3+rKVOm6I477pAk/exnP9Nnn32mrKwsjRkzpsb2Xq9XXq830NMAAADnqIAf+Thx4oSaNPF/2LCwMD5qCwAAJAXhyMfQoUP1xz/+UZdccok6deqknTt3as6cORo/fnygdwUAAEJQwOPj2Wef1e9+9ztNnDhRhw8fVkJCgiZMmKDp06cHelcAACAEBTw+YmJiNHfuXM2dOzfQDw0AABoBftsFAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgKnwhp4AgPpLmbKqoadQZweeGNzQUwDQwDjyAQAATBEfAADAFPEBAABMER8AAMAU8QEAAEwRHwAAwBTxAQAATBEfAADAFPEBAABMER8AAMAU8QEAAEwRHwAAwBTxAQAATBEfAADAFPEBAABMER8AAMAU8QEAAEwRHwAAwBTxAQAATBEfAADAFPEBAABMER8AAMAU8QEAAEwRHwAAwBTxAQAATBEfAADAFPEBAABMER8AAMAU8QEAAEwRHwAAwFRQ4uOLL77QnXfeqZYtWyoqKkpXXXWVtm/fHoxdAQCAEBMe6Ac8fvy4+vbtqwEDBuitt95Sq1at9Omnn+qCCy4I9K4AAEAICnh8PPnkk0pKSlJOTo5vLCUlJdC7AQAAISrgp11WrFihtLQ0/fKXv1SrVq3UpUsXzZ8/P9C7AQAAISrg8bFv3z7NmzdP7du31zvvvKN77rlHDzzwgF566aVat6+oqFBpaanfBQAANF4BP+1SVVWltLQ0zZo1S5LUpUsXffTRR5o3b55+/etf19g+KytLmZmZgZ5Go5IyZVVDTwEAcBqh+G/0gScGN+j+A37kIz4+Xh07dvQbS01N1eeff17r9lOnTlVJSYnvUlhYGOgpAQCAc0jAj3z07dtXn3zyid/Ynj17lJycXOv2Xq9XXq830NMAAADnqIAf+XjooYe0ZcsWzZo1S3v37lVubq5efPFF3XvvvYHeFQAACEEBj4/u3btr+fLlysvLU+fOnfX73/9ec+fO1ahRowK9KwAAEIICftpFkoYMGaIhQ4YE46EBAECI47ddAACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgKnwhp4AgPNLypRVDT2FejnwxOCGngLQaHDkAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgCniAwAAmCI+AACAKeIDAACYIj4AAIAp4gMAAJgiPgAAgKmgx0dWVpY8Ho8efPDBYO8KAACEgKDGx9atW/Xiiy/qiiuuCOZuAABACAlafHz99dcaNWqU5s+frwsvvDBYuwEAACEmaPFx7733avDgwRo0aNCPbldRUaHS0lK/CwAAaLzCg/Ggr776qnbs2KGtW7eecdusrCxlZmYGYxoAEDApU1Y19BSARiPgRz4KCws1adIkvfLKK2rWrNkZt586dapKSkp8l8LCwkBPCQAAnEMCfuRj+/btOnz4sLp16+Ybq6ys1Pr16/Xcc8+poqJCYWFhvtu8Xq+8Xm+gpwEAAM5RAY+PgQMH6sMPP/QbGzdunDp06KBHH33ULzwAAMD5J+DxERMTo86dO/uNRUdHq2XLljXGAQDA+YdvOAUAAKaC8mmXH1q7dq3FbgAAQAjgyAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwFfD4yMrKUvfu3RUTE6NWrVpp+PDh+uSTTwK9GwAAEKICHh/r1q3Tvffeqy1btqigoECnTp1Senq6ysvLA70rAAAQgsID/YBvv/223/WcnBy1atVK27dv1zXXXBPo3QEAgBAT8Pj4oZKSEklSixYtar29oqJCFRUVvuulpaXBnhIAAGhAQX3DqXNOkydP1tVXX63OnTvXuk1WVpbi4uJ8l6SkpGBOCQAANLCgxsd9992nDz74QHl5eafdZurUqSopKfFdCgsLgzklAADQwIJ22uX+++/XihUrtH79eiUmJp52O6/XK6/XG6xpAACAc0zA48M5p/vvv1/Lly/X2rVr1bZt20DvAgAAhLCAx8e9996r3Nxcvfnmm4qJiVFRUZEkKS4uTpGRkYHeHQAACDEBf8/HvHnzVFJSomuvvVbx8fG+y5IlSwK9KwAAEIKCctoFAADgdPhtFwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGCK+AAAAKaIDwAAYIr4AAAApogPAABgivgAAACmiA8AAGAqaPHx/PPPq23btmrWrJm6deumDRs2BGtXAAAghAQlPpYsWaIHH3xQ06ZN086dO9WvXz/ddNNN+vzzz4OxOwAAEEKCEh9z5sxRRkaG7rrrLqWmpmru3LlKSkrSvHnzgrE7AAAQQgIeH9999522b9+u9PR0v/H09HS99957gd4dAAAIMeGBfsAjR46osrJSrVu39htv3bq1ioqKamxfUVGhiooK3/WSkhJJUmlpaaCnJkmqqjgRlMcFACBUBOM1tvoxnXNn3Dbg8VHN4/H4XXfO1RiTpKysLGVmZtYYT0pKCtbUAAA4r8XNDd5jl5WVKS4u7ke3CXh8/OQnP1FYWFiNoxyHDx+ucTREkqZOnarJkyf7rldVVenYsWNq2bJlrbFSF6WlpUpKSlJhYaFiY2P/qcfC2WHNbbHe9lhze6y5vfqsuXNOZWVlSkhIOOO2AY+Ppk2bqlu3biooKNCtt97qGy8oKNCwYcNqbO/1euX1ev3GLrjggoDOKTY2lj9YY6y5LdbbHmtujzW3V9c1P9MRj2pBOe0yefJkjR49Wmlpaerdu7defPFFff7557rnnnuCsTsAABBCghIfI0aM0NGjR/X444/ryy+/VOfOnbV69WolJycHY3cAACCEBO0NpxMnTtTEiROD9fBnxev1asaMGTVO6yB4WHNbrLc91twea24v2GvucWfzmRgAAIAA4YflAACAKeIDAACYIj4AAIAp4gMAAJgK6fh4/vnn1bZtWzVr1kzdunXThg0bfnT7devWqVu3bmrWrJnatWunF154wWimjUdd1nzZsmW6/vrrddFFFyk2Nla9e/fWO++8YzjbxqGuf+fVNm3apPDwcF111VXBnWAjVNc1r6io0LRp05ScnCyv16uf/vSnWrBggdFsG4e6rvnixYt15ZVXKioqSvHx8Ro3bpyOHj1qNNvQt379eg0dOlQJCQnyeDzKz88/430C+hrqQtSrr77qIiIi3Pz5893u3bvdpEmTXHR0tPvss89q3X7fvn0uKirKTZo0ye3evdvNnz/fRUREuNdff9145qGrrms+adIk9+STT7r333/f7dmzx02dOtVFRES4HTt2GM88dNV1zasVFxe7du3aufT0dHfllVfaTLaRqM+a33LLLa5nz56uoKDA7d+/3/3tb39zmzZtMpx1aKvrmm/YsME1adLEPfPMM27fvn1uw4YNrlOnTm748OHGMw9dq1evdtOmTXNvvPGGk+SWL1/+o9sH+jU0ZOOjR48e7p577vEb69Chg5syZUqt2z/yyCOuQ4cOfmMTJkxwvXr1CtocG5u6rnltOnbs6DIzMwM9tUarvms+YsQI99hjj7kZM2YQH3VU1zV/6623XFxcnDt69KjF9Bqluq75008/7dq1a+c39qc//cklJiYGbY6N2dnER6BfQ0PytMt3332n7du3Kz093W88PT1d7733Xq332bx5c43tb7jhBm3btk0nT54M2lwbi/qs+Q9VVVWprKxMLVq0CMYUG536rnlOTo4+/fRTzZgxI9hTbHTqs+YrVqxQWlqannrqKV188cW67LLL9PDDD+ubb76xmHLIq8+a9+nTRwcPHtTq1avlnNNXX32l119/XYMHD7aY8nkp0K+hQfuG02A6cuSIKisra/xKbuvWrWv8mm61oqKiWrc/deqUjhw5ovj4+KDNtzGoz5r/0L/927+pvLxct99+ezCm2OjUZ83/93//V1OmTNGGDRsUHh6S/3k3qPqs+b59+7Rx40Y1a9ZMy5cv15EjRzRx4kQdO3aM932chfqseZ8+fbR48WKNGDFC3377rU6dOqVbbrlFzz77rMWUz0uBfg0NySMf1Twej99151yNsTNtX9s4Tq+ua14tLy9PM2fO1JIlS9SqVatgTa9ROts1r6ys1MiRI5WZmanLLrvManqNUl3+zquqquTxeLR48WL16NFDN998s+bMmaOFCxdy9KMO6rLmu3fv1gMPPKDp06dr+/btevvtt7V//35+vDTIAvkaGpL/a/STn/xEYWFhNar48OHDNcqsWps2bWrdPjw8XC1btgzaXBuL+qx5tSVLligjI0OvvfaaBg0aFMxpNip1XfOysjJt27ZNO3fu1H333Sfp+xdG55zCw8O1Zs0aXXfddSZzD1X1+TuPj4/XxRdf7PdT4qmpqXLO6eDBg2rfvn1Q5xzq6rPmWVlZ6tu3r377299Kkq644gpFR0erX79++sMf/sCR7CAI9GtoSB75aNq0qbp166aCggK/8YKCAvXp06fW+/Tu3bvG9mvWrFFaWpoiIiKCNtfGoj5rLn1/xGPs2LHKzc3lfGwd1XXNY2Nj9eGHH2rXrl2+yz333KPLL79cu3btUs+ePa2mHrLq83fet29fHTp0SF9//bVvbM+ePWrSpIkSExODOt/GoD5rfuLECTVp4v/yFRYWJun//m8cgRXw19B6vU31HFD90azs7Gy3e/du9+CDD7ro6Gh34MAB55xzU6ZMcaNHj/ZtX/0xoYceesjt3r3bZWdn81HbOqrrmufm5rrw8HD35z//2X355Ze+S3FxcUM9hZBT1zX/IT7tUnd1XfOysjKXmJjofvGLX7iPPvrIrVu3zrVv397dddddDfUUQk5d1zwnJ8eFh4e7559/3n366adu48aNLi0tzfXo0aOhnkLIKSsrczt37nQ7d+50ktycOXPczp07fR9vDvZraMjGh3PO/fnPf3bJycmuadOmrmvXrm7dunW+28aMGeP69+/vt/3atWtdly5dXNOmTV1KSoqbN2+e8YxDX13WvH///k5SjcuYMWPsJx7C6vp3/v8jPuqnrmv+8ccfu0GDBrnIyEiXmJjoJk+e7E6cOGE869BW1zX/05/+5Dp27OgiIyNdfHy8GzVqlDt48KDxrEPXu++++6P/Pgf7NdTjHMeoAACAnZB8zwcAAAhdxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfACos7Fjx8rj8cjj8SgiIkLt2rXTww8/rE2bNsnj8Wjjxo213u+GG27QLbfc4rt+8OBBNW3aVB06dKh1e4/Ho/z8/GA8BQANiPgAUC833nijvvzyS+3bt09/+MMf9Pzzz+uVV17RlVdeqZycnBrbFxYW6j//8z+VkZHhG1u4cKFuv/12nThxQps2bbKcPoAGRHwAqBev16s2bdooKSlJI0eO1KhRo5Sfn6+MjAwtXbpU5eXlftsvXLhQF110ke/XjZ1zysnJ0ejRozVy5EhlZ2c3xNMA0ACIDwABERkZqZMnT2rUqFE6efKkXnvtNd9tzjktXLhQY8aMUXh4uCTp3Xff1YkTJzRo0CCNHj1aS5cuVVlZWUNNH4Ah4gPAP+39999Xbm6uBg4cqBYtWmj48OF+p17Wrl2rffv2afz48b6x7Oxs3XHHHQoLC1OnTp106aWXasmSJQ0xfQDGiA8A9bJy5Uo1b95czZo1U+/evXXNNdfo2WeflSRlZGRo/fr12rt3ryRpwYIF6tu3ry6//HJJUnFxsZYtW6Y777zT93h33nmnFixYYP9EAJgLb+gJAAhNAwYM0Lx58xQREaGEhARFRET4bhs0aJCSk5O1cOFCPfLII1q2bJmee+453+25ubn69ttv1bNnT9+Yc05VVVXavXu3OnbsaPpcANgiPgDUS3R0tC699NJab/N4PBo3bpz+8pe/KDExUU2aNNHtt9/uuz07O1v/+q//qrFjx/rd74EHHtCCBQs0e/bsYE4dQAPzOOdcQ08CQGgZO3asiouLf/Q7OD7//HO1bdtWcXFx+vnPf6758+dLknbt2qUuXbro448/rvH9HvPnz9e0adP0xRdfKCIiQh6PR3PmzNGAAQP8trv00kvVvHnzgD8vADZ4zweAoLjkkks0aNAgHT9+vMYbTTt27FjrF4sNHz5cx44d03/8x3/4xiZPnqwuXbr4XbZt22byHAAEB0c+AACAKY58AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMEV8AAAAU8QHAAAwRXwAAABTxAcAADBFfAAAAFPEBwAAMPX/AMse45jOrW1NAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAioAAAHFCAYAAADcytJ5AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAvf0lEQVR4nO3de1xVZb7H8e9WcIMXKElxE94YixCPpmBKjVphmJhpOSfn6JiaXTxdNDkeE51TUTOHpji91C5yNC7j3YokTTM9rxGywkkUJ89oVpMKKuRoCUS5ubjOH73cZ3bgZSOXh83n/XqtP9aznmev33pe1P76rLX3tlmWZQkAAMBAbZq7AAAAgAshqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAM0oMzNTNpvNtfn4+Cg0NFTTp0/X8ePHJUlz5syRzWbT559/fsHXWbhwoWw2m/bu3etqO3XqlOx2u2w2m/Lz8+scN23aNHXs2NHjuo8cOSKbzaaUlJQ6j6ekpMhms+nIkSNu5+rVq5dH5zlx4oSeffZZ7du3z+MaAXgHggpggIyMDOXl5Wn79u166KGHtHbtWg0bNkwVFRWaMWOGJCk9Pb3OsefOndOKFSt04403atCgQa72lStXqrKyUpKUlpbW+BdxCf/xH/+hDRs2eDTmxIkTSkpKIqgArRhBBTBAv379NHToUN1222165plnNG/ePB0+fFjZ2dnq16+fbrrpJq1cuVLV1dW1xm7btk3Hjh1zBZrz0tPT1bVrVw0ePFhr167Vjz/+2FSXU6df/OIXGjhwYLPW4CnLspp93oDWjqACGGjo0KGSpKNHj0qSZsyYoZKSEr3//vu1+mZkZMhut2vy5Mmutj//+c/63//9X02ZMkUPPfSQSktLlZWV1TTFX0Bdt37eeustDRkyRIGBgWrfvr3CwsL0wAMPSJJycnI0ePBgSdL06dNdt8eeffZZ1/iNGzcqJiZG7du3V6dOnXTHHXcoLy+v1rnfffdd9e/fX3a7XWFhYVq8eLGeffZZ2Ww2t342m02PP/64UlNTFRERIbvdrj/+8Y+SpKSkJA0ZMkSdO3dWQECABg0apLS0NP38d1179eqlu+66S++9954GDhwof39/RURE6L333pP00+2+iIgIdejQQTfddNMFb8sB+AlBBTDQV199JUnq0qWLJOlf/uVf1L59+1q3f7777ju9++67uueee3T11Ve72s/f6nnggQf061//Wu3bt2+U2z/nzp1TdXV1re3cuXOXHJuXl6eJEycqLCxM69at0+bNm/X000+7Vo0GDRqkjIwMSdJvf/tb5eXlKS8vTw8++KAkac2aNRo3bpwCAgK0du1apaWl6bvvvtOtt96qjz76yHWerVu36t5771VQUJDWr1+vF198UWvXrnUFkJ/Lzs7W0qVL9fTTT+uDDz7QsGHDJP30XM4jjzyiN998U++8847uvfdePfHEE3r++edrvcZf/vIXJSYm6qmnntI777yjwMBA3XvvvXrmmWf0xhtv6D//8z+1evVqlZaW6q677mLVBrgYC0CzycjIsCRZu3btsqqqqqzy8nLrvffes7p06WJ16tTJKikpcfWdOnWq5evra33zzTeutldeecWSZG3fvt3VVlFRYQUEBFhDhw51G2uz2ayvvvrK7fxTp061OnTo4HHdhw8ftiRdcjt8+LDbuXr27OnaT0lJsSRZZ86cueB5du/ebUmyMjIy3NpramqskJAQ65/+6Z+smpoaV3t5ebnVtWtX6+abb3a1DR482OrevbvldDrd+gUFBVk//1+gJCswMND69ttvL3r9NTU1VlVVlfXcc89ZQUFB1rlz51zHevbsafn7+1vHjh1zte3bt8+SZDkcDquiosLVnp2dbUmyNm7ceNHzAa0ZKyqAAYYOHSpfX1916tRJd911l7p166b3339fwcHBrj4zZsxQVVWVVq5c6WrLyMhQz549FRsb62p78803VVZW5rqFIv20smJZlmuFoqHMnj1bu3fvrrXNnj37kmPP39a577779Oabb7o+5XQ5Dh06pBMnTmjKlClq0+b//zfWsWNHTZgwQbt27dIPP/ygiooK5efna/z48WrXrp1bv7Fjx9b52rfffrvb6tR5f/rTnzRy5EgFBgaqbdu28vX11dNPP63Tp0/r5MmTbn1vvPFGXXvtta79iIgISdKtt96q9u3b12o/f4sPQG0EFcAAK1as0O7du1VQUKATJ07os88+0y233OLWZ9iwYbr++utdYeOzzz7T3r17Xc9vnJeWliY/Pz/deeedOnPmjM6cOaP+/furV69eyszMVE1NTYPVHRoaqujo6FpbaGjoJccOHz5c2dnZqq6u1v3336/Q0FD169dPa9euveTY06dPS5IcDketYyEhITp37py+++47fffdd7Isyy3wnVdX24Ve89NPP1VcXJwkafny5fr444+1e/duLVy4UJJq3brp3Lmz2/75kHSh9rNnz9ZZCwDJp7kLAPDTv6yjo6Mv2e+BBx7Q/Pnz9emnn2rNmjVq06aNpk2b5jr+xRdfuJ7P6NGjR52v8cEHHyg+Pr5B6r5S48aN07hx4+R0OrVr1y4lJydr0qRJ6tWrl2JiYi44LigoSJJUXFxc69iJEyfUpk0bXX311bIsSzabTd98802tfiUlJXW+9s8fsJWkdevWydfXV++99578/Pxc7dnZ2Ze6RABXiBUVoAWZOnWqfHx89N///d9avXq1YmNj1bNnT9fx8w/MLl++XDt27HDbtmzZIl9f3wt+H0tzstvtGjFihP7whz9IkgoKClztUu0Vi/DwcF177bVas2aN26duKioqlJWV5fokUIcOHRQdHa3s7GzXd8pI0vfff+/6FM7lOP9lfG3btnW1/fjjj2634QA0DlZUgBakW7duio+PV0ZGhizLcvvulOrqaq1YsUIRERGuT8b83NixY7Vx40b9/e9/d32iqKamRm+//Xatvh06dNDo0aMb50IkPf300zp27JhiY2MVGhqqM2fOaPHixfL19dWIESMk/fTdK/7+/lq9erUiIiLUsWNHhYSEKCQkRC+++KImT56su+66S4888oicTqdeeuklnTlzRi+88ILrPM8995zGjBmjUaNGafbs2aqpqdFLL72kjh076ttvv72sWseMGaOXX35ZkyZN0sMPP6zTp08rJSXFFaQANB5WVIAWZsaMGbIsS507d9b48eNd7Zs3b1ZJSYkeeeSRC459+OGHaz2Qe/bsWf3zP/9zre1f//VfG/MyNGTIEJWUlOipp55SXFycHn74Yfn7++tPf/qTIiMjJcn1kezTp08rLi5OgwcP1rJlyyRJkyZNUnZ2tk6fPq2JEydq+vTpCggI0I4dO/TLX/7SdZ4777xTWVlZrn4JCQm65557NG7cOF111VWXVevtt9+u9PR07d+/X2PHjtXChQv1q1/9SvPnz2/weQHgzmZZP/u2IgDwclVVVa5P5mzbtq25ywFwEdz6AeD1ZsyYoTvuuEMOh0MlJSVKTU3VwYMHtXjx4uYuDcAlEFQAeL3y8nLNnTtXf//73+Xr66tBgwZpy5YtGjlyZHOXBuASuPUDAACMxcO0AADAWAQVAABgLIIKAAAwVot4mPbcuXM6ceKEOnXqVOfXWwMAAPNYlqXy8nKFhIS4/YCoJ1pEUDlx4oS6d+/e3GUAAIB6KCoquqwfK61LiwgqnTp1kvTThQYEBDRzNQAA4HKUlZWpe/furvfx+mgRQeX87Z6AgACCCgAALcyVPLbBw7QAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYKwrCirJycmy2Wx68sknL9ovNzdXUVFR8vPzU1hYmFJTU6/ktAAAoJWod1DZvXu3li1bpv79+1+03+HDhxUfH69hw4apoKBACxYs0KxZs5SVlVXfUwMAgFaiXkHl+++/1+TJk7V8+XJdffXVF+2bmpqqHj16aNGiRYqIiNCDDz6oBx54QCkpKfUqGAAAtB71CiqPPfaYxowZo5EjR16yb15enuLi4tzaRo0apfz8fFVVVdXn9AAAoJXw8XTAunXrtHfvXu3evfuy+peUlCg4ONitLTg4WNXV1Tp16pQcDketMU6nU06n07VfVlbmaZkAAMALeBRUioqKNHv2bG3btk1+fn6XPc5ms7ntW5ZVZ/t5ycnJSkpK8qS0eus1f3OTnKchHXlhTHOXAABAk/Do1s+ePXt08uRJRUVFycfHRz4+PsrNzdWSJUvk4+OjmpqaWmO6deumkpISt7aTJ0/Kx8dHQUFBdZ4nMTFRpaWlrq2oqMiTMgEAgJfwaEUlNjZW+/fvd2ubPn26brjhBj311FNq27ZtrTExMTHatGmTW9u2bdsUHR0tX1/fOs9jt9tlt9s9KQ0AAHghj4JKp06d1K9fP7e2Dh06KCgoyNWemJio48ePa8WKFZKkmTNn6tVXX1VCQoIeeugh5eXlKS0tTWvXrm2gSwAAAN6qwb+Ztri4WIWFha793r17a8uWLcrJydGNN96o559/XkuWLNGECRMa+tQAAMDL2KzzT7YarKysTIGBgSotLVVAQECDvjYP0wIA0Dga4v2b3/oBAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMbyKKgsXbpU/fv3V0BAgAICAhQTE6P333//gv1zcnJks9lqbZ9//vkVFw4AALyfjyedQ0ND9cILL6hPnz6SpD/+8Y8aN26cCgoKFBkZecFxhw4dUkBAgGu/S5cu9SwXAAC0Jh4FlbFjx7rt//73v9fSpUu1a9euiwaVrl276qqrrqpXgQAAoPWq9zMqNTU1WrdunSoqKhQTE3PRvgMHDpTD4VBsbKx27NhR31MCAIBWxqMVFUnav3+/YmJidPbsWXXs2FEbNmxQ37596+zrcDi0bNkyRUVFyel0auXKlYqNjVVOTo6GDx9+wXM4nU45nU7XfllZmadlAgAAL+BxUAkPD9e+fft05swZZWVlaerUqcrNza0zrISHhys8PNy1HxMTo6KiIqWkpFw0qCQnJyspKcnT0gAAgJfx+NZPu3bt1KdPH0VHRys5OVkDBgzQ4sWLL3v80KFD9eWXX160T2JiokpLS11bUVGRp2UCAAAv4PGKys9ZluV2m+ZSCgoK5HA4LtrHbrfLbrdfaWkAAKCF8yioLFiwQKNHj1b37t1VXl6udevWKScnR1u3bpX000rI8ePHtWLFCknSokWL1KtXL0VGRqqyslKrVq1SVlaWsrKyGv5KAACA1/EoqHzzzTeaMmWKiouLFRgYqP79+2vr1q264447JEnFxcUqLCx09a+srNTcuXN1/Phx+fv7KzIyUps3b1Z8fHzDXgUAAPBKNsuyrOYu4lLKysoUGBio0tJSty+Oawi95m9u0NdrCkdeGNPcJQAAcEkN8f7Nb/0AAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGN5FFSWLl2q/v37KyAgQAEBAYqJidH7779/0TG5ubmKioqSn5+fwsLClJqaekUFAwCA1sOjoBIaGqoXXnhB+fn5ys/P1+23365x48bpr3/9a539Dx8+rPj4eA0bNkwFBQVasGCBZs2apaysrAYpHgAAeDcfTzqPHTvWbf/3v/+9li5dql27dikyMrJW/9TUVPXo0UOLFi2SJEVERCg/P18pKSmaMGFC/asGAACtQr2fUampqdG6detUUVGhmJiYOvvk5eUpLi7OrW3UqFHKz89XVVVVfU8NAABaCY9WVCRp//79iomJ0dmzZ9WxY0dt2LBBffv2rbNvSUmJgoOD3dqCg4NVXV2tU6dOyeFw1DnO6XTK6XS69svKyjwtEwAAeAGPg0p4eLj27dunM2fOKCsrS1OnTlVubu4Fw4rNZnPbtyyrzvZ/lJycrKSkJE9LazV6zd/c3CW0GkdeGNPcJXisJf59MM9Ng3luGi1xnk3m8a2fdu3aqU+fPoqOjlZycrIGDBigxYsX19m3W7duKikpcWs7efKkfHx8FBQUdMFzJCYmqrS01LUVFRV5WiYAAPACHq+o/JxlWW63af5RTEyMNm3a5Na2bds2RUdHy9fX94KvabfbZbfbr7Q0AADQwnm0orJgwQLt3LlTR44c0f79+7Vw4ULl5ORo8uTJkn5aCbn//vtd/WfOnKmjR48qISFBBw8eVHp6utLS0jR37tyGvQoAAOCVPFpR+eabbzRlyhQVFxcrMDBQ/fv319atW3XHHXdIkoqLi1VYWOjq37t3b23ZskVz5szRa6+9ppCQEC1ZsoSPJgMAgMviUVBJS0u76PHMzMxabSNGjNDevXs9KgoAAEDit34AAIDBCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLE8CirJyckaPHiwOnXqpK5du2r8+PE6dOjQRcfk5OTIZrPV2j7//PMrKhwAAHg/j4JKbm6uHnvsMe3atUvbt29XdXW14uLiVFFRccmxhw4dUnFxsWu77rrr6l00AABoHXw86bx161a3/YyMDHXt2lV79uzR8OHDLzq2a9euuuqqqzwuEAAAtF5X9IxKaWmpJKlz586X7Dtw4EA5HA7FxsZqx44dV3JaAADQSni0ovKPLMtSQkKCfvnLX6pfv34X7OdwOLRs2TJFRUXJ6XRq5cqVio2NVU5OzgVXYZxOp5xOp2u/rKysvmUCAIAWrN5B5fHHH9dnn32mjz766KL9wsPDFR4e7tqPiYlRUVGRUlJSLhhUkpOTlZSUVN/SAACAl6jXrZ8nnnhCGzdu1I4dOxQaGurx+KFDh+rLL7+84PHExESVlpa6tqKiovqUCQAAWjiPVlQsy9ITTzyhDRs2KCcnR717967XSQsKCuRwOC543G63y2631+u1AQCA9/AoqDz22GNas2aN3n33XXXq1EklJSWSpMDAQPn7+0v6aTXk+PHjWrFihSRp0aJF6tWrlyIjI1VZWalVq1YpKytLWVlZDXwpAADA23gUVJYuXSpJuvXWW93aMzIyNG3aNElScXGxCgsLXccqKys1d+5cHT9+XP7+/oqMjNTmzZsVHx9/ZZUDAACv5/Gtn0vJzMx02583b57mzZvnUVEAAAASv/UDAAAMRlABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIzlUVBJTk7W4MGD1alTJ3Xt2lXjx4/XoUOHLjkuNzdXUVFR8vPzU1hYmFJTU+tdMAAAaD08Ciq5ubl67LHHtGvXLm3fvl3V1dWKi4tTRUXFBcccPnxY8fHxGjZsmAoKCrRgwQLNmjVLWVlZV1w8AADwbj6edN66davbfkZGhrp27ao9e/Zo+PDhdY5JTU1Vjx49tGjRIklSRESE8vPzlZKSogkTJtSvagAA0Cpc0TMqpaWlkqTOnTtfsE9eXp7i4uLc2kaNGqX8/HxVVVVdyekBAICX82hF5R9ZlqWEhAT98pe/VL9+/S7Yr6SkRMHBwW5twcHBqq6u1qlTp+RwOGqNcTqdcjqdrv2ysrL6lgkAAFqweq+oPP744/rss8+0du3aS/a12Wxu+5Zl1dl+XnJysgIDA11b9+7d61smAABoweoVVJ544glt3LhRO3bsUGho6EX7duvWTSUlJW5tJ0+elI+Pj4KCguock5iYqNLSUtdWVFRUnzIBAEAL59GtH8uy9MQTT2jDhg3KyclR7969LzkmJiZGmzZtcmvbtm2boqOj5evrW+cYu90uu93uSWkAAMALebSi8thjj2nVqlVas2aNOnXqpJKSEpWUlOjHH3909UlMTNT999/v2p85c6aOHj2qhIQEHTx4UOnp6UpLS9PcuXMb7ioAAIBX8iioLF26VKWlpbr11lvlcDhc2/r16119iouLVVhY6Nrv3bu3tmzZopycHN144416/vnntWTJEj6aDAAALsnjWz+XkpmZWattxIgR2rt3ryenAgAA4Ld+AACAuQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxPA4qH374ocaOHauQkBDZbDZlZ2dftH9OTo5sNlut7fPPP69vzQAAoJXw8XRARUWFBgwYoOnTp2vChAmXPe7QoUMKCAhw7Xfp0sXTUwMAgFbG46AyevRojR492uMTde3aVVdddZXH4wAAQOvVZM+oDBw4UA6HQ7GxsdqxY0dTnRYAALRgHq+oeMrhcGjZsmWKioqS0+nUypUrFRsbq5ycHA0fPrzOMU6nU06n07VfVlbW2GUCAAADNXpQCQ8PV3h4uGs/JiZGRUVFSklJuWBQSU5OVlJSUmOXBgAADNcsH08eOnSovvzyywseT0xMVGlpqWsrKipqwuoAAIApGn1FpS4FBQVyOBwXPG6322W325uwIgAAYCKPg8r333+vr776yrV/+PBh7du3T507d1aPHj2UmJio48ePa8WKFZKkRYsWqVevXoqMjFRlZaVWrVqlrKwsZWVlNdxVAAAAr+RxUMnPz9dtt93m2k9ISJAkTZ06VZmZmSouLlZhYaHreGVlpebOnavjx4/L399fkZGR2rx5s+Lj4xugfAAA4M08Diq33nqrLMu64PHMzEy3/Xnz5mnevHkeFwYAAMBv/QAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAY3kcVD788EONHTtWISEhstlsys7OvuSY3NxcRUVFyc/PT2FhYUpNTa1PrQAAoJXxOKhUVFRowIABevXVVy+r/+HDhxUfH69hw4apoKBACxYs0KxZs5SVleVxsQAAoHXx8XTA6NGjNXr06Mvun5qaqh49emjRokWSpIiICOXn5yslJUUTJkzw9PQAAKAVafRnVPLy8hQXF+fWNmrUKOXn56uqqqqxTw8AAFowj1dUPFVSUqLg4GC3tuDgYFVXV+vUqVNyOBy1xjidTjmdTtd+WVlZY5cJAAAM1OhBRZJsNpvbvmVZdbafl5ycrKSkpEavC7iUXvM3N3cJrQLz3DSY56bREuf5yAtjmruEC2r0Wz/dunVTSUmJW9vJkyfl4+OjoKCgOsckJiaqtLTUtRUVFTV2mQAAwECNvqISExOjTZs2ubVt27ZN0dHR8vX1rXOM3W6X3W5v7NIAAIDhPF5R+f7777Vv3z7t27dP0k8fP963b58KCwsl/bQacv/997v6z5w5U0ePHlVCQoIOHjyo9PR0paWlae7cuQ1zBQAAwGt5vKKSn5+v2267zbWfkJAgSZo6daoyMzNVXFzsCi2S1Lt3b23ZskVz5szRa6+9ppCQEC1ZsoSPJgMAgEuyWeefbDVYWVmZAgMDVVpaqoCAgAZ97Zb40BMAAA2psR6mbYj3b37rBwAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYq15B5fXXX1fv3r3l5+enqKgo7dy584J9c3JyZLPZam2ff/55vYsGAACtg8dBZf369XryySe1cOFCFRQUaNiwYRo9erQKCwsvOu7QoUMqLi52bdddd129iwYAAK2Dx0Hl5Zdf1owZM/Tggw8qIiJCixYtUvfu3bV06dKLjuvatau6devm2tq2bVvvogEAQOvgUVCprKzUnj17FBcX59YeFxenTz755KJjBw4cKIfDodjYWO3YscPzSgEAQKvj40nnU6dOqaamRsHBwW7twcHBKikpqXOMw+HQsmXLFBUVJafTqZUrVyo2NlY5OTkaPnx4nWOcTqecTqdrv6yszJMyAQCAl/AoqJxns9nc9i3LqtV2Xnh4uMLDw137MTExKioqUkpKygWDSnJyspKSkupTGgAA8CIe3fq55ppr1LZt21qrJydPnqy1ynIxQ4cO1ZdffnnB44mJiSotLXVtRUVFnpQJAAC8hEdBpV27doqKitL27dvd2rdv366bb775sl+noKBADofjgsftdrsCAgLcNgAA0Pp4fOsnISFBU6ZMUXR0tGJiYrRs2TIVFhZq5syZkn5aDTl+/LhWrFghSVq0aJF69eqlyMhIVVZWatWqVcrKylJWVlbDXgkAAPA6HgeViRMn6vTp03ruuedUXFysfv36acuWLerZs6ckqbi42O07VSorKzV37lwdP35c/v7+ioyM1ObNmxUfH99wVwEAALySzbIsq7mLuJSysjIFBgaqtLS0wW8D9Zq/uUFfDwCAlubIC2Ma5XUb4v2b3/oBAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGMRVAAAgLEIKgAAwFgEFQAAYCyCCgAAMBZBBQAAGIugAgAAjEVQAQAAxiKoAAAAYxFUAACAsQgqAADAWAQVAABgLIIKAAAwFkEFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMaqV1B5/fXX1bt3b/n5+SkqKko7d+68aP/c3FxFRUXJz89PYWFhSk1NrVexAACgdfE4qKxfv15PPvmkFi5cqIKCAg0bNkyjR49WYWFhnf0PHz6s+Ph4DRs2TAUFBVqwYIFmzZqlrKysKy4eAAB4N5tlWZYnA4YMGaJBgwZp6dKlrraIiAiNHz9eycnJtfo/9dRT2rhxow4ePOhqmzlzpv7yl78oLy/vss5ZVlamwMBAlZaWKiAgwJNyL6nX/M0N+noAALQ0R14Y0yiv2xDv3x6tqFRWVmrPnj2Ki4tza4+Li9Mnn3xS55i8vLxa/UeNGqX8/HxVVVV5WC4AAGhNfDzpfOrUKdXU1Cg4ONitPTg4WCUlJXWOKSkpqbN/dXW1Tp06JYfDUWuM0+mU0+l07ZeWlkr6KZk1tHPOHxr8NQEAaEka4/31H1/Xw5s3bjwKKufZbDa3fcuyarVdqn9d7eclJycrKSmpVnv37t09LRUAAFxC4KLGff3y8nIFBgbWa6xHQeWaa65R27Zta62enDx5staqyXndunWrs7+Pj4+CgoLqHJOYmKiEhATX/rlz5/Ttt98qKCjoooHovLKyMnXv3l1FRUUN/kwLLo35b17Mf/Ni/psPc9+86pp/y7JUXl6ukJCQer+uR0GlXbt2ioqK0vbt23XPPfe42rdv365x48bVOSYmJkabNm1ya9u2bZuio6Pl6+tb5xi73S673e7WdtVVV3lSqiQpICCAP9ZmxPw3L+a/eTH/zYe5b14/n//6rqSc5/HHkxMSEvTGG28oPT1dBw8e1Jw5c1RYWKiZM2dK+mk15P7773f1nzlzpo4ePaqEhAQdPHhQ6enpSktL09y5c6+ocAAA4P08fkZl4sSJOn36tJ577jkVFxerX79+2rJli3r27ClJKi4udvtOld69e2vLli2aM2eOXnvtNYWEhGjJkiWaMGFCw10FAADwSvV6mPbRRx/Vo48+WuexzMzMWm0jRozQ3r1763OqerHb7XrmmWdq3T5C02D+mxfz37yY/+bD3Devxpp/j7/wDQAAoKnwo4QAAMBYBBUAAGAsggoAADAWQQUAABirxQaV119/Xb1795afn5+ioqK0c+fOi/bPzc1VVFSU/Pz8FBYWptTU1Caq1Dt5Mv/vvPOO7rjjDnXp0kUBAQGKiYnRBx980ITVeh9P//7P+/jjj+Xj46Mbb7yxcQv0Yp7OvdPp1MKFC9WzZ0/Z7Xb94he/UHp6ehNV6308nf/Vq1drwIABat++vRwOh6ZPn67Tp083UbXe5cMPP9TYsWMVEhIim82m7OzsS45pkPdeqwVat26d5evray1fvtw6cOCANXv2bKtDhw7W0aNH6+z/9ddfW+3bt7dmz55tHThwwFq+fLnl6+trvf32201cuXfwdP5nz55t/eEPf7A+/fRT64svvrASExMtX19fa+/evU1cuXfwdP7PO3PmjBUWFmbFxcVZAwYMaJpivUx95v7uu++2hgwZYm3fvt06fPiw9ec//9n6+OOPm7Bq7+Hp/O/cudNq06aNtXjxYuvrr7+2du7caUVGRlrjx49v4sq9w5YtW6yFCxdaWVlZliRrw4YNF+3fUO+9LTKo3HTTTdbMmTPd2m644QZr/vz5dfafN2+edcMNN7i1PfLII9bQoUMbrUZv5un816Vv375WUlJSQ5fWKtR3/idOnGj99re/tZ555hmCSj15Ovfvv/++FRgYaJ0+fbopyvN6ns7/Sy+9ZIWFhbm1LVmyxAoNDW20GluLywkqDfXe2+Ju/VRWVmrPnj2Ki4tza4+Li9Mnn3xS55i8vLxa/UeNGqX8/HxVVVU1Wq3eqD7z/3Pnzp1TeXm5Onfu3BglerX6zn9GRob+9re/6ZlnnmnsEr1WfeZ+48aNio6O1osvvqhrr71W119/vebOnasff/yxKUr2KvWZ/5tvvlnHjh3Tli1bZFmWvvnmG7399tsaM2ZMU5Tc6jXUe2+9vpm2OZ06dUo1NTW1fq05ODi41q80n1dSUlJn/+rqap06dUoOh6PR6vU29Zn/n/uv//ovVVRU6L777muMEr1afeb/yy+/1Pz587Vz5075+LS4/+SNUZ+5//rrr/XRRx/Jz89PGzZs0KlTp/Too4/q22+/5TkVD9Vn/m+++WatXr1aEydO1NmzZ1VdXa27775br7zySlOU3Oo11Htvi1tROc9ms7ntW5ZVq+1S/etqx+XxdP7PW7t2rZ599lmtX79eXbt2bazyvN7lzn9NTY0mTZqkpKQkXX/99U1Vnlfz5G//3LlzstlsWr16tW666SbFx8fr5ZdfVmZmJqsq9eTJ/B84cECzZs3S008/rT179mjr1q06fPiw60d00fga4r23xf3z6pprrlHbtm1rJeiTJ0/WSm7ndevWrc7+Pj4+CgoKarRavVF95v+89evXa8aMGXrrrbc0cuTIxizTa3k6/+Xl5crPz1dBQYEef/xxST+9eVqWJR8fH23btk233357k9Te0tXnb9/hcOjaa691+5n7iIgIWZalY8eO6brrrmvUmr1JfeY/OTlZt9xyi/793/9dktS/f3916NBBw4YN0+9+9ztW0xtZQ733trgVlXbt2ikqKkrbt293a9++fbtuvvnmOsfExMTU6r9t2zZFR0fL19e30Wr1RvWZf+mnlZRp06ZpzZo13B++Ap7Of0BAgPbv3699+/a5tpkzZyo8PFz79u3TkCFDmqr0Fq8+f/u33HKLTpw4oe+//97V9sUXX6hNmzYKDQ1t1Hq9TX3m/4cfflCbNu5vc23btpX0//+yR+NpsPdejx69NcT5j6ilpaVZBw4csJ588kmrQ4cO1pEjRyzLsqz58+dbU6ZMcfU//xGpOXPmWAcOHLDS0tL4ePIV8HT+16xZY/n4+FivvfaaVVxc7NrOnDnTXJfQonk6/z/Hp37qz9O5Ly8vt0JDQ61f/epX1l//+lcrNzfXuu6666wHH3ywuS6hRfN0/jMyMiwfHx/r9ddft/72t79ZH330kRUdHW3ddNNNzXUJLVp5eblVUFBgFRQUWJKsl19+2SooKHB9PLyx3ntbZFCxLMt67bXXrJ49e1rt2rWzBg0aZOXm5rqOTZ061RoxYoRb/5ycHGvgwIFWu3btrF69ellLly5t4oq9iyfzP2LECEtSrW3q1KlNX7iX8PTv/x8RVK6Mp3N/8OBBa+TIkZa/v78VGhpqJSQkWD/88EMTV+09PJ3/JUuWWH379rX8/f0th8NhTZ482Tp27FgTV+0dduzYcdH/lzfWe6/Nslj/AgAAZmpxz6gAAIDWg6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQXAFZs2bZpsNptsNpt8fX0VFhamuXPn6uOPP5bNZtNHH31U57hRo0bp7rvvdu0fO3ZM7dq10w033FBnf5vNpuzs7Ma4BACGIqgAaBB33nmniouL9fXXX+t3v/udXn/9da1atUoDBgxQRkZGrf5FRUX6n//5H82YMcPVlpmZqfvuu08//PCDPv7446YsH4ChCCoAGoTdble3bt3UvXt3TZo0SZMnT1Z2drZmzJihN998UxUVFW79MzMz1aVLF9evaVuWpYyMDE2ZMkWTJk1SWlpac1wGAMMQVAA0Cn9/f1VVVWny5MmqqqrSW2+95TpmWZYyMzM1depU+fj4SJJ27NihH374QSNHjtSUKVP05ptvqry8vLnKB2AIggqABvfpp59qzZo1io2NVefOnTV+/Hi32z85OTn6+uuv9cADD7ja0tLS9Otf/1pt27ZVZGSk+vTpo/Xr1zdH+QAMQlAB0CDee+89dezYUX5+foqJidHw4cP1yiuvSJJmzJihDz/8UF999ZUkKT09XbfccovCw8MlSWfOnNE777yj3/zmN67X+81vfqP09PSmvxAARvFp7gIAeIfbbrtNS5cula+vr0JCQuTr6+s6NnLkSPXs2VOZmZmaN2+e3nnnHb366quu42vWrNHZs2c1ZMgQV5tlWTp37pwOHDigvn37Num1ADAHQQVAg+jQoYP69OlT5zGbzabp06frjTfeUGhoqNq0aaP77rvPdTwtLU3/9m//pmnTprmNmzVrltLT05WSktKYpQMwmM2yLKu5iwDQsk2bNk1nzpy56HecFBYWqnfv3goMDNSECRO0fPlySdK+ffs0cOBAHTx4sNb3pyxfvlwLFy7U8ePH5evrK5vNppdfflm33XabW78+ffqoY8eODX5dAJofz6gAaBI9evTQyJEj9d1339V6iLZv3751fsnb+PHj9e2332rTpk2utoSEBA0cONBty8/Pb5JrAND0WFEBAADGYkUFAAAYi6ACAACMRVABAADGIqgAAABjEVQAAICxCCoAAMBYBBUAAGAsggoAADAWQQUAABiLoAIAAIxFUAEAAMYiqAAAAGP9H6fKoBT32qPUAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAh8AAAHFCAYAAABSEJsFAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAApwElEQVR4nO3daXRUVb7+8acgoQghiQZliAkJtAgBlHkGAcFgmKFtsEE6DLZ4UUFplWShDHa3EeSyoKGBKw0JKkFQINKAQu6VWVDCoN7GK40ECGKkmZIQpIRk/1+4qP7HhKGwaocK389a58XZZ5+zf7VXpB73OVXlMMYYAQAAWFKhrAsAAAC3F8IHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCB2BRamqqHA6HewsICFBkZKRGjBihb7/9VpL0/PPPy+Fw6P/+7/+uep2JEyfK4XBo79697rZTp07J6XTK4XAoMzOz1POGDx+uqlWrelz3kSNH5HA4NGPGjFKPz5gxQw6HQ0eOHCk2VkxMjEfjnDhxQlOmTNH+/fs9rhGA/yB8AGUgJSVFO3fuVEZGhn7/+99r2bJl6tSpkwoKCjRq1ChJ0uLFi0s9t6ioSG+99ZaaNm2q5s2bu9vffvtt/fjjj5KkRYsW+f5FXMcrr7yi1atXe3TOiRMnNHXqVMIHUM4RPoAy0LhxY7Vt21Zdu3bV5MmT9dJLLykrK0vp6elq3LixWrdurbfffluXL18uce7GjRt1/Phxd0i5YvHixapevbpatWqlZcuW6YcffrD1ckr1q1/9Ss2aNSvTGjxljCnzeQNuB4QP4BbQtm1bSdLRo0clSaNGjVJOTo4+/PDDEn1TUlLkdDo1dOhQd9unn36q//3f/9WwYcP0+9//Xrm5uVq5cqWd4q+itNsu7733ntq0aaOwsDBVqVJFdevW1ciRIyVJmzdvVqtWrSRJI0aMcN+amjJlivv8NWvWqF27dqpSpYpCQkL08MMPa+fOnSXG/uCDD/TAAw/I6XSqbt26mj17tqZMmSKHw1Gsn8Ph0DPPPKMFCxYoNjZWTqdTS5YskSRNnTpVbdq0UXh4uEJDQ9W8eXMtWrRIP/8tzpiYGPXu3Vtr165Vs2bNFBQUpNjYWK1du1bST7faYmNjFRwcrNatW1/1lhhwOyF8ALeAQ4cOSZLuvvtuSdJvf/tbValSpcStl7Nnz+qDDz7QgAEDdOedd7rbr9xmGTlypB577DFVqVLFJ7deioqKdPny5RJbUVHRdc/duXOnBg8erLp16+rdd9/VunXrNGnSJPfqTvPmzZWSkiJJevnll7Vz507t3LlTTzzxhCQpLS1N/fr1U2hoqJYtW6ZFixbp7Nmz6tKli7Zv3+4e56OPPtLAgQNVrVo1LV++XNOnT9eyZcvcoeLn0tPTNX/+fE2aNEkbNmxQp06dJP30nMvo0aO1YsUKrVq1SgMHDtSzzz6rP/7xjyWu8fnnnyspKUkTJkzQqlWrFBYWpoEDB2ry5Mn629/+ptdee01Lly5Vbm6uevfuzeoKYABYk5KSYiSZXbt2mUuXLpn8/Hyzdu1ac/fdd5uQkBCTk5Pj7puQkGACAwPN999/726bM2eOkWQyMjLcbQUFBSY0NNS0bdu22LkOh8McOnSo2PgJCQkmODjY47qzsrKMpOtuWVlZxcaKjo5278+YMcNIMufOnbvqOLt37zaSTEpKSrH2wsJCExERYe6//35TWFjobs/PzzfVq1c37du3d7e1atXKREVFGZfLVaxftWrVzM//yZNkwsLCzJkzZ675+gsLC82lS5fMq6++aqpVq2aKiorcx6Kjo01QUJA5fvy4u23//v1GkqlVq5YpKChwt6enpxtJZs2aNdccDyjvWPkAykDbtm0VGBiokJAQ9e7dWzVr1tSHH36oGjVquPuMGjVKly5d0ttvv+1uS0lJUXR0tLp16+ZuW7FihfLy8ty3L6SfVkCMMe6VBG8ZN26cdu/eXWIbN27cdc+9cktl0KBBWrFihfvTPTfi66+/1okTJzRs2DBVqPDvf7aqVq2qX//619q1a5cuXLiggoICZWZmqn///qpUqVKxfn369Cn12g899FCxVaQrPv74Y3Xv3l1hYWGqWLGiAgMDNWnSJJ0+fVonT54s1rdp06a655573PuxsbGSpC5duqhKlSol2q/cXgNuV4QPoAy89dZb2r17t/bt26cTJ07oiy++UIcOHYr16dSpk+677z53gPjiiy+0d+9e9/MQVyxatEiVK1fWI488onPnzuncuXN64IEHFBMTo9TUVBUWFnqt7sjISLVs2bLEFhkZed1zH3zwQaWnp+vy5cv63e9+p8jISDVu3FjLli277rmnT5+WJNWqVavEsYiICBUVFens2bM6e/asjDHFQtwVpbVd7ZqfffaZ4uLiJEkLFy7Ujh07tHv3bk2cOFGSStw2CQ8PL7Z/Jfhcrf3ixYul1gLcLgLKugDgdhQbG6uWLVtet9/IkSOVmJiozz77TGlpaapQoYKGDx/uPn7w4EH38w61a9cu9RobNmxQz549vVL3L9WvXz/169dPLpdLu3btUnJysoYMGaKYmBi1a9fuqudVq1ZNkvTdd9+VOHbixAlVqFBBd955p4wxcjgc+v7770v0y8nJKfXaP38IVZLeffddBQYGau3atapcubK7PT09/XovEcANYOUDuIUlJCQoICBA//Vf/6WlS5eqW7duio6Odh+/8lDpwoULtWnTpmLb+vXrFRgYeNXvCylLTqdTnTt31rRp0yRJ+/btc7dLJVcW6tevr3vuuUdpaWnFPm1SUFCglStXuj8BExwcrJYtWyo9Pd39nSeSdP78efenT27ElS+Aq1ixorvthx9+KHYLDMDNY+UDuIXVrFlTPXv2VEpKiowxxb7b4/Lly3rrrbcUGxvr/kTIz/Xp00dr1qzRv/71L/cnaQoLC/X++++X6BscHKz4+HjfvBBJkyZN0vHjx9WtWzdFRkbq3Llzmj17tgIDA9W5c2dJP303SFBQkJYuXarY2FhVrVpVERERioiI0PTp0zV06FD17t1bo0ePlsvl0htvvKFz587p9ddfd4/z6quvqlevXurRo4fGjRunwsJCvfHGG6patarOnDlzQ7X26tVLM2fO1JAhQ/Tkk0/q9OnTmjFjhjscAfhlWPkAbnGjRo2SMUbh4eHq37+/u33dunXKycnR6NGjr3ruk08+WeKh1YsXL+o3v/lNie0//uM/fPky1KZNG+Xk5GjChAmKi4vTk08+qaCgIH388cdq1KiRJLk/Xnz69GnFxcWpVatWevPNNyVJQ4YMUXp6uk6fPq3BgwdrxIgRCg0N1aZNm9SxY0f3OI888ohWrlzp7jd+/HgNGDBA/fr10x133HFDtT700ENavHixvvzyS/Xp00cTJ07Uo48+qsTERK/PC3A7chjzs2/MAYBy5tKlS+5PpGzcuLGsywFue9x2AVDujBo1Sg8//LBq1aqlnJwcLViwQF999ZVmz55d1qUBEOEDQDmUn5+vF154Qf/6178UGBio5s2ba/369erevXtZlwZA3HYBAACW8cApAACwivABAACsInwAAACrbrkHTouKinTixAmFhISU+rXHAADg1mOMUX5+viIiIor9AGRpbrnwceLECUVFRZV1GQAA4CZkZ2df98cmb7nwERISIumn4kNDQ8u4GgAAcCPy8vIUFRXlfh+/llsufFy51RIaGkr4AADAz9zIIxM8cAoAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwKqAsi7AtpjEdWVdgseOvN6rrEsAAMBrWPkAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABgFeEDAABYRfgAAABWET4AAIBVhA8AAGCVx+Fj69at6tOnjyIiIuRwOJSenl6iz1dffaW+ffsqLCxMISEhatu2rY4dO+aNegEAgJ/zOHwUFBSoSZMmmjt3bqnHv/nmG3Xs2FENGjTQ5s2b9fnnn+uVV15R5cqVf3GxAADA/wV4ekJ8fLzi4+OvenzixInq2bOnpk+f7m6rW7fuzVUHAADKHa8+81FUVKR169bpvvvuU48ePVS9enW1adOm1FszV7hcLuXl5RXbAABA+eXV8HHy5EmdP39er7/+uh555BFt3LhRAwYM0MCBA7Vly5ZSz0lOTlZYWJh7i4qK8mZJAADgFuP1lQ9J6tevn55//nk1bdpUiYmJ6t27txYsWFDqOUlJScrNzXVv2dnZ3iwJAADcYjx+5uNa7rrrLgUEBKhhw4bF2mNjY7V9+/ZSz3E6nXI6nd4sAwAA3MK8uvJRqVIltWrVSl9//XWx9oMHDyo6OtqbQwEAAD/l8crH+fPndejQIfd+VlaW9u/fr/DwcNWuXVsvvviiBg8erAcffFBdu3bVRx99pL///e/avHmzN+sGAAB+yuPwkZmZqa5du7r3x48fL0lKSEhQamqqBgwYoAULFig5OVljx45V/fr1tXLlSnXs2NF7VQMAAL/lcfjo0qWLjDHX7DNy5EiNHDnyposCAADlF7/tAgAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwLKugCUTzGJ68q6BI8deb1XWZcAALcFVj4AAIBVhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABgFeEDAABYRfgAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFZ5HD62bt2qPn36KCIiQg6HQ+np6VftO3r0aDkcDs2aNesXlAgAAMoTj8NHQUGBmjRporlz516zX3p6uj799FNFRETcdHEAAKD8CfD0hPj4eMXHx1+zz7fffqtnnnlGGzZsUK9evW66OAAAUP54HD6up6ioSMOGDdOLL76oRo0aXbe/y+WSy+Vy7+fl5Xm7JAAAcAvxeviYNm2aAgICNHbs2Bvqn5ycrKlTp3q7jHIlJnFdWZcAAIDXePXTLnv27NHs2bOVmpoqh8NxQ+ckJSUpNzfXvWVnZ3uzJAAAcIvxavjYtm2bTp48qdq1aysgIEABAQE6evSo/vCHPygmJqbUc5xOp0JDQ4ttAACg/PLqbZdhw4ape/fuxdp69OihYcOGacSIEd4cCgAA+CmPw8f58+d16NAh935WVpb279+v8PBw1a5dW9WqVSvWPzAwUDVr1lT9+vV/ebUAAMDveRw+MjMz1bVrV/f++PHjJUkJCQlKTU31WmEAAKB88jh8dOnSRcaYG+5/5MgRT4cAAADlGL/tAgAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwLKugAA8AcxievKugSPHXm9V1mXAJSKlQ8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABgFeEDAABYRfgAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFZ5HD62bt2qPn36KCIiQg6HQ+np6e5jly5d0oQJE3T//fcrODhYERER+t3vfqcTJ054s2YAAODHPA4fBQUFatKkiebOnVvi2IULF7R371698sor2rt3r1atWqWDBw+qb9++XikWAAD4vwBPT4iPj1d8fHypx8LCwpSRkVGsbc6cOWrdurWOHTum2rVr31yVAACg3PA4fHgqNzdXDodDd9xxR6nHXS6XXC6Xez8vL8/XJQEAgDLk0/Bx8eJFJSYmasiQIQoNDS21T3JysqZOnerLMgDgthSTuK6sS/DYkdd7lXUJsMBnn3a5dOmSHnvsMRUVFWnevHlX7ZeUlKTc3Fz3lp2d7auSAADALcAnKx+XLl3SoEGDlJWVpY8//viqqx6S5HQ65XQ6fVEGAAC4BXk9fFwJHv/85z+1adMmVatWzdtDAAAAP+Zx+Dh//rwOHTrk3s/KytL+/fsVHh6uiIgIPfroo9q7d6/Wrl2rwsJC5eTkSJLCw8NVqVIl71UOAAD8ksfhIzMzU127dnXvjx8/XpKUkJCgKVOmaM2aNZKkpk2bFjtv06ZN6tKly81XCgAAygWPw0eXLl1kjLnq8WsdAwAA4LddAACAVYQPAABgFeEDAABYRfgAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABgVUBZFwDg5sUkrivrEjx25PVeZV0CgDLGygcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKs8Dh9bt25Vnz59FBERIYfDofT09GLHjTGaMmWKIiIiFBQUpC5duugf//iHt+oFAAB+zuPwUVBQoCZNmmju3LmlHp8+fbpmzpypuXPnavfu3apZs6Yefvhh5efn/+JiAQCA/wvw9IT4+HjFx8eXeswYo1mzZmnixIkaOHCgJGnJkiWqUaOG0tLSNHr06F9WLQAA8HtefeYjKytLOTk5iouLc7c5nU517txZn3zyiTeHAgAAfsrjlY9rycnJkSTVqFGjWHuNGjV09OjRUs9xuVxyuVzu/by8PG+WBAAAbjFeDR9XOByOYvvGmBJtVyQnJ2vq1Km+KAPwSEziurIu4bbAPKO88ce/6SOv9yrT8b1626VmzZqS/r0CcsXJkydLrIZckZSUpNzcXPeWnZ3tzZIAAMAtxqvho06dOqpZs6YyMjLcbT/++KO2bNmi9u3bl3qO0+lUaGhosQ0AAJRfHt92OX/+vA4dOuTez8rK0v79+xUeHq7atWvrueee02uvvaZ69eqpXr16eu2111SlShUNGTLEq4UDAAD/5HH4yMzMVNeuXd3748ePlyQlJCQoNTVVL730kn744QeNGTNGZ8+eVZs2bbRx40aFhIR4r2oAAOC3PA4fXbp0kTHmqscdDoemTJmiKVOm/JK6AABAOcVvuwAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwyuvh4/Lly3r55ZdVp04dBQUFqW7dunr11VdVVFTk7aEAAIAfCvD2BadNm6YFCxZoyZIlatSokTIzMzVixAiFhYVp3Lhx3h4OAAD4Ga+Hj507d6pfv37q1auXJCkmJkbLli1TZmamt4cCAAB+yOu3XTp27Kj/+Z//0cGDByVJn3/+ubZv366ePXuW2t/lcikvL6/YBgAAyi+vr3xMmDBBubm5atCggSpWrKjCwkL9+c9/1m9/+9tS+ycnJ2vq1KneLgMAANyivL7ysXz5cr3zzjtKS0vT3r17tWTJEs2YMUNLliwptX9SUpJyc3PdW3Z2trdLAgAAtxCvr3y8+OKLSkxM1GOPPSZJuv/++3X06FElJycrISGhRH+n0ymn0+ntMgAAwC3K6ysfFy5cUIUKxS9bsWJFPmoLAAAk+WDlo0+fPvrzn/+s2rVrq1GjRtq3b59mzpypkSNHensoAADgh7wePubMmaNXXnlFY8aM0cmTJxUREaHRo0dr0qRJ3h4KAAD4Ia+Hj5CQEM2aNUuzZs3y9qUBAEA5wG+7AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAqoCyLgAAgCtiEteVdQmwgJUPAABgFeEDAABYRfgAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABgFeEDAABYRfgAAABW+SR8fPvtt3r88cdVrVo1ValSRU2bNtWePXt8MRQAAPAzAd6+4NmzZ9WhQwd17dpVH374oapXr65vvvlGd9xxh7eHAgAAfsjr4WPatGmKiopSSkqKuy0mJsbbwwAAAD/l9dsua9asUcuWLfWb3/xG1atXV7NmzbRw4cKr9ne5XMrLyyu2AQCA8svr4ePw4cOaP3++6tWrpw0bNuipp57S2LFj9dZbb5XaPzk5WWFhYe4tKirK2yUBAIBbiMMYY7x5wUqVKqlly5b65JNP3G1jx47V7t27tXPnzhL9XS6XXC6Xez8vL09RUVHKzc1VaGioN0uTJMUkrvP6NQEA8CdHXu/l9Wvm5eUpLCzsht6/vb7yUatWLTVs2LBYW2xsrI4dO1Zqf6fTqdDQ0GIbAAAov7wePjp06KCvv/66WNvBgwcVHR3t7aEAAIAf8nr4eP7557Vr1y699tprOnTokNLS0vTmm2/q6aef9vZQAADAD3k9fLRq1UqrV6/WsmXL1LhxY/3xj3/UrFmzNHToUG8PBQAA/JDXv+dDknr37q3evXv74tIAAMDP8dsuAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKzyefhITk6Ww+HQc8895+uhAACAH/Bp+Ni9e7fefPNNPfDAA74cBgAA+BGfhY/z589r6NChWrhwoe68805fDQMAAPyMz8LH008/rV69eql79+7X7OdyuZSXl1dsAwAA5VeALy767rvvau/evdq9e/d1+yYnJ2vq1Km+KAMAANyCvL7ykZ2drXHjxumdd95R5cqVr9s/KSlJubm57i07O9vbJQEAgFuI11c+9uzZo5MnT6pFixbutsLCQm3dulVz586Vy+VSxYoV3cecTqecTqe3ywAAALcor4ePbt266csvvyzWNmLECDVo0EATJkwoFjwAAMDtx+vhIyQkRI0bNy7WFhwcrGrVqpVoBwAAtx++4RQAAFjlk0+7/NzmzZttDAMAAPwAKx8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsIrwAQAArCJ8AAAAqwgfAADAKsIHAACwivABAACsInwAAACrCB8AAMAqwgcAALCK8AEAAKwifAAAAKsIHwAAwCrCBwAAsMrr4SM5OVmtWrVSSEiIqlevrv79++vrr7/29jAAAMBPeT18bNmyRU8//bR27dqljIwMXb58WXFxcSooKPD2UAAAwA8FePuCH330UbH9lJQUVa9eXXv27NGDDz7o7eEAAICf8Xr4+Lnc3FxJUnh4eKnHXS6XXC6Xez8vL8/XJQEAgDLk0wdOjTEaP368OnbsqMaNG5faJzk5WWFhYe4tKirKlyUBAIAy5tPw8cwzz+iLL77QsmXLrtonKSlJubm57i07O9uXJQEAgDLms9suzz77rNasWaOtW7cqMjLyqv2cTqecTqevygAAALcYr4cPY4yeffZZrV69Wps3b1adOnW8PQQAAPBjXg8fTz/9tNLS0vTBBx8oJCREOTk5kqSwsDAFBQV5ezgAAOBnvP7Mx/z585Wbm6suXbqoVq1a7m358uXeHgoAAPghn9x2AQAAuBp+2wUAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABgFeEDAABYRfgAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABgFeEDAABYRfgAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVvksfMybN0916tRR5cqV1aJFC23bts1XQwEAAD/ik/CxfPlyPffcc5o4caL27dunTp06KT4+XseOHfPFcAAAwI/4JHzMnDlTo0aN0hNPPKHY2FjNmjVLUVFRmj9/vi+GAwAAfsTr4ePHH3/Unj17FBcXV6w9Li5On3zyibeHAwAAfibA2xc8deqUCgsLVaNGjWLtNWrUUE5OTon+LpdLLpfLvZ+bmytJysvL83ZpkqQi1wWfXBcAAH/hi/fYK9c0xly3r9fDxxUOh6PYvjGmRJskJScna+rUqSXao6KifFUaAAC3tbBZvrt2fn6+wsLCrtnH6+HjrrvuUsWKFUuscpw8ebLEaogkJSUlafz48e79oqIinTlzRtWqVSs1rHgiLy9PUVFRys7OVmho6C+6Fq6NubaHubaL+baHubbHF3NtjFF+fr4iIiKu29fr4aNSpUpq0aKFMjIyNGDAAHd7RkaG+vXrV6K/0+mU0+ks1nbHHXd4tabQ0FD+kC1hru1hru1ivu1hru3x9lxfb8XjCp/cdhk/fryGDRumli1bql27dnrzzTd17NgxPfXUU74YDgAA+BGfhI/Bgwfr9OnTevXVV/Xdd9+pcePGWr9+vaKjo30xHAAA8CM+e+B0zJgxGjNmjK8uf0OcTqcmT55c4rYOvI+5toe5tov5toe5tqes59phbuQzMQAAAF7CD8sBAACrCB8AAMAqwgcAALCK8AEAAKzy+/Axb9481alTR5UrV1aLFi20bdu2a/bfsmWLWrRoocqVK6tu3bpasGCBpUr9nydzvWrVKj388MO6++67FRoaqnbt2mnDhg0Wq/Vvnv5dX7Fjxw4FBASoadOmvi2wHPF0rl0ulyZOnKjo6Gg5nU796le/0uLFiy1V6/88ne+lS5eqSZMmqlKlimrVqqURI0bo9OnTlqr1T1u3blWfPn0UEREhh8Oh9PT0655j/b3R+LF3333XBAYGmoULF5oDBw6YcePGmeDgYHP06NFS+x8+fNhUqVLFjBs3zhw4cMAsXLjQBAYGmvfff99y5f7H07keN26cmTZtmvnss8/MwYMHTVJSkgkMDDR79+61XLn/8XSurzh37pypW7euiYuLM02aNLFTrJ+7mbnu27evadOmjcnIyDBZWVnm008/NTt27LBYtf/ydL63bdtmKlSoYGbPnm0OHz5stm3bZho1amT69+9vuXL/sn79ejNx4kSzcuVKI8msXr36mv3L4r3Rr8NH69atzVNPPVWsrUGDBiYxMbHU/i+99JJp0KBBsbbRo0ebtm3b+qzG8sLTuS5Nw4YNzdSpU71dWrlzs3M9ePBg8/LLL5vJkycTPm6Qp3P94YcfmrCwMHP69Gkb5ZU7ns73G2+8YerWrVus7S9/+YuJjIz0WY3lzY2Ej7J4b/Tb2y4//vij9uzZo7i4uGLtcXFx+uSTT0o9Z+fOnSX69+jRQ5mZmbp06ZLPavV3NzPXP1dUVKT8/HyFh4f7osRy42bnOiUlRd98840mT57s6xLLjZuZ6zVr1qhly5aaPn267rnnHt1333164YUX9MMPP9go2a/dzHy3b99ex48f1/r162WM0ffff6/3339fvXr1slHybaMs3ht99g2nvnbq1CkVFhaW+KXcGjVqlPhF3StycnJK7X/58mWdOnVKtWrV8lm9/uxm5vrn/vM//1MFBQUaNGiQL0osN25mrv/5z38qMTFR27ZtU0CA3/4nbd3NzPXhw4e1fft2Va5cWatXr9apU6c0ZswYnTlzhuc+ruNm5rt9+/ZaunSpBg8erIsXL+ry5cvq27ev5syZY6Pk20ZZvDf67crHFQ6Ho9i+MaZE2/X6l9aOkjyd6yuWLVumKVOmaPny5apevbqvyitXbnSuCwsLNWTIEE2dOlX33XefrfLKFU/+rouKiuRwOLR06VK1bt1aPXv21MyZM5Wamsrqxw3yZL4PHDigsWPHatKkSdqzZ48++ugjZWVl8SOlPmD7vdFv/zfprrvuUsWKFUsk5pMnT5ZIcFfUrFmz1P4BAQGqVq2az2r1dzcz11csX75co0aN0nvvvafu3bv7ssxywdO5zs/PV2Zmpvbt26dnnnlG0k9vkMYYBQQEaOPGjXrooYes1O5vbubvulatWrrnnnuK/Wx4bGysjDE6fvy46tWr59Oa/dnNzHdycrI6dOigF198UZL0wAMPKDg4WJ06ddKf/vQnVqu9pCzeG/125aNSpUpq0aKFMjIyirVnZGSoffv2pZ7Trl27Ev03btyoli1bKjAw0Ge1+rubmWvppxWP4cOHKy0tjXu0N8jTuQ4NDdWXX36p/fv3u7ennnpK9evX1/79+9WmTRtbpfudm/m77tChg06cOKHz58+72w4ePKgKFSooMjLSp/X6u5uZ7wsXLqhCheJvUxUrVpT07/8zxy9XJu+NPnuU1YIrH9tatGiROXDggHnuuedMcHCwOXLkiDHGmMTERDNs2DB3/ysfJ3r++efNgQMHzKJFi/io7Q3ydK7T0tJMQECA+etf/2q+++4793bu3Lmyegl+w9O5/jk+7XLjPJ3r/Px8ExkZaR599FHzj3/8w2zZssXUq1fPPPHEE2X1EvyKp/OdkpJiAgICzLx588w333xjtm/fblq2bGlat25dVi/BL+Tn55t9+/aZffv2GUlm5syZZt++fe6PNN8K741+HT6MMeavf/2riY6ONpUqVTLNmzc3W7ZscR9LSEgwnTt3LtZ/8+bNplmzZqZSpUomJibGzJ8/33LF/suTue7cubORVGJLSEiwX7gf8vTv+v9H+PCMp3P91Vdfme7du5ugoCATGRlpxo8fby5cuGC5av/l6Xz/5S9/MQ0bNjRBQUGmVq1aZujQoeb48eOWq/YvmzZtuua/v7fCe6PDGNauAACAPX77zAcAAPBPhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AHhs+PDhcjgccjgcCgwMVN26dfXCCy9ox44dcjgc2r59e6nn9ejRQ3379nXvHz9+XJUqVVKDBg1K7e9wOJSenu6LlwCgDBE+ANyURx55RN99950OHz6sP/3pT5o3b57eeecdNWnSRCkpKSX6Z2dn67//+781atQod1tqaqoGDRqkCxcuaMeOHTbLB1CGCB8AborT6VTNmjUVFRWlIUOGaOjQoUpPT9eoUaO0YsUKFRQUFOufmpqqu+++2/0Lx8YYpaSkaNiwYRoyZIgWLVpUFi8DQBkgfADwiqCgIF26dElDhw7VpUuX9N5777mPGWOUmpqqhIQEBQQESJI2bdqkCxcuqHv37ho2bJhWrFih/Pz8siofgEWEDwC/2Geffaa0tDR169ZN4eHh6t+/f7FbL5s3b9bhw4c1cuRId9uiRYv02GOPqWLFimrUqJHuvfdeLV++vCzKB2AZ4QPATVm7dq2qVq2qypUrq127dnrwwQc1Z84cSdKoUaO0detWHTp0SJK0ePFidejQQfXr15cknTt3TqtWrdLjjz/uvt7jjz+uxYsX238hAKwLKOsCAPinrl27av78+QoMDFRERIQCAwPdx7p3767o6GilpqbqpZde0qpVqzR37lz38bS0NF28eFFt2rRxtxljVFRUpAMHDqhhw4ZWXwsAuwgfAG5KcHCw7r333lKPORwOjRgxQn/7298UGRmpChUqaNCgQe7jixYt0h/+8AcNHz682Hljx47V4sWLNWPGDF+WDqCMOYwxpqyLAOBfhg8frnPnzl3zOziOHTumOnXqKCwsTL/+9a+1cOFCSdL+/fvVrFkzffXVVyW+32PhwoWaOHGivv32WwUGBsrhcGjmzJnq2rVrsX733nuvqlat6vXXBcAOnvkA4BO1a9dW9+7ddfbs2RIPmjZs2LDULxbr37+/zpw5o7///e/utvHjx6tZs2bFtszMTCuvAYBvsPIBAACsYuUDAABYRfgAAABWET4AAIBVhA8AAGAV4QMAAFhF+AAAAFYRPgAAgFWEDwAAYBXhAwAAWEX4AAAAVhE+AACAVYQPAABg1f8DScp0/wuTBr0AAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n",
      "findfont: Generic family 'sans-serif' not found because none of the following families were found: Arial\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAiwAAAHFCAYAAADR1KI+AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAvUUlEQVR4nO3deXBUZb7G8achocOWOEEgCQkBFJBFkT3AsAkEwyIICApCEFy4OormcpEMLuC1jAvjIItwxSyirBoIDKASSnajEkkcR1BhCIQlkQHJAkgI5Nw/LHqmzQKN3cmb8P1UnSrPe9739O+8laIf33O622ZZliUAAACDVavoAgAAAK6GwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAhguISFBNpvNsXl5eSk4OFgPPfSQjh8/Lkl65plnZLPZ9P3335d6npkzZ8pms2nv3r2OtlOnTslut8tmsyk1NbXEcRMnTlSdOnVcrvvw4cOy2WyaM2dOicfnzJkjm82mw4cPO71WkyZNXHqdEydOaNasWUpPT3e5RgCVB4EFqCTi4+OVkpKi5ORkPfLII1qxYoV69uypc+fOafLkyZKkuLi4EscWFRVp6dKluvPOO9WhQwdH+/vvv6+LFy9KkmJjYz1/EVfx/PPPa+3atS6NOXHihGbPnk1gAao4AgtQSbRt21ZhYWHq27evXnzxRU2fPl0ZGRlKSkpS27Zt1aVLF73//vu6dOlSsbGbN2/WsWPHHMHmiri4ODVo0ECdO3fWihUr9Msvv5TX5ZTolltuUfv27Su0BldZllXh8wbcCAgsQCUVFhYmSTpy5IgkafLkycrOztbHH39crG98fLzsdrvGjRvnaPvyyy/1j3/8Q+PHj9cjjzyi3NxcJSYmlk/xpSjpltCHH36orl27ys/PT7Vq1VKzZs00adIkSdK2bdvUuXNnSdJDDz3kuG02a9Ysx/j169erW7duqlWrlurWrasBAwYoJSWl2GuvW7dOd9xxh+x2u5o1a6a33npLs2bNks1mc+pns9n0pz/9SYsXL1arVq1kt9v13nvvSZJmz56trl27yt/fX76+vurQoYNiY2P129+YbdKkiYYMGaINGzaoffv2qlmzplq1aqUNGzZI+vU2YKtWrVS7dm116dKl1Nt1wI2EwAJUUgcPHpQk1a9fX5L0wAMPqFatWsVuC505c0br1q3Tvffeqz/84Q+O9iu3gCZNmqT7779ftWrV8shtoaKiIl26dKnYVlRUdNWxKSkpGjNmjJo1a6aVK1dq48aNeuGFFxyrSB06dFB8fLwk6bnnnlNKSopSUlL08MMPS5KWL1+uYcOGydfXVytWrFBsbKzOnDmjPn36aNeuXY7X+eSTTzRixAjVq1dPq1at0uuvv64VK1Y4gshvJSUladGiRXrhhRf06aefqmfPnpJ+fW7nscce0+rVq7VmzRqNGDFCTz75pP73f/+32Dm++eYbRUdH69lnn9WaNWvk5+enESNG6MUXX9S7776rV155RcuWLVNubq6GDBnCKg5gATBafHy8Jcn64osvrMLCQis/P9/asGGDVb9+fatu3bpWdna2o29kZKTl7e1t/fTTT462+fPnW5Ks5ORkR9u5c+csX19fKywszGmszWazDh486PT6kZGRVu3atV2uOyMjw5J01S0jI8PptUJDQx37c+bMsSRZOTk5pb7Onj17LElWfHy8U/vly5etoKAg6/bbb7cuX77saM/Pz7caNGhgde/e3dHWuXNnKyQkxCooKHDqV69ePeu3/0xKsvz8/Kyff/65zOu/fPmyVVhYaL300ktWvXr1rKKiIsex0NBQq2bNmtaxY8ccbenp6ZYkKzAw0Dp37pyjPSkpyZJkrV+/vszXA6o6VliASiIsLEze3t6qW7euhgwZooCAAH388cdq2LCho8/kyZNVWFio999/39EWHx+v0NBQ9evXz9G2evVq5eXlOW6tSL+utFiW5VixcJepU6dqz549xbapU6dedeyV2z2jR4/W6tWrHZ+KuhY//PCDTpw4ofHjx6tatX//U1enTh2NHDlSX3zxhc6fP69z584pNTVVw4cPV40aNZz6DR06tMRz33XXXU6rVVd89tln6t+/v/z8/FS9enV5e3vrhRde0OnTp3Xy5EmnvnfeeacaNWrk2G/VqpUkqU+fPqpVq1ax9iu3/oAbFYEFqCSWLl2qPXv2KC0tTSdOnNDf//539ejRw6lPz5491aJFC0fo+Pvf/669e/c6nu+4IjY2Vj4+Prr77ruVk5OjnJwc3XHHHWrSpIkSEhJ0+fJlt9UdHBysTp06FduCg4OvOrZXr15KSkrSpUuXNGHCBAUHB6tt27ZasWLFVceePn1akhQYGFjsWFBQkIqKinTmzBmdOXNGlmU5Bb8rSmor7ZxfffWVwsPDJUlLlizR7t27tWfPHs2cOVOSit3S8ff3d9q/EpZKa79w4UKJtQA3Cq+KLgDAtWnVqpU6dep01X6TJk3SjBkz9NVXX2n58uWqVq2aJk6c6Dj+448/Op7faNy4cYnn+PTTTzVo0CC31P17DRs2TMOGDVNBQYG++OILxcTEaOzYsWrSpIm6detW6rh69epJkrKysoodO3HihKpVq6Y//OEPsixLNptNP/30U7F+2dnZJZ77tw/iStLKlSvl7e2tDRs2yMfHx9GelJR0tUsEcA1YYQGqmMjISHl5een//u//tGzZMvXr10+hoaGO41cerF2yZIm2bt3qtG3atEne3t6lfp9LRbLb7erdu7dee+01SVJaWpqjXSq+gtGyZUs1atRIy5cvd/qUzrlz55SYmOj45FDt2rXVqVMnJSUlOb6TRpLOnj3r+NTOtbjypX7Vq1d3tP3yyy9Ot+cAXD9WWIAqJiAgQIMGDVJ8fLwsy3L67pVLly5p6dKlatWqleOTNL81dOhQrV+/Xv/6178cn0C6fPmyPvroo2J9a9eurYiICM9ciKQXXnhBx44dU79+/RQcHKycnBy99dZb8vb2Vu/evSX9+t0tNWvW1LJly9SqVSvVqVNHQUFBCgoK0uuvv65x48ZpyJAheuyxx1RQUKA33nhDOTk5evXVVx2v89JLL2nw4MEaOHCgpk6dqsuXL+uNN95QnTp19PPPP19TrYMHD9abb76psWPH6tFHH9Xp06c1Z84cR6AC8PuwwgJUQZMnT5ZlWfL399fw4cMd7Rs3blR2drYee+yxUsc++uijxR7cvXDhgu67775i23/913958jLUtWtXZWdn69lnn1V4eLgeffRR1axZU5999pnatGkjSY6Pcp8+fVrh4eHq3Lmz3nnnHUnS2LFjlZSUpNOnT2vMmDF66KGH5Ovrq61bt+qPf/yj43XuvvtuJSYmOvpFRUXp3nvv1bBhw3TTTTddU6133XWX4uLi9O2332ro0KGaOXOmRo0apRkzZrh9XoAbkc2yfvONRgAAFRYWOj7Js3nz5oouB7jhcUsIAPTrqtSAAQMUGBio7OxsLV68WPv379dbb71V0aUBEIEFACRJ+fn5mjZtmv71r3/J29tbHTp00KZNm9S/f/+KLg2AuCUEAAAqAR66BQAAxiOwAAAA4xFYAACA8arMQ7dFRUU6ceKE6tatW+LXZgMAAPNYlqX8/HwFBQU5/VDpb1WZwHLixAmFhIRUdBkAAOA6HD16tMwfRa0ygaVu3bqSfr1gX1/fCq4GAABci7y8PIWEhDjex0tTZQLLldtAvr6+BBYAACqZqz3OwUO3AADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4BBYAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAON5VXQBlUGTGRsrugSXHX51cEWXAACA27DCAgAAjEdgAQAAxiOwAAAA4xFYAACA8VwOLDt27NDQoUMVFBQkm82mpKQkp+M2m63E7Y033ij1nAkJCSWOuXDhgssXBAAAqh6XA8u5c+fUrl07LViwoMTjWVlZTltcXJxsNptGjhxZ5nl9fX2LjfXx8XG1PAAAUAW5/LHmiIgIRURElHo8ICDAaX/dunXq27evmjVrVuZ5bTZbsbEAAACSh59h+emnn7Rx40ZNnjz5qn3Pnj2r0NBQBQcHa8iQIUpLS/NkaQAAoBLxaGB57733VLduXY0YMaLMfrfddpsSEhK0fv16rVixQj4+PurRo4cOHDhQ6piCggLl5eU5bQAAoGryaGCJi4vTuHHjrvosSlhYmB588EG1a9dOPXv21OrVq9WiRQvNnz+/1DExMTHy8/NzbCEhIe4uHwAAGMJjgWXnzp364Ycf9PDDD7s8tlq1aurcuXOZKyzR0dHKzc11bEePHv095QIAAIN57LeEYmNj1bFjR7Vr187lsZZlKT09Xbfffnupfex2u+x2++8pEQAAVBIuB5azZ8/q4MGDjv2MjAylp6fL399fjRs3liTl5eXpww8/1F/+8pcSzzFhwgQ1atRIMTExkqTZs2crLCxMzZs3V15enubNm6f09HQtXLjweq4JAABUMS4HltTUVPXt29exHxUVJUmKjIxUQkKCJGnlypWyLEsPPPBAiefIzMxUtWr/vhuVk5OjRx99VNnZ2fLz81P79u21Y8cOdenSxdXyAABAFWSzLMuq6CLcIS8vT35+fsrNzZWvr69bz91kxka3nq88HH51cEWXAADAVV3r+ze/JQQAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4BBYAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPJcDy44dOzR06FAFBQXJZrMpKSnJ6fjEiRNls9mctrCwsKueNzExUa1bt5bdblfr1q21du1aV0sDAABVlMuB5dy5c2rXrp0WLFhQap+7775bWVlZjm3Tpk1lnjMlJUVjxozR+PHj9c0332j8+PEaPXq0vvzyS1fLAwAAVZCXqwMiIiIUERFRZh+73a6AgIBrPufcuXM1YMAARUdHS5Kio6O1fft2zZ07VytWrHC1RAAAUMV45BmWbdu2qUGDBmrRooUeeeQRnTx5ssz+KSkpCg8Pd2obOHCgPv/881LHFBQUKC8vz2kDAABVk8srLFcTERGh++67T6GhocrIyNDzzz+vu+66S19//bXsdnuJY7Kzs9WwYUOntoYNGyo7O7vU14mJidHs2bPdWjsqVpMZGyu6hOty+NXBFV0CAFR5bg8sY8aMcfx327Zt1alTJ4WGhmrjxo0aMWJEqeNsNpvTvmVZxdr+U3R0tKKiohz7eXl5CgkJ+R2VAwAAU7k9sPxWYGCgQkNDdeDAgVL7BAQEFFtNOXnyZLFVl/9kt9tLXbEBAABVi8e/h+X06dM6evSoAgMDS+3TrVs3JScnO7Vt3rxZ3bt393R5AACgEnB5heXs2bM6ePCgYz8jI0Pp6eny9/eXv7+/Zs2apZEjRyowMFCHDx/Wn//8Z91888269957HWMmTJigRo0aKSYmRpI0depU9erVS6+99pqGDRumdevWacuWLdq1a5cbLhEAAFR2LgeW1NRU9e3b17F/5TmSyMhILVq0SN9++62WLl2qnJwcBQYGqm/fvlq1apXq1q3rGJOZmalq1f69uNO9e3etXLlSzz33nJ5//nndcsstWrVqlbp27fp7rg0AAFQRLgeWPn36yLKsUo9/+umnVz3Htm3birWNGjVKo0aNcrUcAABwA+C3hAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYz+VfawbgrMmMjRVdgssOvzq4oksAbmj8u+E6VlgAAIDxCCwAAMB4BBYAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjuRxYduzYoaFDhyooKEg2m01JSUmOY4WFhXr22Wd1++23q3bt2goKCtKECRN04sSJMs+ZkJAgm81WbLtw4YLLFwQAAKoelwPLuXPn1K5dOy1YsKDYsfPnz2vv3r16/vnntXfvXq1Zs0Y//vij7rnnnque19fXV1lZWU6bj4+Pq+UBAIAqyMvVAREREYqIiCjxmJ+fn5KTk53a5s+fry5duigzM1ONGzcu9bw2m00BAQGulgMAAG4AHn+GJTc3VzabTTfddFOZ/c6ePavQ0FAFBwdryJAhSktLK7N/QUGB8vLynDYAAFA1eTSwXLhwQTNmzNDYsWPl6+tbar/bbrtNCQkJWr9+vVasWCEfHx/16NFDBw4cKHVMTEyM/Pz8HFtISIgnLgEAABjAY4GlsLBQ999/v4qKivT222+X2TcsLEwPPvig2rVrp549e2r16tVq0aKF5s+fX+qY6Oho5ebmOrajR4+6+xIAAIAhXH6G5VoUFhZq9OjRysjI0GeffVbm6kpJqlWrps6dO5e5wmK322W3239vqQAAoBJw+wrLlbBy4MABbdmyRfXq1XP5HJZlKT09XYGBge4uDwAAVEIur7CcPXtWBw8edOxnZGQoPT1d/v7+CgoK0qhRo7R3715t2LBBly9fVnZ2tiTJ399fNWrUkCRNmDBBjRo1UkxMjCRp9uzZCgsLU/PmzZWXl6d58+YpPT1dCxcudMc1AgCASs7lwJKamqq+ffs69qOioiRJkZGRmjVrltavXy9JuvPOO53Gbd26VX369JEkZWZmqlq1fy/u5OTk6NFHH1V2drb8/PzUvn177dixQ126dHG1PAAAUAW5HFj69Okjy7JKPV7WsSu2bdvmtP/Xv/5Vf/3rX10tBQAA3CD4LSEAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4BBYAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA47kcWHbs2KGhQ4cqKChINptNSUlJTscty9KsWbMUFBSkmjVrqk+fPvruu++uet7ExES1bt1adrtdrVu31tq1a10tDQAAVFEuB5Zz586pXbt2WrBgQYnHX3/9db355ptasGCB9uzZo4CAAA0YMED5+fmlnjMlJUVjxozR+PHj9c0332j8+PEaPXq0vvzyS1fLAwAAVZCXqwMiIiIUERFR4jHLsjR37lzNnDlTI0aMkCS99957atiwoZYvX67HHnusxHFz587VgAEDFB0dLUmKjo7W9u3bNXfuXK1YscLVEgEAQBXj1mdYMjIylJ2drfDwcEeb3W5X79699fnnn5c6LiUlxWmMJA0cOLDMMQUFBcrLy3PaAABA1eTyCktZsrOzJUkNGzZ0am/YsKGOHDlS5riSxlw5X0liYmI0e/bs31EtgMqkyYyNFV3CDeHwq4MrugSgRB75lJDNZnPatyyrWNvvHRMdHa3c3FzHdvTo0esvGAAAGM2tKywBAQGSfl0xCQwMdLSfPHmy2ArKb8f9djXlamPsdrvsdvvvrBgAAFQGbl1hadq0qQICApScnOxou3jxorZv367u3buXOq5bt25OYyRp8+bNZY4BAAA3DpdXWM6ePauDBw869jMyMpSeni5/f381btxYTz/9tF555RU1b95czZs31yuvvKJatWpp7NixjjETJkxQo0aNFBMTI0maOnWqevXqpddee03Dhg3TunXrtGXLFu3atcsNlwgAACo7lwNLamqq+vbt69iPioqSJEVGRiohIUHTp0/XL7/8oscff1xnzpxR165dtXnzZtWtW9cxJjMzU9Wq/Xtxp3v37lq5cqWee+45Pf/887rlllu0atUqde3a9fdcGwAAqCJcDix9+vSRZVmlHrfZbJo1a5ZmzZpVap9t27YVaxs1apRGjRrlajkAAOAGwG8JAQAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4BBYAAGA8AgsAADCey7/WDACouprM2FjRJbjs8KuDK7oElANWWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4XhVdAAAAv0eTGRsrugSUA1ZYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYz+2BpUmTJrLZbMW2J554osT+27ZtK7H/999/7+7SAABAJeX2jzXv2bNHly9fduz/4x//0IABA3TfffeVOe6HH36Qr6+vY79+/fruLg0AAFRSbg8svw0ar776qm655Rb17t27zHENGjTQTTfd5O5yAABAFeDRZ1guXryoDz74QJMmTZLNZiuzb/v27RUYGKh+/fpp69atVz13QUGB8vLynDYAAFA1eTSwJCUlKScnRxMnTiy1T2BgoN555x0lJiZqzZo1atmypfr166cdO3aUee6YmBj5+fk5tpCQEDdXDwAATOHRr+aPjY1VRESEgoKCSu3TsmVLtWzZ0rHfrVs3HT16VHPmzFGvXr1KHRcdHa2oqCjHfl5eHqEFAIAqymOB5ciRI9qyZYvWrFnj8tiwsDB98MEHZfax2+2y2+3XWx4AAKhEPHZLKD4+Xg0aNNDgwYNdHpuWlqbAwEAPVAUAACojj6ywFBUVKT4+XpGRkfLycn6J6OhoHT9+XEuXLpUkzZ07V02aNFGbNm0cD+kmJiYqMTHRE6UBAIBKyCOBZcuWLcrMzNSkSZOKHcvKylJmZqZj/+LFi5o2bZqOHz+umjVrqk2bNtq4caMGDRrkidIAAEAl5JHAEh4eLsuySjyWkJDgtD99+nRNnz7dE2UAAIAqgt8SAgAAxiOwAAAA4xFYAACA8Tz6xXEAzNRkxsaKLgEAXMIKCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4BBYAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYz+2BZdasWbLZbE5bQEBAmWO2b9+ujh07ysfHR82aNdPixYvdXRYAAKjEvDxx0jZt2mjLli2O/erVq5faNyMjQ4MGDdIjjzyiDz74QLt379bjjz+u+vXra+TIkZ4oDwAAVDIeCSxeXl5XXVW5YvHixWrcuLHmzp0rSWrVqpVSU1M1Z84cAgsAAJDkoWdYDhw4oKCgIDVt2lT333+/Dh06VGrflJQUhYeHO7UNHDhQqampKiwsLHVcQUGB8vLynDYAAFA1uX2FpWvXrlq6dKlatGihn376SS+//LK6d++u7777TvXq1SvWPzs7Ww0bNnRqa9iwoS5duqRTp04pMDCwxNeJiYnR7Nmz3V1+ldFkxsaKLgEAALdx+wpLRESERo4cqdtvv139+/fXxo2/vnG+9957pY6x2WxO+5Zlldj+n6Kjo5Wbm+vYjh496obqAQCAiTzyDMt/ql27tm6//XYdOHCgxOMBAQHKzs52ajt58qS8vLxKXJG5wm63y263u7VWAABgJo9/D0tBQYH2799f6q2dbt26KTk52alt8+bN6tSpk7y9vT1dHgAAqATcHlimTZum7du3KyMjQ19++aVGjRqlvLw8RUZGSvr1Vs6ECRMc/adMmaIjR44oKipK+/fvV1xcnGJjYzVt2jR3lwYAACopt98SOnbsmB544AGdOnVK9evXV1hYmL744guFhoZKkrKyspSZmeno37RpU23atEnPPPOMFi5cqKCgIM2bN4+PNAMAAAebdeUJ10ouLy9Pfn5+ys3Nla+vr1vPzSduAAA3usOvDvbIea/1/ZvfEgIAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4BBYAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwntsDS0xMjDp37qy6deuqQYMGGj58uH744Ycyx2zbtk02m63Y9v3337u7PAAAUAm5PbBs375dTzzxhL744gslJyfr0qVLCg8P17lz56469ocfflBWVpZja968ubvLAwAAlZCXu0/4ySefOO3Hx8erQYMG+vrrr9WrV68yxzZo0EA33XSTu0sCAACVnMefYcnNzZUk+fv7X7Vv+/btFRgYqH79+mnr1q1l9i0oKFBeXp7TBgAAqiaPBhbLshQVFaU//vGPatu2ban9AgMD9c477ygxMVFr1qxRy5Yt1a9fP+3YsaPUMTExMfLz83NsISEhnrgEAABgAJtlWZanTv7EE09o48aN2rVrl4KDg10aO3ToUNlsNq1fv77E4wUFBSooKHDs5+XlKSQkRLm5ufL19f1ddf9Wkxkb3Xo+AAAqm8OvDvbIefPy8uTn53fV92+PrbA8+eSTWr9+vbZu3epyWJGksLAwHThwoNTjdrtdvr6+ThsAAKia3P7QrWVZevLJJ7V27Vpt27ZNTZs2va7zpKWlKTAw0M3VAQCAysjtgeWJJ57Q8uXLtW7dOtWtW1fZ2dmSJD8/P9WsWVOSFB0drePHj2vp0qWSpLlz56pJkyZq06aNLl68qA8++ECJiYlKTEx0d3kAAKAScntgWbRokSSpT58+Tu3x8fGaOHGiJCkrK0uZmZmOYxcvXtS0adN0/Phx1axZU23atNHGjRs1aNAgd5cHAAAqIY8+dFuervWhnevBQ7cAgBtdlX3oFgAAwF0ILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYj8ACAACMR2ABAADGI7AAAADjEVgAAIDxCCwAAMB4BBYAAGA8AgsAADAegQUAABiPwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAYz2OB5e2331bTpk3l4+Ojjh07aufOnWX23759uzp27CgfHx81a9ZMixcv9lRpAACgkvFIYFm1apWefvppzZw5U2lpaerZs6ciIiKUmZlZYv+MjAwNGjRIPXv2VFpamv785z/rqaeeUmJioifKAwAAlYzNsizL3Sft2rWrOnTooEWLFjnaWrVqpeHDhysmJqZY/2effVbr16/X/v37HW1TpkzRN998o5SUlGt6zby8PPn5+Sk3N1e+vr6//yL+Q5MZG916PgAAKpvDrw72yHmv9f3b7SssFy9e1Ndff63w8HCn9vDwcH3++ecljklJSSnWf+DAgUpNTVVhYaG7SwQAAJWMl7tPeOrUKV2+fFkNGzZ0am/YsKGys7NLHJOdnV1i/0uXLunUqVMKDAwsNqagoEAFBQWO/dzcXEm/JjV3Kyo47/ZzAgBQmXji/fU/z3u1Gz5uDyxX2Gw2p33Lsoq1Xa1/Se1XxMTEaPbs2cXaQ0JCXC0VAABchd9cz54/Pz9ffn5+pR53e2C5+eabVb169WKrKSdPniy2inJFQEBAif29vLxUr169EsdER0crKirKsV9UVKSff/5Z9erVKzMYXU1eXp5CQkJ09OhRtz8Lg+KY7/LFfJcf5rp8Md/lx91zbVmW8vPzFRQUVGY/tweWGjVqqGPHjkpOTta9997raE9OTtawYcNKHNOtWzf97W9/c2rbvHmzOnXqJG9v7xLH2O122e12p7abbrrp9xX/H3x9ffmjL0fMd/livssPc12+mO/y4865Lmtl5QqPfKw5KipK7777ruLi4rR//34988wzyszM1JQpUyT9ujoyYcIER/8pU6boyJEjioqK0v79+xUXF6fY2FhNmzbNE+UBAIBKxiPPsIwZM0anT5/WSy+9pKysLLVt21abNm1SaGioJCkrK8vpO1maNm2qTZs26ZlnntHChQsVFBSkefPmaeTIkZ4oDwAAVDIee+j28ccf1+OPP17isYSEhGJtvXv31t69ez1VzjWz2+168cUXi91ugmcw3+WL+S4/zHX5Yr7LT0XNtUe+OA4AAMCd+PFDAABgPAILAAAwHoEFAAAYj8ACAACMd0MGlrfffltNmzaVj4+POnbsqJ07d5bZf/v27erYsaN8fHzUrFkzLV68uJwqrRpcme81a9ZowIABql+/vnx9fdWtWzd9+umn5Vht5ebq3/YVu3fvlpeXl+68807PFljFuDrfBQUFmjlzpkJDQ2W323XLLbcoLi6unKqt3Fyd62XLlqldu3aqVauWAgMD9dBDD+n06dPlVG3ltmPHDg0dOlRBQUGy2WxKSkq66phyeZ+0bjArV660vL29rSVLllj79u2zpk6datWuXds6cuRIif0PHTpk1apVy5o6daq1b98+a8mSJZa3t7f10UcflXPllZOr8z116lTrtddes7766ivrxx9/tKKjoy1vb29r79695Vx55ePqXF+Rk5NjNWvWzAoPD7fatWtXPsVWAdcz3/fcc4/VtWtXKzk52crIyLC+/PJLa/fu3eVYdeXk6lzv3LnTqlatmvXWW29Zhw4dsnbu3Gm1adPGGj58eDlXXjlt2rTJmjlzppWYmGhJstauXVtm//J6n7zhAkuXLl2sKVOmOLXddttt1owZM0rsP336dOu2225zanvsscessLAwj9VYlbg63yVp3bq1NXv2bHeXVuVc71yPGTPGeu6556wXX3yRwOICV+f7448/tvz8/KzTp0+XR3lViqtz/cYbb1jNmjVzaps3b54VHBzssRqrqmsJLOX1PnlD3RK6ePGivv76a4WHhzu1h4eH6/PPPy9xTEpKSrH+AwcOVGpqqgoLCz1Wa1VwPfP9W0VFRcrPz5e/v78nSqwyrneu4+Pj9c9//lMvvviip0usUq5nvtevX69OnTrp9ddfV6NGjdSiRQtNmzZNv/zyS3mUXGldz1x3795dx44d06ZNm2RZln766Sd99NFHGjx4cHmUfMMpr/dJj33TrYlOnTqly5cvF/vV6IYNGxb7tegrsrOzS+x/6dIlnTp1SoGBgR6rt7K7nvn+rb/85S86d+6cRo8e7YkSq4zrmesDBw5oxowZ2rlzp7y8bqh/Cn6365nvQ4cOadeuXfLx8dHatWt16tQpPf744/r55595jqUM1zPX3bt317JlyzRmzBhduHBBly5d0j333KP58+eXR8k3nPJ6n7yhVliusNlsTvuWZRVru1r/ktpRMlfn+4oVK1Zo1qxZWrVqlRo0aOCp8qqUa53ry5cva+zYsZo9e7ZatGhRXuVVOa78bRcVFclms2nZsmXq0qWLBg0apDfffFMJCQmsslwDV+Z63759euqpp/TCCy/o66+/1ieffKKMjAzHD/DC/crjffKG+t+qm2++WdWrVy+Wyk+ePFksHV4REBBQYn8vLy/Vq1fPY7VWBdcz31esWrVKkydP1ocffqj+/ft7sswqwdW5zs/PV2pqqtLS0vSnP/1J0q9vqJZlycvLS5s3b9Zdd91VLrVXRtfztx0YGKhGjRrJz8/P0daqVStZlqVjx46pefPmHq25srqeuY6JiVGPHj30P//zP5KkO+64Q7Vr11bPnj318ssvszLuZuX1PnlDrbDUqFFDHTt2VHJyslN7cnKyunfvXuKYbt26Feu/efNmderUSd7e3h6rtSq4nvmWfl1ZmThxopYvX84952vk6lz7+vrq22+/VXp6umObMmWKWrZsqfT0dHXt2rW8Sq+Urudvu0ePHjpx4oTOnj3raPvxxx9VrVo1BQcHe7Teyux65vr8+fOqVs357a169eqS/v1//nCfcnufdOsjvJXAlY/HxcbGWvv27bOefvppq3bt2tbhw4cty7KsGTNmWOPHj3f0v/JxrWeeecbat2+fFRsby8eaXeDqfC9fvtzy8vKyFi5caGVlZTm2nJycirqESsPVuf4tPiXkGlfnOz8/3woODrZGjRplfffdd9b27dut5s2bWw8//HBFXUKl4epcx8fHW15eXtbbb79t/fOf/7R27dplderUyerSpUtFXUKlkp+fb6WlpVlpaWmWJOvNN9+00tLSHB8jr6j3yRsusFiWZS1cuNAKDQ21atSoYXXo0MHavn2741hkZKTVu3dvp/7btm2z2rdvb9WoUcNq0qSJtWjRonKuuHJzZb579+5tSSq2RUZGln/hlZCrf9v/icDiOlfne//+/Vb//v2tmjVrWsHBwVZUVJR1/vz5cq66cnJ1rufNm2e1bt3aqlmzphUYGGiNGzfOOnbsWDlXXTlt3bq1zH+HK+p90mZZrI8BAACz3VDPsAAAgMqJwAIAAIxHYAEAAMYjsAAAAOMRWAAAgPEILAAAwHgEFgAAYDwCCwAAMB6BBUC5mDhxomw2m2w2m7y9vdWsWTNNmzZNu3fvls1m065du0ocN3DgQN1zzz2O/WPHjqlGjRq67bbbSuxvs9mUlJTkiUsAUIEILADKzd13362srCwdOnRIL7/8st5++2198MEHateuneLj44v1P3r0qLZs2aLJkyc72hISEjR69GidP39eu3fvLs/yAVQgAguAcmO32xUQEKCQkBCNHTtW48aNU1JSkiZPnqzVq1fr3LlzTv0TEhJUv359x692W5al+Ph4jR8/XmPHjlVsbGxFXAaACkBgAVBhatasqcLCQo0bN06FhYX68MMPHccsy1JCQoIiIyPl5eUlSdq6davOnz+v/v37a/z48Vq9erXy8/MrqnwA5YjAAqBCfPXVV1q+fLn69esnf39/DR8+3Om20LZt23To0CFNmjTJ0RYbG6v7779f1atXV5s2bXTrrbdq1apVFVE+gHJGYAFQbjZs2KA6derIx8dH3bp1U69evTR//nxJ0uTJk7Vjxw4dPHhQkhQXF6cePXqoZcuWkqScnBytWbNGDz74oON8Dz74oOLi4sr/QgCUO6+KLgDAjaNv375atGiRvL29FRQUJG9vb8ex/v37KzQ0VAkJCZo+fbrWrFmjBQsWOI4vX75cFy5cUNeuXR1tlmWpqKhI+/btU+vWrcv1WgCULwILgHJTu3Zt3XrrrSUes9lseuihh/Tuu+8qODhY1apV0+jRox3HY2Nj9d///d+aOHGi07innnpKcXFxmjNnjidLB1DBbJZlWRVdBICqb+LEicrJySnzO1IyMzPVtGlT+fn5aeTIkVqyZIkkKT09Xe3bt9f+/fuLff/KkiVLNHPmTB0/flze3t6y2Wx688031bdvX6d+t956q+rUqeP26wJQPniGBYAxGjdurP79++vMmTPFHrZt3bp1iV8WN3z4cP3888/629/+5miLiopS+/btnbbU1NRyuQYAnsEKCwAAMB4rLAAAwHgEFgAAYDwCCwAAMB6BBQAAGI/AAgAAjEdgAQAAxiOwAAAA4xFYAACA8QgsAADAeAQWAABgPAILAAAwHoEFAAAY7/8Bqveo5nrNNjMAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# make pval histogram and qval outputs\n",
    "for p in ['MergeEpoch_NG_Died_vs_Survived', 'MergeEpoch_SL_Died_vs_Survived',\n",
    "          'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl', 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl']:\n",
    "    output_directory = '/home/sraju/{rdir}/{p}/'.format(p=p, rdir=resdirmap[p])\n",
    "    make_hla_assoc_outputs(res = res_all[p], output_directory = output_directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# meta-analysis with METAL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# outcome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaAnalysis Helper - (c) 2007 - 2009 Goncalo Abecasis\n",
      "This version released on 2011-03-25\n",
      "\n",
      "# This program faciliates meta-analysis of genome-wide association studies.\n",
      "# Commonly used commands are listed below:\n",
      "#\n",
      "# Options for describing input files ...\n",
      "#   SEPARATOR        [WHITESPACE|COMMA|BOTH|TAB] (default = WHITESPACE)\n",
      "#   COLUMNCOUNTING   [STRICT|LENIENT]            (default = 'STRICT')\n",
      "#   MARKERLABEL      [LABEL]                     (default = 'MARKER')\n",
      "#   ALLELELABELS     [LABEL1 LABEL2]             (default = 'ALLELE1','ALLELE2')\n",
      "#   EFFECTLABEL      [LABEL|log(LABEL)]          (default = 'EFFECT')\n",
      "#   FLIP\n",
      "#\n",
      "# Options for filtering input files ...\n",
      "#   ADDFILTER        [LABEL CONDITION VALUE]     (example = ADDFILTER N > 10)\n",
      "#                    (available conditions are <, >, <=, >=, =, !=, IN)\n",
      "#   REMOVEFILTERS\n",
      "#\n",
      "# Options for sample size weighted meta-analysis ...\n",
      "#   WEIGHTLABEL      [LABEL]                     (default = 'N')\n",
      "#   PVALUELABEL      [LABEL]                     (default = 'PVALUE')\n",
      "#   DEFAULTWEIGHT    [NUMBER]                    (default = 1.0)\n",
      "#   MINWEIGHT        [NUMBER]                    (default = 1.0)\n",
      "#\n",
      "# Options for inverse variance weighted meta-analysis ...\n",
      "#   STDERRLABEL      [LABEL]                     (default = 'STDERR')\n",
      "#   SCHEME           [SAMPLESIZE|STDERR]         (default = SAMPLESIZE)\n",
      "#\n",
      "# Options to enable tracking of allele frequencies ...\n",
      "#   AVERAGEFREQ      [ON|OFF]                    (default = OFF)\n",
      "#   MINMAXFREQ       [ON|OFF]                    (default = OFF)\n",
      "#   FREQLABEL        [LABEL]                     (default = 'FREQ')\n",
      "#\n",
      "# Options to enable tracking of user defined variables ...\n",
      "#   CUSTOMVARIABLE   [VARNAME]\n",
      "#   LABEL            [VARNAME] AS [HEADER]\n",
      "#\n",
      "# Options to enable explicit strand information ...\n",
      "#   USESTRAND        [ON|OFF]                    (default = OFF)\n",
      "#   STRANDLABEL      [LABEL]                     (default = 'STRAND')\n",
      "#\n",
      "# Automatic genomic control correction of input statistics ...\n",
      "#   GENOMICCONTROL   [ON|OFF|VALUE|LIST snps.txt](default = OFF)\n",
      "#\n",
      "# Options for general analysis control ...\n",
      "#   PROCESSFILE      [FILENAME]\n",
      "#   OUTFILE          [PREFIX SUFFIX]             (default = 'METAANALYSIS','.TBL')\n",
      "#   MAXWARNINGS      [NUMBER]                    (default = 20)\n",
      "#   VERBOSE          [ON|OFF]                    (default = 'OFF')\n",
      "#   LOGPVALUE        [ON|OFF]                    (default = 'OFF')\n",
      "#   ANALYZE          [HETEROGENEITY]\n",
      "#   CLEAR\n",
      "\n",
      "# Options for general run control ...\n",
      "#   SOURCE           [SCRIPTFILE]\n",
      "#   RETURN\n",
      "#   QUIT\n",
      "\n",
      "# Processing commands in /home/sraju/meta_outcome.cmd ...\n",
      "## Set column separator to TAB ...\n",
      "## Verbose output during meta-analysis disabled\n",
      "## Set output file prefix and suffix to /home/sraju/outcome_H3covar_20220315/meta_analysis_HLA and .tsv ...\n",
      "## Set marker header to MarkerID ...\n",
      "## Set weight header to N ...\n",
      "## Set allele headers to Allele1 and Allele2 ...\n",
      "## Set effect header to BETA ...\n",
      "## Set standard error header to SE ...\n",
      "## Set p-value header to p.value ...\n",
      "###########################################################################\n",
      "## Processing file '/home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/combined_HLAs.tsv'\n",
      "## Processed 71 markers ...\n",
      "\n",
      "## Set marker header to MarkerID ...\n",
      "## Set weight header to N ...\n",
      "## Set allele headers to Allele1 and Allele2 ...\n",
      "## Set effect header to BETA ...\n",
      "## Set standard error header to SE ...\n",
      "## Set p-value header to p.value ...\n",
      "###########################################################################\n",
      "## Processing file '/home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/combined_HLAs.tsv'\n",
      "## Processed 23 markers ...\n",
      "\n",
      "###########################################################################\n",
      "## Executing meta-analysis ...\n",
      "## Complete results will be stored in file '/home/sraju/outcome_H3covar_20220315/meta_analysis_HLA1.tsv'\n",
      "## Column descriptions will be stored in file '/home/sraju/outcome_H3covar_20220315/meta_analysis_HLA1.tsv.info'\n",
      "## Completed meta-analysis for 71 markers!\n",
      "## Smallest p-value is 0.009338 at marker '13:01DPB1'\n",
      "\n",
      "## Clearing all stored statistics ...\n",
      "# Clearing user defined filters ...\n"
     ]
    }
   ],
   "source": [
    "cmd = \"\"\"#LOAD THE FIRST TWO INPUT FILES\n",
    "\n",
    "SEPARATOR TAB\n",
    "VERBOSE OFF\n",
    "#COLUMNCOUNTING LENIENT\n",
    "OUTFILE /home/sraju/outcome_H3covar_20220315/meta_analysis_HLA .tsv   \n",
    "\n",
    "# === DESCRIBE AND PROCESS THE FIRST INPUT FILE ===\n",
    "MARKER   MarkerID\n",
    "WEIGHT   N\n",
    "ALLELE   Allele1 Allele2\n",
    "EFFECT   BETA\n",
    "STDERR   SE\n",
    "PVAL     p.value\n",
    "\n",
    "PROCESS /home/sraju/outcome_H3covar_20220315/MergeEpoch_NG_Died_vs_Survived/combined_HLAs.tsv\n",
    "\n",
    "# === THE SECOND INPUT FILE HAS THE SAME FORMAT AND CAN BE PROCESSED IMMEDIATELY ===\n",
    "MARKER   MarkerID\n",
    "WEIGHT   N\n",
    "ALLELE   Allele1 Allele2\n",
    "EFFECT   BETA\n",
    "STDERR   SE\n",
    "PVAL     p.value\n",
    "\n",
    "PROCESS /home/sraju/outcome_H3covar_20220315/MergeEpoch_SL_Died_vs_Survived/combined_HLAs.tsv\n",
    "\n",
    "ANALYZE \n",
    "\"\"\"\n",
    "\n",
    "with open('/home/sraju/meta_outcome.cmd', 'w') as F:\n",
    "    F.write(cmd)\n",
    "    \n",
    "!{'/home/sraju/generic-metal/metal /home/sraju/meta_outcome.cmd'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# susceptibility"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaAnalysis Helper - (c) 2007 - 2009 Goncalo Abecasis\n",
      "This version released on 2011-03-25\n",
      "\n",
      "# This program faciliates meta-analysis of genome-wide association studies.\n",
      "# Commonly used commands are listed below:\n",
      "#\n",
      "# Options for describing input files ...\n",
      "#   SEPARATOR        [WHITESPACE|COMMA|BOTH|TAB] (default = WHITESPACE)\n",
      "#   COLUMNCOUNTING   [STRICT|LENIENT]            (default = 'STRICT')\n",
      "#   MARKERLABEL      [LABEL]                     (default = 'MARKER')\n",
      "#   ALLELELABELS     [LABEL1 LABEL2]             (default = 'ALLELE1','ALLELE2')\n",
      "#   EFFECTLABEL      [LABEL|log(LABEL)]          (default = 'EFFECT')\n",
      "#   FLIP\n",
      "#\n",
      "# Options for filtering input files ...\n",
      "#   ADDFILTER        [LABEL CONDITION VALUE]     (example = ADDFILTER N > 10)\n",
      "#                    (available conditions are <, >, <=, >=, =, !=, IN)\n",
      "#   REMOVEFILTERS\n",
      "#\n",
      "# Options for sample size weighted meta-analysis ...\n",
      "#   WEIGHTLABEL      [LABEL]                     (default = 'N')\n",
      "#   PVALUELABEL      [LABEL]                     (default = 'PVALUE')\n",
      "#   DEFAULTWEIGHT    [NUMBER]                    (default = 1.0)\n",
      "#   MINWEIGHT        [NUMBER]                    (default = 1.0)\n",
      "#\n",
      "# Options for inverse variance weighted meta-analysis ...\n",
      "#   STDERRLABEL      [LABEL]                     (default = 'STDERR')\n",
      "#   SCHEME           [SAMPLESIZE|STDERR]         (default = SAMPLESIZE)\n",
      "#\n",
      "# Options to enable tracking of allele frequencies ...\n",
      "#   AVERAGEFREQ      [ON|OFF]                    (default = OFF)\n",
      "#   MINMAXFREQ       [ON|OFF]                    (default = OFF)\n",
      "#   FREQLABEL        [LABEL]                     (default = 'FREQ')\n",
      "#\n",
      "# Options to enable tracking of user defined variables ...\n",
      "#   CUSTOMVARIABLE   [VARNAME]\n",
      "#   LABEL            [VARNAME] AS [HEADER]\n",
      "#\n",
      "# Options to enable explicit strand information ...\n",
      "#   USESTRAND        [ON|OFF]                    (default = OFF)\n",
      "#   STRANDLABEL      [LABEL]                     (default = 'STRAND')\n",
      "#\n",
      "# Automatic genomic control correction of input statistics ...\n",
      "#   GENOMICCONTROL   [ON|OFF|VALUE|LIST snps.txt](default = OFF)\n",
      "#\n",
      "# Options for general analysis control ...\n",
      "#   PROCESSFILE      [FILENAME]\n",
      "#   OUTFILE          [PREFIX SUFFIX]             (default = 'METAANALYSIS','.TBL')\n",
      "#   MAXWARNINGS      [NUMBER]                    (default = 20)\n",
      "#   VERBOSE          [ON|OFF]                    (default = 'OFF')\n",
      "#   LOGPVALUE        [ON|OFF]                    (default = 'OFF')\n",
      "#   ANALYZE          [HETEROGENEITY]\n",
      "#   CLEAR\n",
      "\n",
      "# Options for general run control ...\n",
      "#   SOURCE           [SCRIPTFILE]\n",
      "#   RETURN\n",
      "#   QUIT\n",
      "\n",
      "# Processing commands in /home/sraju/meta_suscept.cmd ...\n",
      "## Set column separator to TAB ...\n",
      "## Verbose output during meta-analysis disabled\n",
      "## Set output file prefix and suffix to /home/sraju/results_20220311/meta_analysis_HLA and .tsv ...\n",
      "## Set marker header to MarkerID ...\n",
      "## Set weight header to N ...\n",
      "## Set allele headers to Allele1 and Allele2 ...\n",
      "## Set effect header to BETA ...\n",
      "## Set standard error header to SE ...\n",
      "## Set p-value header to p.value ...\n",
      "###########################################################################\n",
      "## Processing file '/home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/combined_HLAs.tsv'\n",
      "## Processed 119 markers ...\n",
      "\n",
      "## Set marker header to MarkerID ...\n",
      "## Set weight header to N ...\n",
      "## Set allele headers to Allele1 and Allele2 ...\n",
      "## Set effect header to BETA ...\n",
      "## Set standard error header to SE ...\n",
      "## Set p-value header to p.value ...\n",
      "###########################################################################\n",
      "## Processing file '/home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/combined_HLAs.tsv'\n",
      "## Processed 110 markers ...\n",
      "\n",
      "###########################################################################\n",
      "## Executing meta-analysis ...\n",
      "## Complete results will be stored in file '/home/sraju/results_20220311/meta_analysis_HLA1.tsv'\n",
      "## Column descriptions will be stored in file '/home/sraju/results_20220311/meta_analysis_HLA1.tsv.info'\n",
      "## Completed meta-analysis for 133 markers!\n",
      "## Smallest p-value is 0.01106 at marker '15:16B'\n",
      "\n",
      "## Clearing all stored statistics ...\n",
      "# Clearing user defined filters ...\n"
     ]
    }
   ],
   "source": [
    "cmd = \"\"\"#LOAD THE FIRST TWO INPUT FILES\n",
    "\n",
    "SEPARATOR TAB\n",
    "VERBOSE OFF\n",
    "#COLUMNCOUNTING LENIENT\n",
    "OUTFILE /home/sraju/results_20220311/meta_analysis_HLA .tsv   \n",
    "\n",
    "# === DESCRIBE AND PROCESS THE FIRST INPUT FILE ===\n",
    "MARKER   MarkerID\n",
    "WEIGHT   N\n",
    "ALLELE   Allele1 Allele2\n",
    "EFFECT   BETA\n",
    "STDERR   SE\n",
    "PVAL     p.value\n",
    "\n",
    "PROCESS /home/sraju/results_20220311/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/combined_HLAs.tsv\n",
    "\n",
    "# === THE SECOND INPUT FILE HAS THE SAME FORMAT AND CAN BE PROCESSED IMMEDIATELY ===\n",
    "MARKER   MarkerID\n",
    "WEIGHT   N\n",
    "ALLELE   Allele1 Allele2\n",
    "EFFECT   BETA\n",
    "STDERR   SE\n",
    "PVAL     p.value\n",
    "\n",
    "PROCESS /home/sraju/results_20220311/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/combined_HLAs.tsv\n",
    "\n",
    "ANALYZE \n",
    "\"\"\"\n",
    "\n",
    "with open('/home/sraju/meta_suscept.cmd', 'w') as F:\n",
    "    F.write(cmd)\n",
    "    \n",
    "!{'/home/sraju/generic-metal/metal /home/sraju/meta_suscept.cmd'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_outcome = pd.read_csv('/home/sraju/outcome_H3covar_20220315/meta_analysis_HLA1.tsv', sep = '\\t')\n",
    "meta_outcome['qval'] = sm.fdrcorrection(meta_outcome['P-value'], alpha = 0.05)[1]\n",
    "meta_outcome.to_csv('/home/sraju/outcome_H3covar_20220315/meta_analysis_HLA1_withq.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_sus = pd.read_csv('/home/sraju/results_20220311/meta_analysis_HLA1.tsv', sep = '\\t')\n",
    "meta_sus['qval'] = sm.fdrcorrection(meta_sus['P-value'], alpha = 0.05)[1]\n",
    "meta_sus.to_csv('/home/sraju/results_20220311/meta_analysis_HLA1_withq.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdirmap = {'MergeEpoch_NG_Died_vs_Survived':'outcome_H3covar_20220315',\n",
    "         'MergeEpoch_SL_Died_vs_Survived':'outcome_H3covar_20220315',\n",
    "          'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl':'results_20220311',\n",
    "             'MergeEpoch_SL_AgORSeqPos_Vs_PopControl':'results_20220311'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MergeEpoch_NG_Died_vs_Survived\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerID</th>\n",
       "      <th>p.value</th>\n",
       "      <th>AF_case</th>\n",
       "      <th>AF_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MarkerID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13:01DPB1</th>\n",
       "      <td>13:01DPB1</td>\n",
       "      <td>0.009338</td>\n",
       "      <td>0.011434</td>\n",
       "      <td>0.047285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>07:02B</th>\n",
       "      <td>07:02B</td>\n",
       "      <td>0.030088</td>\n",
       "      <td>0.093188</td>\n",
       "      <td>0.058160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:01DRB1</th>\n",
       "      <td>11:01DRB1</td>\n",
       "      <td>0.032952</td>\n",
       "      <td>0.052760</td>\n",
       "      <td>0.108265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57:03B</th>\n",
       "      <td>57:03B</td>\n",
       "      <td>0.041747</td>\n",
       "      <td>0.077114</td>\n",
       "      <td>0.044965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>04:01C</th>\n",
       "      <td>04:01C</td>\n",
       "      <td>0.059026</td>\n",
       "      <td>0.212629</td>\n",
       "      <td>0.283003</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            MarkerID   p.value   AF_case   AF_ctrl\n",
       "MarkerID                                          \n",
       "13:01DPB1  13:01DPB1  0.009338  0.011434  0.047285\n",
       "07:02B        07:02B  0.030088  0.093188  0.058160\n",
       "11:01DRB1  11:01DRB1  0.032952  0.052760  0.108265\n",
       "57:03B        57:03B  0.041747  0.077114  0.044965\n",
       "04:01C        04:01C  0.059026  0.212629  0.283003"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MergeEpoch_SL_Died_vs_Survived\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerID</th>\n",
       "      <th>p.value</th>\n",
       "      <th>AF_case</th>\n",
       "      <th>AF_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MarkerID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>03:19DQB1</th>\n",
       "      <td>03:19DQB1</td>\n",
       "      <td>0.021013</td>\n",
       "      <td>0.114570</td>\n",
       "      <td>0.201809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05:05DQA1</th>\n",
       "      <td>05:05DQA1</td>\n",
       "      <td>0.073178</td>\n",
       "      <td>0.149057</td>\n",
       "      <td>0.230483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53:01B</th>\n",
       "      <td>53:01B</td>\n",
       "      <td>0.080190</td>\n",
       "      <td>0.129845</td>\n",
       "      <td>0.065694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17:01C</th>\n",
       "      <td>17:01C</td>\n",
       "      <td>0.090293</td>\n",
       "      <td>0.150084</td>\n",
       "      <td>0.040762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>03:01DPA1</th>\n",
       "      <td>03:01DPA1</td>\n",
       "      <td>0.136209</td>\n",
       "      <td>0.132936</td>\n",
       "      <td>0.092926</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            MarkerID   p.value   AF_case   AF_ctrl\n",
       "MarkerID                                          \n",
       "03:19DQB1  03:19DQB1  0.021013  0.114570  0.201809\n",
       "05:05DQA1  05:05DQA1  0.073178  0.149057  0.230483\n",
       "53:01B        53:01B  0.080190  0.129845  0.065694\n",
       "17:01C        17:01C  0.090293  0.150084  0.040762\n",
       "03:01DPA1  03:01DPA1  0.136209  0.132936  0.092926"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerID</th>\n",
       "      <th>p.value</th>\n",
       "      <th>AF_case</th>\n",
       "      <th>AF_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MarkerID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>104:01DPB1</th>\n",
       "      <td>104:01DPB1</td>\n",
       "      <td>0.008465</td>\n",
       "      <td>0.027175</td>\n",
       "      <td>0.038900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39:10B</th>\n",
       "      <td>39:10B</td>\n",
       "      <td>0.012090</td>\n",
       "      <td>0.005909</td>\n",
       "      <td>0.010816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:16B</th>\n",
       "      <td>15:16B</td>\n",
       "      <td>0.012384</td>\n",
       "      <td>0.032253</td>\n",
       "      <td>0.024621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14:02C</th>\n",
       "      <td>14:02C</td>\n",
       "      <td>0.025946</td>\n",
       "      <td>0.024108</td>\n",
       "      <td>0.016713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:03C</th>\n",
       "      <td>12:03C</td>\n",
       "      <td>0.026699</td>\n",
       "      <td>0.010147</td>\n",
       "      <td>0.016972</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              MarkerID   p.value   AF_case   AF_ctrl\n",
       "MarkerID                                            \n",
       "104:01DPB1  104:01DPB1  0.008465  0.027175  0.038900\n",
       "39:10B          39:10B  0.012090  0.005909  0.010816\n",
       "15:16B          15:16B  0.012384  0.032253  0.024621\n",
       "14:02C          14:02C  0.025946  0.024108  0.016713\n",
       "12:03C          12:03C  0.026699  0.010147  0.016972"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MergeEpoch_SL_AgORSeqPos_Vs_PopControl\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerID</th>\n",
       "      <th>p.value</th>\n",
       "      <th>AF_case</th>\n",
       "      <th>AF_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MarkerID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>05:01DQB1</th>\n",
       "      <td>05:01DQB1</td>\n",
       "      <td>0.008330</td>\n",
       "      <td>0.183992</td>\n",
       "      <td>0.114109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>07:01C</th>\n",
       "      <td>07:01C</td>\n",
       "      <td>0.011715</td>\n",
       "      <td>0.078152</td>\n",
       "      <td>0.040949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>02:01DPA1</th>\n",
       "      <td>02:01DPA1</td>\n",
       "      <td>0.026820</td>\n",
       "      <td>0.505705</td>\n",
       "      <td>0.436557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>02:01A</th>\n",
       "      <td>02:01A</td>\n",
       "      <td>0.041370</td>\n",
       "      <td>0.171894</td>\n",
       "      <td>0.144997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>02:02DPA1</th>\n",
       "      <td>02:02DPA1</td>\n",
       "      <td>0.044274</td>\n",
       "      <td>0.213948</td>\n",
       "      <td>0.283172</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            MarkerID   p.value   AF_case   AF_ctrl\n",
       "MarkerID                                          \n",
       "05:01DQB1  05:01DQB1  0.008330  0.183992  0.114109\n",
       "07:01C        07:01C  0.011715  0.078152  0.040949\n",
       "02:01DPA1  02:01DPA1  0.026820  0.505705  0.436557\n",
       "02:01A        02:01A  0.041370  0.171894  0.144997\n",
       "02:02DPA1  02:02DPA1  0.044274  0.213948  0.283172"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res = {}\n",
    "for p in ['MergeEpoch_NG_Died_vs_Survived', 'MergeEpoch_SL_Died_vs_Survived',\n",
    "          'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl', 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl']:\n",
    "    \n",
    "    fn = 'gs://popgen-gwas/20230824_HLA/{rdir}/{p}/combined_HLAs.tsv'.format(p=p, rdir=resdirmap[p])\n",
    "    res[p] = pd.read_csv(fn, sep='\\t')\n",
    "    res[p].index = res[p]['MarkerID']\n",
    "    res[p]['OR'] = np.exp(res[p]['BETA'])\n",
    "    print(p)\n",
    "    display(res[p][['MarkerID', 'p.value', 'AF_case', 'AF_ctrl']].sort_values(by='p.value').head(5))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Weight</th>\n",
       "      <th>Zscore</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Direction</th>\n",
       "      <th>qval</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MarkerName</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15:16B</th>\n",
       "      <td>15:16B</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>2519.0</td>\n",
       "      <td>2.541</td>\n",
       "      <td>0.01106</td>\n",
       "      <td>++</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:03DRB1</th>\n",
       "      <td>15:03DRB1</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>2519.0</td>\n",
       "      <td>2.473</td>\n",
       "      <td>0.01338</td>\n",
       "      <td>++</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39:10B</th>\n",
       "      <td>39:10B</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>1598.0</td>\n",
       "      <td>-2.510</td>\n",
       "      <td>0.01209</td>\n",
       "      <td>-?</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>02:01DPA1</th>\n",
       "      <td>02:01DPA1</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>2519.0</td>\n",
       "      <td>2.373</td>\n",
       "      <td>0.01764</td>\n",
       "      <td>++</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:03C</th>\n",
       "      <td>12:03C</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>1598.0</td>\n",
       "      <td>-2.216</td>\n",
       "      <td>0.02670</td>\n",
       "      <td>-?</td>\n",
       "      <td>0.591850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14:02C</th>\n",
       "      <td>14:02C</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>2519.0</td>\n",
       "      <td>2.281</td>\n",
       "      <td>0.02255</td>\n",
       "      <td>++</td>\n",
       "      <td>0.591850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>08:06DRB1</th>\n",
       "      <td>08:06DRB1</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>921.0</td>\n",
       "      <td>-1.964</td>\n",
       "      <td>0.04954</td>\n",
       "      <td>?-</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01:02A</th>\n",
       "      <td>01:02A</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>921.0</td>\n",
       "      <td>1.977</td>\n",
       "      <td>0.04803</td>\n",
       "      <td>?+</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35:01B</th>\n",
       "      <td>35:01B</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>2519.0</td>\n",
       "      <td>-2.036</td>\n",
       "      <td>0.04175</td>\n",
       "      <td>--</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           MarkerName Allele1 Allele2  Weight  Zscore  P-value Direction  \\\n",
       "MarkerName                                                                 \n",
       "15:16B         15:16B       a       g  2519.0   2.541  0.01106        ++   \n",
       "15:03DRB1   15:03DRB1       a       g  2519.0   2.473  0.01338        ++   \n",
       "39:10B         39:10B       a       g  1598.0  -2.510  0.01209        -?   \n",
       "02:01DPA1   02:01DPA1       a       g  2519.0   2.373  0.01764        ++   \n",
       "12:03C         12:03C       a       g  1598.0  -2.216  0.02670        -?   \n",
       "14:02C         14:02C       a       g  2519.0   2.281  0.02255        ++   \n",
       "08:06DRB1   08:06DRB1       a       g   921.0  -1.964  0.04954        ?-   \n",
       "01:02A         01:02A       a       g   921.0   1.977  0.04803        ?+   \n",
       "35:01B         35:01B       a       g  2519.0  -2.036  0.04175        --   \n",
       "\n",
       "                qval  \n",
       "MarkerName            \n",
       "15:16B      0.586530  \n",
       "15:03DRB1   0.586530  \n",
       "39:10B      0.586530  \n",
       "02:01DPA1   0.586530  \n",
       "12:03C      0.591850  \n",
       "14:02C      0.591850  \n",
       "08:06DRB1   0.692797  \n",
       "01:02A      0.692797  \n",
       "35:01B      0.692797  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sus_meta = pd.read_csv('gs://popgen-gwas/20230824_HLA/results_20220311/meta_analysis_HLA1_withq.tsv', sep='\\t', index_col=0)\n",
    "sus_meta.index = sus_meta['MarkerName']\n",
    "sus_meta = sus_meta.sort_values(by='qval')\n",
    "sus_meta_sig = sus_meta.loc[sus_meta['P-value']<.05, :]\n",
    "sus_meta_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">Nigeria</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Sierra Leone</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Meta-analysis</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Q-value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MarkerName</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15:16B</th>\n",
       "      <td>0.026584</td>\n",
       "      <td>3.020614</td>\n",
       "      <td>0.012384</td>\n",
       "      <td>0.018076</td>\n",
       "      <td>1.776452</td>\n",
       "      <td>0.364046</td>\n",
       "      <td>0.01106</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:03DRB1</th>\n",
       "      <td>0.243574</td>\n",
       "      <td>1.319887</td>\n",
       "      <td>0.089463</td>\n",
       "      <td>0.022443</td>\n",
       "      <td>2.598153</td>\n",
       "      <td>0.063795</td>\n",
       "      <td>0.01338</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>02:01DPA1</th>\n",
       "      <td>0.354312</td>\n",
       "      <td>1.203705</td>\n",
       "      <td>0.194115</td>\n",
       "      <td>0.445717</td>\n",
       "      <td>1.451697</td>\n",
       "      <td>0.026820</td>\n",
       "      <td>0.01764</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14:02C</th>\n",
       "      <td>0.018615</td>\n",
       "      <td>3.215169</td>\n",
       "      <td>0.025946</td>\n",
       "      <td>0.019571</td>\n",
       "      <td>1.665962</td>\n",
       "      <td>0.401618</td>\n",
       "      <td>0.02255</td>\n",
       "      <td>0.591850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35:01B</th>\n",
       "      <td>0.088392</td>\n",
       "      <td>0.671095</td>\n",
       "      <td>0.107814</td>\n",
       "      <td>0.144118</td>\n",
       "      <td>0.742437</td>\n",
       "      <td>0.211714</td>\n",
       "      <td>0.04175</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Nigeria                          Sierra Leone             \\\n",
       "           Allele Frequency Odds Ratio   P-value Allele Frequency Odds Ratio   \n",
       "MarkerName                                                                     \n",
       "15:16B             0.026584   3.020614  0.012384         0.018076   1.776452   \n",
       "15:03DRB1          0.243574   1.319887  0.089463         0.022443   2.598153   \n",
       "02:01DPA1          0.354312   1.203705  0.194115         0.445717   1.451697   \n",
       "14:02C             0.018615   3.215169  0.025946         0.019571   1.665962   \n",
       "35:01B             0.088392   0.671095  0.107814         0.144118   0.742437   \n",
       "\n",
       "                     Meta-analysis            \n",
       "             P-value       P-value   Q-value  \n",
       "MarkerName                                    \n",
       "15:16B      0.364046       0.01106  0.586530  \n",
       "15:03DRB1   0.063795       0.01338  0.586530  \n",
       "02:01DPA1   0.026820       0.01764  0.586530  \n",
       "14:02C      0.401618       0.02255  0.591850  \n",
       "35:01B      0.211714       0.04175  0.692797  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.concat([ res['MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl'].loc[sus_meta_sig.index, ['AF_Allele2', 'OR', 'p.value']],\n",
    "            res['MergeEpoch_SL_AgORSeqPos_Vs_PopControl'].loc[sus_meta_sig.index, ['AF_Allele2', 'OR', 'p.value']],\n",
    "           sus_meta_sig[['P-value', 'qval']]], axis=1)\n",
    "\n",
    "\n",
    "merged.columns = [('Nigeria', 'Allele Frequency'), ('Nigeria', 'Odds Ratio'), ('Nigeria', 'P-value'),\n",
    "                 ('Sierra Leone', 'Allele Frequency'), ('Sierra Leone', 'Odds Ratio'), ('Sierra Leone', 'P-value'),\n",
    "                 ('Meta-analysis', 'P-value'), ('Meta-analysis', 'Q-value')]\n",
    "\n",
    "merged.columns = pd.MultiIndex.from_tuples(merged.columns)\n",
    "merged.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Nigeria        Allele Frequency    0.354312\n",
       "               Odds Ratio          1.203705\n",
       "               P-value             0.194115\n",
       "Sierra Leone   Allele Frequency    0.445717\n",
       "               Odds Ratio          1.451697\n",
       "               P-value             0.026820\n",
       "Meta-analysis  P-value             0.017640\n",
       "               Q-value             0.586530\n",
       "Name: 02:01DPA1, dtype: float64"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.loc['02:01DPA1', :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['15:16B', '15:03DRB1', '39:10B', '02:01DPA1', '12:03C', '14:02C',\n",
       "       '08:06DRB1', '01:02A', '35:01B'],\n",
       "      dtype='object', name='MarkerName')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged.index = ['B*15:16', 'DRB1*15:03', 'B*39:10', 'DPA1*02:01', 'C*12:03', 'C*14:02',\n",
    "       'DRB1*08:06', 'A*01:02', 'B*35:01']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">Nigeria</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Sierra Leone</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Meta-analysis</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Q-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>B*15:16</th>\n",
       "      <td>0.026584</td>\n",
       "      <td>1.105460</td>\n",
       "      <td>0.012384</td>\n",
       "      <td>0.018076</td>\n",
       "      <td>0.574618</td>\n",
       "      <td>0.364046</td>\n",
       "      <td>0.01106</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*15:03</th>\n",
       "      <td>0.243574</td>\n",
       "      <td>0.277546</td>\n",
       "      <td>0.089463</td>\n",
       "      <td>0.022443</td>\n",
       "      <td>0.954801</td>\n",
       "      <td>0.063795</td>\n",
       "      <td>0.01338</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*39:10</th>\n",
       "      <td>0.009554</td>\n",
       "      <td>-1.989230</td>\n",
       "      <td>0.012090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01209</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPA1*02:01</th>\n",
       "      <td>0.354312</td>\n",
       "      <td>0.185404</td>\n",
       "      <td>0.194115</td>\n",
       "      <td>0.445717</td>\n",
       "      <td>0.372733</td>\n",
       "      <td>0.026820</td>\n",
       "      <td>0.01764</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*12:03</th>\n",
       "      <td>0.015216</td>\n",
       "      <td>-1.349210</td>\n",
       "      <td>0.026699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02670</td>\n",
       "      <td>0.591850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*14:02</th>\n",
       "      <td>0.018615</td>\n",
       "      <td>1.167880</td>\n",
       "      <td>0.025946</td>\n",
       "      <td>0.019571</td>\n",
       "      <td>0.510403</td>\n",
       "      <td>0.401618</td>\n",
       "      <td>0.02255</td>\n",
       "      <td>0.591850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*08:06</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.022114</td>\n",
       "      <td>-1.167890</td>\n",
       "      <td>0.049540</td>\n",
       "      <td>0.04954</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A*01:02</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.015443</td>\n",
       "      <td>1.344970</td>\n",
       "      <td>0.048033</td>\n",
       "      <td>0.04803</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*35:01</th>\n",
       "      <td>0.088392</td>\n",
       "      <td>-0.398844</td>\n",
       "      <td>0.107814</td>\n",
       "      <td>0.144118</td>\n",
       "      <td>-0.297817</td>\n",
       "      <td>0.211714</td>\n",
       "      <td>0.04175</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Nigeria                          Sierra Leone             \\\n",
       "           Allele Frequency Odds Ratio   P-value Allele Frequency Odds Ratio   \n",
       "B*15:16            0.026584   1.105460  0.012384         0.018076   0.574618   \n",
       "DRB1*15:03         0.243574   0.277546  0.089463         0.022443   0.954801   \n",
       "B*39:10            0.009554  -1.989230  0.012090              NaN        NaN   \n",
       "DPA1*02:01         0.354312   0.185404  0.194115         0.445717   0.372733   \n",
       "C*12:03            0.015216  -1.349210  0.026699              NaN        NaN   \n",
       "C*14:02            0.018615   1.167880  0.025946         0.019571   0.510403   \n",
       "DRB1*08:06              NaN        NaN       NaN         0.022114  -1.167890   \n",
       "A*01:02                 NaN        NaN       NaN         0.015443   1.344970   \n",
       "B*35:01            0.088392  -0.398844  0.107814         0.144118  -0.297817   \n",
       "\n",
       "                     Meta-analysis            \n",
       "             P-value       P-value   Q-value  \n",
       "B*15:16     0.364046       0.01106  0.586530  \n",
       "DRB1*15:03  0.063795       0.01338  0.586530  \n",
       "B*39:10          NaN       0.01209  0.586530  \n",
       "DPA1*02:01  0.026820       0.01764  0.586530  \n",
       "C*12:03          NaN       0.02670  0.591850  \n",
       "C*14:02     0.401618       0.02255  0.591850  \n",
       "DRB1*08:06  0.049540       0.04954  0.692797  \n",
       "A*01:02     0.048033       0.04803  0.692797  \n",
       "B*35:01     0.211714       0.04175  0.692797  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">Nigeria</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Sierra Leone</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Meta-analysis</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>P-value</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Q-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>B*15:16</th>\n",
       "      <td>0.026584</td>\n",
       "      <td>1.105460</td>\n",
       "      <td>0.012384</td>\n",
       "      <td>0.018076</td>\n",
       "      <td>0.574618</td>\n",
       "      <td>0.364046</td>\n",
       "      <td>0.01106</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*15:03</th>\n",
       "      <td>0.243574</td>\n",
       "      <td>0.277546</td>\n",
       "      <td>0.089463</td>\n",
       "      <td>0.022443</td>\n",
       "      <td>0.954801</td>\n",
       "      <td>0.063795</td>\n",
       "      <td>0.01338</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPA1*02:01</th>\n",
       "      <td>0.354312</td>\n",
       "      <td>0.185404</td>\n",
       "      <td>0.194115</td>\n",
       "      <td>0.445717</td>\n",
       "      <td>0.372733</td>\n",
       "      <td>0.026820</td>\n",
       "      <td>0.01764</td>\n",
       "      <td>0.586530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*14:02</th>\n",
       "      <td>0.018615</td>\n",
       "      <td>1.167880</td>\n",
       "      <td>0.025946</td>\n",
       "      <td>0.019571</td>\n",
       "      <td>0.510403</td>\n",
       "      <td>0.401618</td>\n",
       "      <td>0.02255</td>\n",
       "      <td>0.591850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*35:01</th>\n",
       "      <td>0.088392</td>\n",
       "      <td>-0.398844</td>\n",
       "      <td>0.107814</td>\n",
       "      <td>0.144118</td>\n",
       "      <td>-0.297817</td>\n",
       "      <td>0.211714</td>\n",
       "      <td>0.04175</td>\n",
       "      <td>0.692797</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Nigeria                          Sierra Leone             \\\n",
       "           Allele Frequency Odds Ratio   P-value Allele Frequency Odds Ratio   \n",
       "B*15:16            0.026584   1.105460  0.012384         0.018076   0.574618   \n",
       "DRB1*15:03         0.243574   0.277546  0.089463         0.022443   0.954801   \n",
       "DPA1*02:01         0.354312   0.185404  0.194115         0.445717   0.372733   \n",
       "C*14:02            0.018615   1.167880  0.025946         0.019571   0.510403   \n",
       "B*35:01            0.088392  -0.398844  0.107814         0.144118  -0.297817   \n",
       "\n",
       "                     Meta-analysis            \n",
       "             P-value       P-value   Q-value  \n",
       "B*15:16     0.364046       0.01106  0.586530  \n",
       "DRB1*15:03  0.063795       0.01338  0.586530  \n",
       "DPA1*02:01  0.026820       0.01764  0.586530  \n",
       "C*14:02     0.401618       0.02255  0.591850  \n",
       "B*35:01     0.211714       0.04175  0.692797  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
