{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from statsmodels.stats.multitest import fdrcorrection, multipletests\n",
    "import seaborn as sns\n",
    "\n",
    "from matplotlib import rcParams, font_manager\n",
    "rcParams['pdf.fonttype'] = 42\n",
    "rcParams['ps.fonttype'] = 42\n",
    "\n",
    "figuredir = '../../figures'\n",
    "if not os.path.exists(figuredir):\n",
    "    os.mkdir(figuredir)\n",
    "    \n",
    "assocdir = '../../data/tmp/HLA/Association'\n",
    "if not os.path.exists(assocdir):\n",
    "    os.mkdir(assocdir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "hlabase = '../../data/tmp/HLA/Stanford/Predictions/Omni25AffyMerged.{g}.Nclass_7.Flank_500K.4dig.Model.20220206.AllProbabilities.tsv'\n",
    "genes = ['A', 'B', 'C', 'DRB1', 'DQB1', 'DQA1', 'DPB1', 'DPA1']\n",
    "all_allele_data = []\n",
    "import sys\n",
    "for g in genes:\n",
    "    hlafn = hlabase.format(g=g)\n",
    "    hladat = pd.read_csv(hlafn, sep='\\t')\n",
    "    x1 = hladat.groupby(lambda x: x.split('/')[0], axis=0).sum()\n",
    "    x2 = hladat.groupby(lambda x: x.split('/')[1], axis=0).sum()\n",
    "    all_alleles = sorted(set(x1.index).union(x2.index))\n",
    "    dosage = x1.loc[all_alleles, :].replace(np.nan, 0) + x2.loc[all_alleles, :].replace(np.nan, 0)\n",
    "    dosage.index = [g+'*'+x for x in dosage.index]\n",
    "    all_allele_data.append(dosage.T)\n",
    "    \n",
    "all_allele_data = pd.concat(all_allele_data, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DQA1*03:02    0.000010\n",
       "DPB1*05:01    0.000015\n",
       "C*12:02       0.000019\n",
       "DPB1*02:02    0.000022\n",
       "A*25:01       0.000031\n",
       "dtype: float64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AF = all_allele_data.sum(axis=0) / (all_allele_data.shape[0]*2)\n",
    "AF.sort_values().head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "dosagefn = os.path.join(assocdir, 'AllGene_Dosage.tsv')\n",
    "all_allele_data.to_csv(dosagefn, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_allele_data_maf01 = all_allele_data.loc[:, (AF>.01) & (AF<.99)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "dosagefn = os.path.join(assocdir, 'AllGene_Dosage_Maf_01.tsv')\n",
    "all_allele_data_maf01.to_csv(dosagefn, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "phenotypes = ['MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl', 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl']\n",
    "covarmap = {'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl': ['SEX', 'PCs'],\n",
    " 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl': ['SEX','PCs']}\n",
    "\n",
    "tmpdir = '../../data/tmp/'\n",
    "gwasdir = os.path.join(tmpdir, 'GWAS')\n",
    "resdir = os.path.join(gwasdir, 'results_noH3covar_20220207')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    1598\n",
       "Name: IID, dtype: int64"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript ../../code/gwas_analysis/run_GMMAT_premade_model_text_dosage.R ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/gmmat.SEXPCs.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.nullmodel.rds ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.tsv ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.results.tsv\n",
      "Warning message:\n",
      "In glmm.score(model0, infile = genotypefn, outfile = outfn, is.dosage = FALSE,  :\n",
      "  Argument select is unspecified... Assuming the order of individuals in infile matches unique id_include in obj...\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True    921\n",
       "Name: IID, dtype: int64"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript ../../code/gwas_analysis/run_GMMAT_premade_model_text_dosage.R ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/gmmat.SEXPCs.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.nullmodel.rds ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.tsv ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.results.tsv\n",
      "Warning message:\n",
      "In glmm.score(model0, infile = genotypefn, outfile = outfn, is.dosage = FALSE,  :\n",
      "  Argument select is unspecified... Assuming the order of individuals in infile matches unique id_include in obj...\n"
     ]
    }
   ],
   "source": [
    "\n",
    "results = {}\n",
    "AF_thresh = .01\n",
    "AF_both = {}\n",
    "for p in phenotypes:\n",
    "    pdir = '{base}/{p}'.format(base=resdir, p=p)\n",
    "    phenoresdir = os.path.join(pdir, 'results')\n",
    "    rawgwasdir = os.path.join(phenoresdir, 'raw_gwas')\n",
    "\n",
    "    covarstr = ''.join(covarmap[p])\n",
    "    nulloutfile = os.path.join(rawgwasdir, \"gmmat.%s.%s.nullmodel.rds\" % (covarstr, p))\n",
    "    phenofn = os.path.join(pdir, 'Covar_Pheno_UnrelatedPCs.txt')\n",
    "    relfn = os.path.join(pdir, 'rel/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220107.Deduped.{p}.maf_1e-2.hwe_1e-6.geno_5e-2.rel.withids.tsv').format(p=p)\n",
    "    pheno = pd.read_csv(phenofn, sep='\\t')\n",
    "    rel = pd.read_csv(relfn, sep='\\t', index_col=0)\n",
    "    display((pheno['IID']==rel.index).value_counts())\n",
    "    genot = all_allele_data.loc[rel.index,:].T\n",
    "    AF = genot.sum(axis=1) / (genot.shape[1]*2)\n",
    "    AF_both[p] = AF\n",
    "    genot = genot.loc[(AF>AF_thresh) & (AF < (1- AF_thresh)), :]    \n",
    "    samps = list(rel.index)\n",
    "    genot['SNP'] = genot.index\n",
    "    genot['REF'] = 'A'\n",
    "    genot['ALT'] = 'T'\n",
    "    genot = genot.loc[:, ['SNP', 'REF', 'ALT'] + samps]\n",
    "    genotfn = os.path.join(assocdir, 'AllGene_Dosage_Maf_01.{p}.tsv'.format(p=p))\n",
    "    genot.to_csv(genotfn, sep='\\t', index=False)\n",
    "    outfn = os.path.join(assocdir, 'AllGene_Dosage_Maf_01.{p}.results.tsv'.format(p=p))    \n",
    "    cmd = 'Rscript ../../code/gwas_analysis/run_GMMAT_premade_model_text_dosage.R %s %s %s' % (nulloutfile, genotfn, outfn)\n",
    "    print(cmd)\n",
    "    !{cmd}\n",
    "    results[p] = pd.read_csv(outfn, sep='\\t')\n",
    "    results[p].index = results[p]['SNP']\n",
    "    _,q,_,_ = multipletests(results[p]['PVAL'].values, method='fdr_bh')\n",
    "    results[p]['QVAL'] = q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>N</th>\n",
       "      <th>AF</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>VAR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C*14:02</th>\n",
       "      <td>C*14:02</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.018615</td>\n",
       "      <td>7.05560</td>\n",
       "      <td>6.09635</td>\n",
       "      <td>0.004269</td>\n",
       "      <td>0.381949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*15:16</th>\n",
       "      <td>B*15:16</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.026584</td>\n",
       "      <td>8.06405</td>\n",
       "      <td>8.94377</td>\n",
       "      <td>0.007008</td>\n",
       "      <td>0.381949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*52:01</th>\n",
       "      <td>B*52:01</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.014350</td>\n",
       "      <td>-4.81087</td>\n",
       "      <td>4.00562</td>\n",
       "      <td>0.016228</td>\n",
       "      <td>0.584849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A*02:01</th>\n",
       "      <td>A*02:01</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.096311</td>\n",
       "      <td>-11.67760</td>\n",
       "      <td>29.38770</td>\n",
       "      <td>0.031230</td>\n",
       "      <td>0.584849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQA1*05:01</th>\n",
       "      <td>DQA1*05:01</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.051717</td>\n",
       "      <td>-8.47301</td>\n",
       "      <td>17.27290</td>\n",
       "      <td>0.041479</td>\n",
       "      <td>0.584849</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP     N        AF     SCORE       VAR      PVAL      QVAL\n",
       "SNP                                                                           \n",
       "C*14:02        C*14:02  1598  0.018615   7.05560   6.09635  0.004269  0.381949\n",
       "B*15:16        B*15:16  1598  0.026584   8.06405   8.94377  0.007008  0.381949\n",
       "B*52:01        B*52:01  1598  0.014350  -4.81087   4.00562  0.016228  0.584849\n",
       "A*02:01        A*02:01  1598  0.096311 -11.67760  29.38770  0.031230  0.584849\n",
       "DQA1*05:01  DQA1*05:01  1598  0.051717  -8.47301  17.27290  0.041479  0.584849"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "MergeEpoch_SL_AgORSeqPos_Vs_PopControl\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>N</th>\n",
       "      <th>AF</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>VAR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DPA1*02:02</th>\n",
       "      <td>DPA1*02:02</td>\n",
       "      <td>921</td>\n",
       "      <td>0.274253</td>\n",
       "      <td>-14.32660</td>\n",
       "      <td>26.92320</td>\n",
       "      <td>0.005761</td>\n",
       "      <td>0.378817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQB1*05:01</th>\n",
       "      <td>DQB1*05:01</td>\n",
       "      <td>921</td>\n",
       "      <td>0.123551</td>\n",
       "      <td>11.21200</td>\n",
       "      <td>18.08820</td>\n",
       "      <td>0.008383</td>\n",
       "      <td>0.378817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPA1*02:01</th>\n",
       "      <td>DPA1*02:01</td>\n",
       "      <td>921</td>\n",
       "      <td>0.445456</td>\n",
       "      <td>15.57400</td>\n",
       "      <td>38.93680</td>\n",
       "      <td>0.012566</td>\n",
       "      <td>0.378817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*07:01</th>\n",
       "      <td>C*07:01</td>\n",
       "      <td>921</td>\n",
       "      <td>0.045904</td>\n",
       "      <td>6.47016</td>\n",
       "      <td>6.99465</td>\n",
       "      <td>0.014428</td>\n",
       "      <td>0.378817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*15:03</th>\n",
       "      <td>DRB1*15:03</td>\n",
       "      <td>921</td>\n",
       "      <td>0.022577</td>\n",
       "      <td>4.52815</td>\n",
       "      <td>3.58392</td>\n",
       "      <td>0.016762</td>\n",
       "      <td>0.378817</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP    N        AF     SCORE       VAR      PVAL      QVAL\n",
       "SNP                                                                          \n",
       "DPA1*02:02  DPA1*02:02  921  0.274253 -14.32660  26.92320  0.005761  0.378817\n",
       "DQB1*05:01  DQB1*05:01  921  0.123551  11.21200  18.08820  0.008383  0.378817\n",
       "DPA1*02:01  DPA1*02:01  921  0.445456  15.57400  38.93680  0.012566  0.378817\n",
       "C*07:01        C*07:01  921  0.045904   6.47016   6.99465  0.014428  0.378817\n",
       "DRB1*15:03  DRB1*15:03  921  0.022577   4.52815   3.58392  0.016762  0.378817"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for p in phenotypes:\n",
    "    print(p)\n",
    "    display(results[p].sort_values(by='PVAL').head(5))\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "in1 = '../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.results.tsv'\n",
    "in2 = '../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.results.tsv'\n",
    "outfn = '../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.NG_SL_Susceptibility_MetaAnalysis.results.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for fn in [in1, in2]:\n",
    "    dat = pd.read_csv(fn, sep='\\t')\n",
    "    ## The run_GMMAT_meta.R script expects a REF and ALT column\n",
    "    dat['REF'] = 'A'\n",
    "    dat['ALT'] = 'T'\n",
    "    dat = dat[['SNP', 'REF', 'ALT', 'N', 'AF', 'SCORE', 'VAR', 'PVAL']]\n",
    "    dat.to_csv(fn.replace('.tsv', '.fixed.tsv'), sep='\\t', index=False)\n",
    "\n",
    "Rcmd = 'Rscript ../../code/gwas_analysis/run_GMMAT_meta.R %s %s %s' % (in1.replace('.tsv', '.fixed.tsv'), in2.replace('.tsv', '.fixed.tsv'), outfn)\n",
    "print(Rcmd)\n",
    "!{Rcmd}\n",
    "\n",
    "cmd = 'rm ../../data/tmp/HLA/Association/*.fixed.tsv'\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/py36/lib/python3.6/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \"\"\"\n"
     ]
    }
   ],
   "source": [
    "meta = pd.read_csv(outfn, sep='\\t')\n",
    "meta.index = meta['SNP']\n",
    "meta_both = meta.loc[meta['N']==2519, :]\n",
    "_,q,_,_ = multipletests(meta_both['PVAL'].values, method='fdr_bh')\n",
    "meta_both['QVAL'] = q\n",
    "meta_both.sort_values(by='PVAL').head(10)\n",
    "\n",
    "_,q,_,_ = multipletests(meta['PVAL'].values, method='fdr_bh')\n",
    "meta['QVAL'] = q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript --vanilla ../../code/gwas_analysis/run_GMMAT_fromDosage_text_Wald.R ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/Covar_Pheno_UnrelatedPCs.txt ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.tsv ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/rel/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220107.Deduped.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.maf_1e-2.hwe_1e-6.geno_5e-2.rel.withids.tsv \"MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl ~ SEX + PC1 + PC2 + PC3 + PC4 + PC5 + PC6\" ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.results.wald.tsv 15\n",
      "Rscript --vanilla ../../code/gwas_analysis/run_GMMAT_fromDosage_text_Wald.R ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/Covar_Pheno_UnrelatedPCs.txt ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.tsv ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/rel/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220107.Deduped.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.maf_1e-2.hwe_1e-6.geno_5e-2.rel.withids.tsv \"MergeEpoch_SL_AgORSeqPos_Vs_PopControl ~ SEX + PC1 + PC2 + PC3 + PC4 + PC5 + PC6\" ../../data/tmp/HLA/Association/AllGene_Dosage_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.results.wald.tsv 15\n"
     ]
    }
   ],
   "source": [
    "Rcmdbase = 'Rscript --vanilla ../../code/gwas_analysis/run_GMMAT_fromDosage_text_Wald.R %s %s %s %s %s %d'\n",
    "ncores = 15\n",
    "for p in phenotypes:\n",
    "    pdir = '{base}/{p}'.format(base=resdir, p=p)\n",
    "    phenoresdir = os.path.join(pdir, 'results')\n",
    "    rawgwasdir = os.path.join(phenoresdir, 'raw_gwas')\n",
    "    formula ='\"{p} ~ SEX + PC1 + PC2 + PC3 + PC4 + PC5 + PC6\"'.format(p=p)\n",
    "    covarstr = ''.join(covarmap[p])\n",
    "    nulloutfile = os.path.join(rawgwasdir, \"gmmat.%s.%s.nullmodel.rds\" % (covarstr, p))\n",
    "    phenofn = os.path.join(pdir, 'Covar_Pheno_UnrelatedPCs.txt')\n",
    "    relfn = os.path.join(pdir, 'rel/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220107.Deduped.{p}.maf_1e-2.hwe_1e-6.geno_5e-2.rel.withids.tsv').format(p=p)\n",
    "    pheno = pd.read_csv(phenofn, sep='\\t')\n",
    "    genotfn = os.path.join(assocdir, 'AllGene_Dosage_Maf_01.{p}.tsv'.format(p=p))\n",
    "    outfn = os.path.join(assocdir, 'AllGene_Dosage_Maf_01.{p}.results.wald.tsv'.format(p=p))    \n",
    "\n",
    "    Rcmd = Rcmdbase % (phenofn, genotfn, relfn, formula, outfn, ncores)\n",
    "    print(Rcmd)\n",
    "    #!{Rcmd}\n",
    "    wald_dat = pd.read_csv(outfn, sep='\\t')\n",
    "    wald_dat.index = wald_dat['SNP']\n",
    "    results[p]['OR'] = wald_dat.loc[results[p].index, 'BETA'].apply(np.exp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>N</th>\n",
       "      <th>AF</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>VAR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C*14:02</th>\n",
       "      <td>C*14:02</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.018615</td>\n",
       "      <td>7.05560</td>\n",
       "      <td>6.09635</td>\n",
       "      <td>0.004269</td>\n",
       "      <td>0.381949</td>\n",
       "      <td>2.793635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*15:16</th>\n",
       "      <td>B*15:16</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.026584</td>\n",
       "      <td>8.06405</td>\n",
       "      <td>8.94377</td>\n",
       "      <td>0.007008</td>\n",
       "      <td>0.381949</td>\n",
       "      <td>2.271244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*52:01</th>\n",
       "      <td>B*52:01</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.014350</td>\n",
       "      <td>-4.81087</td>\n",
       "      <td>4.00562</td>\n",
       "      <td>0.016228</td>\n",
       "      <td>0.584849</td>\n",
       "      <td>0.262185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A*02:01</th>\n",
       "      <td>A*02:01</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.096311</td>\n",
       "      <td>-11.67760</td>\n",
       "      <td>29.38770</td>\n",
       "      <td>0.031230</td>\n",
       "      <td>0.584849</td>\n",
       "      <td>0.660209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQA1*05:01</th>\n",
       "      <td>DQA1*05:01</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.051717</td>\n",
       "      <td>-8.47301</td>\n",
       "      <td>17.27290</td>\n",
       "      <td>0.041479</td>\n",
       "      <td>0.584849</td>\n",
       "      <td>0.592874</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP     N        AF     SCORE       VAR      PVAL  \\\n",
       "SNP                                                                    \n",
       "C*14:02        C*14:02  1598  0.018615   7.05560   6.09635  0.004269   \n",
       "B*15:16        B*15:16  1598  0.026584   8.06405   8.94377  0.007008   \n",
       "B*52:01        B*52:01  1598  0.014350  -4.81087   4.00562  0.016228   \n",
       "A*02:01        A*02:01  1598  0.096311 -11.67760  29.38770  0.031230   \n",
       "DQA1*05:01  DQA1*05:01  1598  0.051717  -8.47301  17.27290  0.041479   \n",
       "\n",
       "                QVAL        OR  \n",
       "SNP                             \n",
       "C*14:02     0.381949  2.793635  \n",
       "B*15:16     0.381949  2.271244  \n",
       "B*52:01     0.584849  0.262185  \n",
       "A*02:01     0.584849  0.660209  \n",
       "DQA1*05:01  0.584849  0.592874  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>N</th>\n",
       "      <th>AF</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>VAR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DPA1*02:02</th>\n",
       "      <td>DPA1*02:02</td>\n",
       "      <td>921</td>\n",
       "      <td>0.274253</td>\n",
       "      <td>-14.32660</td>\n",
       "      <td>26.92320</td>\n",
       "      <td>0.005761</td>\n",
       "      <td>0.378817</td>\n",
       "      <td>0.565302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQB1*05:01</th>\n",
       "      <td>DQB1*05:01</td>\n",
       "      <td>921</td>\n",
       "      <td>0.123551</td>\n",
       "      <td>11.21200</td>\n",
       "      <td>18.08820</td>\n",
       "      <td>0.008383</td>\n",
       "      <td>0.378817</td>\n",
       "      <td>1.760462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPA1*02:01</th>\n",
       "      <td>DPA1*02:01</td>\n",
       "      <td>921</td>\n",
       "      <td>0.445456</td>\n",
       "      <td>15.57400</td>\n",
       "      <td>38.93680</td>\n",
       "      <td>0.012566</td>\n",
       "      <td>0.378817</td>\n",
       "      <td>1.494460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*07:01</th>\n",
       "      <td>C*07:01</td>\n",
       "      <td>921</td>\n",
       "      <td>0.045904</td>\n",
       "      <td>6.47016</td>\n",
       "      <td>6.99465</td>\n",
       "      <td>0.014428</td>\n",
       "      <td>0.378817</td>\n",
       "      <td>2.180002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*15:03</th>\n",
       "      <td>DRB1*15:03</td>\n",
       "      <td>921</td>\n",
       "      <td>0.022577</td>\n",
       "      <td>4.52815</td>\n",
       "      <td>3.58392</td>\n",
       "      <td>0.016762</td>\n",
       "      <td>0.378817</td>\n",
       "      <td>2.681294</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP    N        AF     SCORE       VAR      PVAL      QVAL  \\\n",
       "SNP                                                                             \n",
       "DPA1*02:02  DPA1*02:02  921  0.274253 -14.32660  26.92320  0.005761  0.378817   \n",
       "DQB1*05:01  DQB1*05:01  921  0.123551  11.21200  18.08820  0.008383  0.378817   \n",
       "DPA1*02:01  DPA1*02:01  921  0.445456  15.57400  38.93680  0.012566  0.378817   \n",
       "C*07:01        C*07:01  921  0.045904   6.47016   6.99465  0.014428  0.378817   \n",
       "DRB1*15:03  DRB1*15:03  921  0.022577   4.52815   3.58392  0.016762  0.378817   \n",
       "\n",
       "                  OR  \n",
       "SNP                   \n",
       "DPA1*02:02  0.565302  \n",
       "DQB1*05:01  1.760462  \n",
       "DPA1*02:01  1.494460  \n",
       "C*07:01     2.180002  \n",
       "DRB1*15:03  2.681294  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for p in phenotypes:\n",
    "    display(results[p].sort_values(by='PVAL').head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/py36/lib/python3.6/site-packages/pandas/core/indexing.py:1404: FutureWarning: \n",
      "Passing list-likes to .loc or [] with any missing label will raise\n",
      "KeyError in the future, you can use .reindex() as an alternative.\n",
      "\n",
      "See the documentation here:\n",
      "https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike\n",
      "  return self._getitem_tuple(key)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">NG</th>\n",
       "      <th colspan=\"3\" halign=\"left\">SL</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Meta</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>AF</th>\n",
       "      <th>OR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>AF</th>\n",
       "      <th>OR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>B*15:16</th>\n",
       "      <td>0.026584</td>\n",
       "      <td>2.271244</td>\n",
       "      <td>0.007008</td>\n",
       "      <td>0.018077</td>\n",
       "      <td>1.482257</td>\n",
       "      <td>0.458528</td>\n",
       "      <td>0.006684</td>\n",
       "      <td>0.443502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*14:02</th>\n",
       "      <td>0.018615</td>\n",
       "      <td>2.793635</td>\n",
       "      <td>0.004269</td>\n",
       "      <td>0.019571</td>\n",
       "      <td>1.398048</td>\n",
       "      <td>0.515944</td>\n",
       "      <td>0.006876</td>\n",
       "      <td>0.443502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*15:03</th>\n",
       "      <td>0.243576</td>\n",
       "      <td>1.266862</td>\n",
       "      <td>0.055630</td>\n",
       "      <td>0.022577</td>\n",
       "      <td>2.681294</td>\n",
       "      <td>0.016762</td>\n",
       "      <td>0.015897</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPA1*02:02</th>\n",
       "      <td>0.180110</td>\n",
       "      <td>0.904495</td>\n",
       "      <td>0.482959</td>\n",
       "      <td>0.274253</td>\n",
       "      <td>0.565302</td>\n",
       "      <td>0.005761</td>\n",
       "      <td>0.027561</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*52:01</th>\n",
       "      <td>0.014350</td>\n",
       "      <td>0.262185</td>\n",
       "      <td>0.016228</td>\n",
       "      <td>0.010389</td>\n",
       "      <td>0.999957</td>\n",
       "      <td>0.999707</td>\n",
       "      <td>0.027666</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*78:01</th>\n",
       "      <td>0.002675</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.029817</td>\n",
       "      <td>2.262743</td>\n",
       "      <td>0.033891</td>\n",
       "      <td>0.033891</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*08:06</th>\n",
       "      <td>0.000968</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.022056</td>\n",
       "      <td>0.135602</td>\n",
       "      <td>0.041165</td>\n",
       "      <td>0.041165</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPA1*02:01</th>\n",
       "      <td>0.354411</td>\n",
       "      <td>1.071581</td>\n",
       "      <td>0.520416</td>\n",
       "      <td>0.445456</td>\n",
       "      <td>1.494460</td>\n",
       "      <td>0.012566</td>\n",
       "      <td>0.054253</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*03:02</th>\n",
       "      <td>0.026070</td>\n",
       "      <td>0.797934</td>\n",
       "      <td>0.527066</td>\n",
       "      <td>0.045552</td>\n",
       "      <td>0.402719</td>\n",
       "      <td>0.049165</td>\n",
       "      <td>0.070048</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQB1*02:01</th>\n",
       "      <td>0.054898</td>\n",
       "      <td>0.603755</td>\n",
       "      <td>0.047375</td>\n",
       "      <td>0.057394</td>\n",
       "      <td>0.995807</td>\n",
       "      <td>0.990269</td>\n",
       "      <td>0.101615</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*03:01</th>\n",
       "      <td>0.052193</td>\n",
       "      <td>0.602298</td>\n",
       "      <td>0.046988</td>\n",
       "      <td>0.054675</td>\n",
       "      <td>1.043818</td>\n",
       "      <td>0.904740</td>\n",
       "      <td>0.111988</td>\n",
       "      <td>0.669291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQA1*05:01</th>\n",
       "      <td>0.051717</td>\n",
       "      <td>0.592874</td>\n",
       "      <td>0.041479</td>\n",
       "      <td>0.058139</td>\n",
       "      <td>1.135128</td>\n",
       "      <td>0.704067</td>\n",
       "      <td>0.146163</td>\n",
       "      <td>0.724313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPB1*104:01</th>\n",
       "      <td>0.035886</td>\n",
       "      <td>0.554805</td>\n",
       "      <td>0.047785</td>\n",
       "      <td>0.013688</td>\n",
       "      <td>1.985782</td>\n",
       "      <td>0.211152</td>\n",
       "      <td>0.173401</td>\n",
       "      <td>0.724313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQB1*05:01</th>\n",
       "      <td>0.144763</td>\n",
       "      <td>0.955528</td>\n",
       "      <td>0.746068</td>\n",
       "      <td>0.123551</td>\n",
       "      <td>1.760462</td>\n",
       "      <td>0.008383</td>\n",
       "      <td>0.285913</td>\n",
       "      <td>0.773177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C*07:01</th>\n",
       "      <td>0.058280</td>\n",
       "      <td>0.943455</td>\n",
       "      <td>0.793514</td>\n",
       "      <td>0.045904</td>\n",
       "      <td>2.180002</td>\n",
       "      <td>0.014428</td>\n",
       "      <td>0.313120</td>\n",
       "      <td>0.773177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A*02:01</th>\n",
       "      <td>0.096311</td>\n",
       "      <td>0.660209</td>\n",
       "      <td>0.031230</td>\n",
       "      <td>0.148679</td>\n",
       "      <td>1.570313</td>\n",
       "      <td>0.090363</td>\n",
       "      <td>0.383552</td>\n",
       "      <td>0.773177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   NG                            SL                      \\\n",
       "                   AF        OR      PVAL        AF        OR      PVAL   \n",
       "SNP                                                                       \n",
       "B*15:16      0.026584  2.271244  0.007008  0.018077  1.482257  0.458528   \n",
       "C*14:02      0.018615  2.793635  0.004269  0.019571  1.398048  0.515944   \n",
       "DRB1*15:03   0.243576  1.266862  0.055630  0.022577  2.681294  0.016762   \n",
       "DPA1*02:02   0.180110  0.904495  0.482959  0.274253  0.565302  0.005761   \n",
       "B*52:01      0.014350  0.262185  0.016228  0.010389  0.999957  0.999707   \n",
       "B*78:01      0.002675       NaN       NaN  0.029817  2.262743  0.033891   \n",
       "DRB1*08:06   0.000968       NaN       NaN  0.022056  0.135602  0.041165   \n",
       "DPA1*02:01   0.354411  1.071581  0.520416  0.445456  1.494460  0.012566   \n",
       "C*03:02      0.026070  0.797934  0.527066  0.045552  0.402719  0.049165   \n",
       "DQB1*02:01   0.054898  0.603755  0.047375  0.057394  0.995807  0.990269   \n",
       "DRB1*03:01   0.052193  0.602298  0.046988  0.054675  1.043818  0.904740   \n",
       "DQA1*05:01   0.051717  0.592874  0.041479  0.058139  1.135128  0.704067   \n",
       "DPB1*104:01  0.035886  0.554805  0.047785  0.013688  1.985782  0.211152   \n",
       "DQB1*05:01   0.144763  0.955528  0.746068  0.123551  1.760462  0.008383   \n",
       "C*07:01      0.058280  0.943455  0.793514  0.045904  2.180002  0.014428   \n",
       "A*02:01      0.096311  0.660209  0.031230  0.148679  1.570313  0.090363   \n",
       "\n",
       "                 Meta            \n",
       "                 PVAL      QVAL  \n",
       "SNP                              \n",
       "B*15:16      0.006684  0.443502  \n",
       "C*14:02      0.006876  0.443502  \n",
       "DRB1*15:03   0.015897  0.669291  \n",
       "DPA1*02:02   0.027561  0.669291  \n",
       "B*52:01      0.027666  0.669291  \n",
       "B*78:01      0.033891  0.669291  \n",
       "DRB1*08:06   0.041165  0.669291  \n",
       "DPA1*02:01   0.054253  0.669291  \n",
       "C*03:02      0.070048  0.669291  \n",
       "DQB1*02:01   0.101615  0.669291  \n",
       "DRB1*03:01   0.111988  0.669291  \n",
       "DQA1*05:01   0.146163  0.724313  \n",
       "DPB1*104:01  0.173401  0.724313  \n",
       "DQB1*05:01   0.285913  0.773177  \n",
       "C*07:01      0.313120  0.773177  \n",
       "A*02:01      0.383552  0.773177  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hits = []\n",
    "pthresh = .05\n",
    "for p in phenotypes:\n",
    "    hits += list(results[p].loc[results[p]['PVAL']< pthresh, 'SNP'])\n",
    "\n",
    "\n",
    "#hits = meta_both.loc[meta_both['PVAL']<.05, 'SNP']\n",
    "    \n",
    "ngdat = results['MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl'].loc[hits, ['AF', 'OR', 'PVAL']]\n",
    "sldat = results['MergeEpoch_SL_AgORSeqPos_Vs_PopControl'].loc[hits, ['AF', 'OR', 'PVAL']]\n",
    "\n",
    "\n",
    "for v in ['DRB1*08:06', 'B*78:01']:\n",
    "    ngdat.loc[v, 'AF'] = AF_both['MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl'].loc[v]\n",
    "\n",
    "\n",
    "ngdat.columns = [('NG', x) for x in ngdat.columns]\n",
    "sldat.columns = [('SL', x) for x in sldat.columns]\n",
    "zmeta = meta.loc[hits, ['PVAL', 'QVAL']]\n",
    "zmeta.columns = [('Meta', x) for x in zmeta.columns]\n",
    "\n",
    "comb = pd.concat([ngdat, sldat, zmeta], axis=1)\n",
    "comb.columns = pd.MultiIndex.from_tuples(comb.columns)\n",
    "comb.sort_values(by=('Meta', 'PVAL'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/py36/lib/python3.6/site-packages/ipykernel_launcher.py:5: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  \"\"\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AF</th>\n",
       "      <th>N</th>\n",
       "      <th>OR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>SNP</th>\n",
       "      <th>VAR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>B*52:01</th>\n",
       "      <td>0.0143497</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.262185</td>\n",
       "      <td>0.0162281</td>\n",
       "      <td>0.584849</td>\n",
       "      <td>-4.81087</td>\n",
       "      <td>B*52:01</td>\n",
       "      <td>4.00562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*52:01</th>\n",
       "      <td>0.0103885</td>\n",
       "      <td>921</td>\n",
       "      <td>0.999957</td>\n",
       "      <td>0.999707</td>\n",
       "      <td>0.999707</td>\n",
       "      <td>-0.000321926</td>\n",
       "      <td>B*52:01</td>\n",
       "      <td>0.768302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*52:01</th>\n",
       "      <td>0.0129014</td>\n",
       "      <td>2519</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0276661</td>\n",
       "      <td>0.669291</td>\n",
       "      <td>-4.81119</td>\n",
       "      <td>B*52:01</td>\n",
       "      <td>4.77392</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                AF     N        OR       PVAL      QVAL        SCORE      SNP  \\\n",
       "B*52:01  0.0143497  1598  0.262185  0.0162281  0.584849     -4.81087  B*52:01   \n",
       "B*52:01  0.0103885   921  0.999957   0.999707  0.999707 -0.000321926  B*52:01   \n",
       "B*52:01  0.0129014  2519       NaN  0.0276661  0.669291     -4.81119  B*52:01   \n",
       "\n",
       "              VAR  \n",
       "B*52:01   4.00562  \n",
       "B*52:01  0.768302  \n",
       "B*52:01   4.77392  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.concat([\n",
    "    results['MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl'].loc['B*52:01', :],\n",
    "    results['MergeEpoch_SL_AgORSeqPos_Vs_PopControl'].loc['B*52:01', :],\n",
    "    meta.loc['B*52:01', :].drop(['A1', 'A2'])\n",
    "], axis=1).T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SNP        B*52:01\n",
       "A1            True\n",
       "A2               A\n",
       "N             2519\n",
       "AF       0.0129014\n",
       "SCORE     -4.81119\n",
       "VAR        4.77392\n",
       "PVAL     0.0276661\n",
       "QVAL      0.669291\n",
       "Name: B*52:01, dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.loc['B*52:01', :]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Repeat with 2 digit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A*01</th>\n",
       "      <th>A*02</th>\n",
       "      <th>A*03</th>\n",
       "      <th>A*11</th>\n",
       "      <th>A*23</th>\n",
       "      <th>A*24</th>\n",
       "      <th>A*25</th>\n",
       "      <th>A*26</th>\n",
       "      <th>A*29</th>\n",
       "      <th>A*30</th>\n",
       "      <th>...</th>\n",
       "      <th>DPB1*39</th>\n",
       "      <th>DPB1*40</th>\n",
       "      <th>DPB1*417</th>\n",
       "      <th>DPB1*461</th>\n",
       "      <th>DPB1*463</th>\n",
       "      <th>DPB1*85</th>\n",
       "      <th>DPA1*01</th>\n",
       "      <th>DPA1*02</th>\n",
       "      <th>DPA1*03</th>\n",
       "      <th>DPA1*04</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>WG0284565-DNA_A02_G-5231</th>\n",
       "      <td>8.980594e-10</td>\n",
       "      <td>3.592218e-04</td>\n",
       "      <td>5.966318e-05</td>\n",
       "      <td>9.884059e-28</td>\n",
       "      <td>0.001356</td>\n",
       "      <td>6.517996e-26</td>\n",
       "      <td>1.178850e-11</td>\n",
       "      <td>2.337478e-06</td>\n",
       "      <td>2.469260e-18</td>\n",
       "      <td>0.998225</td>\n",
       "      <td>...</td>\n",
       "      <td>1.498780e-28</td>\n",
       "      <td>0.001438</td>\n",
       "      <td>5.414226e-10</td>\n",
       "      <td>7.268555e-07</td>\n",
       "      <td>7.144158e-07</td>\n",
       "      <td>8.437959e-09</td>\n",
       "      <td>0.001149</td>\n",
       "      <td>1.998851</td>\n",
       "      <td>8.745754e-09</td>\n",
       "      <td>1.303046e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WG0284565-DNA_A03_G-5686</th>\n",
       "      <td>8.980982e-10</td>\n",
       "      <td>1.000359e+00</td>\n",
       "      <td>5.966575e-05</td>\n",
       "      <td>1.141340e-27</td>\n",
       "      <td>0.002020</td>\n",
       "      <td>6.890107e-30</td>\n",
       "      <td>3.488029e-32</td>\n",
       "      <td>4.674195e-19</td>\n",
       "      <td>2.559077e-18</td>\n",
       "      <td>0.997561</td>\n",
       "      <td>...</td>\n",
       "      <td>1.110960e-48</td>\n",
       "      <td>0.001508</td>\n",
       "      <td>1.893355e-09</td>\n",
       "      <td>4.785941e-04</td>\n",
       "      <td>7.322280e-04</td>\n",
       "      <td>3.749667e-04</td>\n",
       "      <td>0.004427</td>\n",
       "      <td>1.992840</td>\n",
       "      <td>2.733096e-03</td>\n",
       "      <td>1.604225e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WG0284565-DNA_A04_G-5690</th>\n",
       "      <td>5.607632e-10</td>\n",
       "      <td>2.242875e-04</td>\n",
       "      <td>5.939966e-05</td>\n",
       "      <td>6.542605e-19</td>\n",
       "      <td>0.996537</td>\n",
       "      <td>1.654490e-18</td>\n",
       "      <td>3.051315e-40</td>\n",
       "      <td>2.073262e-14</td>\n",
       "      <td>9.195031e-19</td>\n",
       "      <td>0.999054</td>\n",
       "      <td>...</td>\n",
       "      <td>2.338300e-02</td>\n",
       "      <td>0.976616</td>\n",
       "      <td>7.352054e-27</td>\n",
       "      <td>3.223394e-23</td>\n",
       "      <td>1.032338e-26</td>\n",
       "      <td>1.520536e-12</td>\n",
       "      <td>0.134395</td>\n",
       "      <td>0.949052</td>\n",
       "      <td>9.165531e-01</td>\n",
       "      <td>3.764090e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WG0284565-DNA_A05_G-5135</th>\n",
       "      <td>8.900894e-10</td>\n",
       "      <td>2.045765e-02</td>\n",
       "      <td>5.954924e-05</td>\n",
       "      <td>1.149885e-27</td>\n",
       "      <td>0.002043</td>\n",
       "      <td>1.019281e-25</td>\n",
       "      <td>3.309208e-39</td>\n",
       "      <td>4.663327e-19</td>\n",
       "      <td>2.469627e-18</td>\n",
       "      <td>0.997541</td>\n",
       "      <td>...</td>\n",
       "      <td>1.189321e-17</td>\n",
       "      <td>0.001541</td>\n",
       "      <td>9.483453e-10</td>\n",
       "      <td>1.336075e-04</td>\n",
       "      <td>1.336194e-04</td>\n",
       "      <td>7.503476e-09</td>\n",
       "      <td>1.006336</td>\n",
       "      <td>0.991862</td>\n",
       "      <td>1.801766e-03</td>\n",
       "      <td>1.591168e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WG0284565-DNA_A06_G-5045</th>\n",
       "      <td>3.665769e-18</td>\n",
       "      <td>2.174426e-16</td>\n",
       "      <td>1.148462e-09</td>\n",
       "      <td>7.008001e-20</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>1.155316e-09</td>\n",
       "      <td>3.674382e-28</td>\n",
       "      <td>2.300854e-27</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>...</td>\n",
       "      <td>1.247548e-01</td>\n",
       "      <td>0.870925</td>\n",
       "      <td>1.335608e-31</td>\n",
       "      <td>1.175743e-42</td>\n",
       "      <td>5.877356e-32</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>9.999962e-01</td>\n",
       "      <td>1.419585e-21</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  121 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  A*01          A*02          A*03  \\\n",
       "WG0284565-DNA_A02_G-5231  8.980594e-10  3.592218e-04  5.966318e-05   \n",
       "WG0284565-DNA_A03_G-5686  8.980982e-10  1.000359e+00  5.966575e-05   \n",
       "WG0284565-DNA_A04_G-5690  5.607632e-10  2.242875e-04  5.939966e-05   \n",
       "WG0284565-DNA_A05_G-5135  8.900894e-10  2.045765e-02  5.954924e-05   \n",
       "WG0284565-DNA_A06_G-5045  3.665769e-18  2.174426e-16  1.148462e-09   \n",
       "\n",
       "                                  A*11      A*23          A*24          A*25  \\\n",
       "WG0284565-DNA_A02_G-5231  9.884059e-28  0.001356  6.517996e-26  1.178850e-11   \n",
       "WG0284565-DNA_A03_G-5686  1.141340e-27  0.002020  6.890107e-30  3.488029e-32   \n",
       "WG0284565-DNA_A04_G-5690  6.542605e-19  0.996537  1.654490e-18  3.051315e-40   \n",
       "WG0284565-DNA_A05_G-5135  1.149885e-27  0.002043  1.019281e-25  3.309208e-39   \n",
       "WG0284565-DNA_A06_G-5045  7.008001e-20  0.000017  1.155316e-09  3.674382e-28   \n",
       "\n",
       "                                  A*26          A*29      A*30  ...  \\\n",
       "WG0284565-DNA_A02_G-5231  2.337478e-06  2.469260e-18  0.998225  ...   \n",
       "WG0284565-DNA_A03_G-5686  4.674195e-19  2.559077e-18  0.997561  ...   \n",
       "WG0284565-DNA_A04_G-5690  2.073262e-14  9.195031e-19  0.999054  ...   \n",
       "WG0284565-DNA_A05_G-5135  4.663327e-19  2.469627e-18  0.997541  ...   \n",
       "WG0284565-DNA_A06_G-5045  2.300854e-27  1.000000e+00  0.000001  ...   \n",
       "\n",
       "                               DPB1*39   DPB1*40      DPB1*417      DPB1*461  \\\n",
       "WG0284565-DNA_A02_G-5231  1.498780e-28  0.001438  5.414226e-10  7.268555e-07   \n",
       "WG0284565-DNA_A03_G-5686  1.110960e-48  0.001508  1.893355e-09  4.785941e-04   \n",
       "WG0284565-DNA_A04_G-5690  2.338300e-02  0.976616  7.352054e-27  3.223394e-23   \n",
       "WG0284565-DNA_A05_G-5135  1.189321e-17  0.001541  9.483453e-10  1.336075e-04   \n",
       "WG0284565-DNA_A06_G-5045  1.247548e-01  0.870925  1.335608e-31  1.175743e-42   \n",
       "\n",
       "                              DPB1*463       DPB1*85   DPA1*01   DPA1*02  \\\n",
       "WG0284565-DNA_A02_G-5231  7.144158e-07  8.437959e-09  0.001149  1.998851   \n",
       "WG0284565-DNA_A03_G-5686  7.322280e-04  3.749667e-04  0.004427  1.992840   \n",
       "WG0284565-DNA_A04_G-5690  1.032338e-26  1.520536e-12  0.134395  0.949052   \n",
       "WG0284565-DNA_A05_G-5135  1.336194e-04  7.503476e-09  1.006336  0.991862   \n",
       "WG0284565-DNA_A06_G-5045  5.877356e-32  1.000000e+00  0.000007  0.999996   \n",
       "\n",
       "                               DPA1*03       DPA1*04  \n",
       "WG0284565-DNA_A02_G-5231  8.745754e-09  1.303046e-08  \n",
       "WG0284565-DNA_A03_G-5686  2.733096e-03  1.604225e-10  \n",
       "WG0284565-DNA_A04_G-5690  9.165531e-01  3.764090e-17  \n",
       "WG0284565-DNA_A05_G-5135  1.801766e-03  1.591168e-12  \n",
       "WG0284565-DNA_A06_G-5045  9.999962e-01  1.419585e-21  \n",
       "\n",
       "[5 rows x 121 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hlabase = '../../data/tmp/HLA/Stanford/Predictions/Omni25AffyMerged.{g}.Nclass_7.Flank_500K.4dig.Model.20220206.AllProbabilities.tsv'\n",
    "genes = ['A', 'B', 'C', 'DRB1', 'DQB1', 'DQA1', 'DPB1', 'DPA1']\n",
    "all_allele_data = []\n",
    "import sys\n",
    "for g in genes:\n",
    "    hlafn = hlabase.format(g=g)\n",
    "    hladat = pd.read_csv(hlafn, sep='\\t')\n",
    "    x1 = hladat.groupby(lambda x: x.split('/')[0].split(':')[0], axis=0).sum()\n",
    "    x2 = hladat.groupby(lambda x: x.split('/')[1].split(':')[0], axis=0).sum()\n",
    "    all_alleles = sorted(set(x1.index).union(x2.index))\n",
    "    dosage = x1.loc[all_alleles, :].replace(np.nan, 0) + x2.loc[all_alleles, :].replace(np.nan, 0)\n",
    "    dosage.index = [g+'*'+x for x in dosage.index]\n",
    "    all_allele_data.append(dosage.T)\n",
    "    \n",
    "all_allele_data = pd.concat(all_allele_data, axis=1)\n",
    "all_allele_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "dosagefn = os.path.join(assocdir, 'AllGene_Dosage_2dig.tsv')\n",
    "all_allele_data.to_csv(dosagefn, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DPB1*05     0.000015\n",
       "A*25        0.000031\n",
       "DPB1*20     0.000060\n",
       "DPB1*106    0.000097\n",
       "A*11        0.000203\n",
       "dtype: float64"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AF = all_allele_data.sum(axis=0) / (all_allele_data.shape[0]*2)\n",
    "AF.sort_values().head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_allele_data_maf01 = all_allele_data.loc[:, (AF>.01) & (AF<.99)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "dosagefn = os.path.join(assocdir, 'AllGene_Dosage_2dig_Maf_01.tsv')\n",
    "all_allele_data_maf01.to_csv(dosagefn, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "phenotypes = ['MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl', 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl']\n",
    "covarmap = {'MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl': ['SEX', 'PCs'],\n",
    " 'MergeEpoch_SL_AgORSeqPos_Vs_PopControl': ['SEX','PCs']}\n",
    "\n",
    "tmpdir = '../../data/tmp/'\n",
    "gwasdir = os.path.join(tmpdir, 'GWAS')\n",
    "resdir = os.path.join(gwasdir, 'results_noH3covar_20220207')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    1598\n",
       "Name: IID, dtype: int64"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript ../../code/gwas_analysis/run_GMMAT_premade_model_text_dosage.R ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl/results/raw_gwas/gmmat.SEXPCs.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.nullmodel.rds ../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.tsv ../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.results.tsv\n",
      "Warning message:\n",
      "In glmm.score(model0, infile = genotypefn, outfile = outfn, is.dosage = FALSE,  :\n",
      "  Argument select is unspecified... Assuming the order of individuals in infile matches unique id_include in obj...\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True    921\n",
       "Name: IID, dtype: int64"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript ../../code/gwas_analysis/run_GMMAT_premade_model_text_dosage.R ../../data/tmp/GWAS/results_noH3covar_20220207/MergeEpoch_SL_AgORSeqPos_Vs_PopControl/results/raw_gwas/gmmat.SEXPCs.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.nullmodel.rds ../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.tsv ../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.results.tsv\n",
      "Warning message:\n",
      "In glmm.score(model0, infile = genotypefn, outfile = outfn, is.dosage = FALSE,  :\n",
      "  Argument select is unspecified... Assuming the order of individuals in infile matches unique id_include in obj...\n"
     ]
    }
   ],
   "source": [
    "results = {}\n",
    "AF_thresh = .01\n",
    "AF_both = {}\n",
    "for p in phenotypes:\n",
    "    pdir = '{base}/{p}'.format(base=resdir, p=p)\n",
    "    phenoresdir = os.path.join(pdir, 'results')\n",
    "    rawgwasdir = os.path.join(phenoresdir, 'raw_gwas')\n",
    "\n",
    "    covarstr = ''.join(covarmap[p])\n",
    "    nulloutfile = os.path.join(rawgwasdir, \"gmmat.%s.%s.nullmodel.rds\" % (covarstr, p))\n",
    "    phenofn = os.path.join(pdir, 'Covar_Pheno_UnrelatedPCs.txt')\n",
    "    relfn = os.path.join(pdir, 'rel/OmniH3Merged_PrePost2016.20210817.FiltSLOmni5.Geno_1e-1.filtBatchvars20220107.Deduped.{p}.maf_1e-2.hwe_1e-6.geno_5e-2.rel.withids.tsv').format(p=p)\n",
    "    pheno = pd.read_csv(phenofn, sep='\\t')\n",
    "    rel = pd.read_csv(relfn, sep='\\t', index_col=0)\n",
    "    display((pheno['IID']==rel.index).value_counts())\n",
    "    genot = all_allele_data.loc[rel.index,:].T\n",
    "    AF = genot.sum(axis=1) / (genot.shape[1]*2)\n",
    "    AF_both[p] = AF\n",
    "    genot = genot.loc[(AF>AF_thresh) & (AF < (1- AF_thresh)), :]    \n",
    "    samps = list(rel.index)\n",
    "    genot['SNP'] = genot.index\n",
    "    genot['REF'] = 'A'\n",
    "    genot['ALT'] = 'T'\n",
    "    genot = genot.loc[:, ['SNP', 'REF', 'ALT'] + samps]\n",
    "    genotfn = os.path.join(assocdir, 'AllGene_Dosage_2dig_Maf_01.{p}.tsv'.format(p=p))\n",
    "    genot.to_csv(genotfn, sep='\\t', index=False)\n",
    "    outfn = os.path.join(assocdir, 'AllGene_Dosage_2dig_Maf_01.{p}.results.tsv'.format(p=p))    \n",
    "    cmd = 'Rscript ../../code/gwas_analysis/run_GMMAT_premade_model_text_dosage.R %s %s %s' % (nulloutfile, genotfn, outfn)\n",
    "    print(cmd)\n",
    "    !{cmd}\n",
    "    results[p] = pd.read_csv(outfn, sep='\\t')\n",
    "    results[p].index = results[p]['SNP']\n",
    "    _,q,_,_ = multipletests(results[p]['PVAL'].values, method='fdr_bh')\n",
    "    results[p]['QVAL'] = q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>N</th>\n",
       "      <th>AF</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>VAR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C*14</th>\n",
       "      <td>C*14</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.022431</td>\n",
       "      <td>7.69268</td>\n",
       "      <td>7.50715</td>\n",
       "      <td>0.004991</td>\n",
       "      <td>0.379295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*52</th>\n",
       "      <td>B*52</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.014350</td>\n",
       "      <td>-4.81087</td>\n",
       "      <td>4.00562</td>\n",
       "      <td>0.016228</td>\n",
       "      <td>0.495068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQB1*02</th>\n",
       "      <td>DQB1*02</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.221116</td>\n",
       "      <td>-17.07740</td>\n",
       "      <td>57.58010</td>\n",
       "      <td>0.024415</td>\n",
       "      <td>0.495068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A*02</th>\n",
       "      <td>A*02</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.155092</td>\n",
       "      <td>-15.03510</td>\n",
       "      <td>45.64630</td>\n",
       "      <td>0.026056</td>\n",
       "      <td>0.495068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPB1*104</th>\n",
       "      <td>DPB1*104</td>\n",
       "      <td>1598</td>\n",
       "      <td>0.035886</td>\n",
       "      <td>-7.20772</td>\n",
       "      <td>13.26120</td>\n",
       "      <td>0.047785</td>\n",
       "      <td>0.498404</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               SNP     N        AF     SCORE       VAR      PVAL      QVAL\n",
       "SNP                                                                       \n",
       "C*14          C*14  1598  0.022431   7.69268   7.50715  0.004991  0.379295\n",
       "B*52          B*52  1598  0.014350  -4.81087   4.00562  0.016228  0.495068\n",
       "DQB1*02    DQB1*02  1598  0.221116 -17.07740  57.58010  0.024415  0.495068\n",
       "A*02          A*02  1598  0.155092 -15.03510  45.64630  0.026056  0.495068\n",
       "DPB1*104  DPB1*104  1598  0.035886  -7.20772  13.26120  0.047785  0.498404"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>N</th>\n",
       "      <th>AF</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>VAR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DQB1*05</th>\n",
       "      <td>DQB1*05</td>\n",
       "      <td>921</td>\n",
       "      <td>0.168250</td>\n",
       "      <td>13.57760</td>\n",
       "      <td>23.66380</td>\n",
       "      <td>0.005253</td>\n",
       "      <td>0.400038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A*66</th>\n",
       "      <td>A*66</td>\n",
       "      <td>921</td>\n",
       "      <td>0.019237</td>\n",
       "      <td>-4.47821</td>\n",
       "      <td>3.08643</td>\n",
       "      <td>0.010802</td>\n",
       "      <td>0.400038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*15</th>\n",
       "      <td>DRB1*15</td>\n",
       "      <td>921</td>\n",
       "      <td>0.022852</td>\n",
       "      <td>4.60287</td>\n",
       "      <td>3.59448</td>\n",
       "      <td>0.015191</td>\n",
       "      <td>0.400038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*78</th>\n",
       "      <td>B*78</td>\n",
       "      <td>921</td>\n",
       "      <td>0.029817</td>\n",
       "      <td>4.55013</td>\n",
       "      <td>4.60062</td>\n",
       "      <td>0.033891</td>\n",
       "      <td>0.558163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*27</th>\n",
       "      <td>B*27</td>\n",
       "      <td>921</td>\n",
       "      <td>0.027429</td>\n",
       "      <td>-3.60119</td>\n",
       "      <td>3.37605</td>\n",
       "      <td>0.050004</td>\n",
       "      <td>0.558163</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             SNP    N        AF     SCORE       VAR      PVAL      QVAL\n",
       "SNP                                                                    \n",
       "DQB1*05  DQB1*05  921  0.168250  13.57760  23.66380  0.005253  0.400038\n",
       "A*66        A*66  921  0.019237  -4.47821   3.08643  0.010802  0.400038\n",
       "DRB1*15  DRB1*15  921  0.022852   4.60287   3.59448  0.015191  0.400038\n",
       "B*78        B*78  921  0.029817   4.55013   4.60062  0.033891  0.558163\n",
       "B*27        B*27  921  0.027429  -3.60119   3.37605  0.050004  0.558163"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for p in phenotypes:\n",
    "    display(results[p].sort_values(by='PVAL').head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "in1 = '../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.results.tsv'\n",
    "in2 = '../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.results.tsv'\n",
    "outfn = '../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.NG_SL_Susceptibility_MetaAnalysis.results.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript ../../code/gwas_analysis/run_GMMAT_meta.R ../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_NG_ExactLVMatchqPCR2xPosOrSeqPos_Vs_PopControl.results.fixed.tsv ../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.MergeEpoch_SL_AgORSeqPos_Vs_PopControl.results.fixed.tsv ../../data/tmp/HLA/Association/AllGene_Dosage_2dig_Maf_01.NG_SL_Susceptibility_MetaAnalysis.results.tsv\n"
     ]
    }
   ],
   "source": [
    "for fn in [in1, in2]:\n",
    "    dat = pd.read_csv(fn, sep='\\t')\n",
    "    ## The run_GMMAT_meta.R script expects a REF and ALT column\n",
    "    dat['REF'] = 'A'\n",
    "    dat['ALT'] = 'T'\n",
    "    dat = dat[['SNP', 'REF', 'ALT', 'N', 'AF', 'SCORE', 'VAR', 'PVAL']]\n",
    "    dat.to_csv(fn.replace('.tsv', '.fixed.tsv'), sep='\\t', index=False)\n",
    "\n",
    "Rcmd = 'Rscript ../../code/gwas_analysis/run_GMMAT_meta.R %s %s %s' % (in1.replace('.tsv', '.fixed.tsv'), in2.replace('.tsv', '.fixed.tsv'), outfn)\n",
    "print(Rcmd)\n",
    "!{Rcmd}\n",
    "\n",
    "cmd = 'rm ../../data/tmp/HLA/Association/*.fixed.tsv'\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = pd.read_csv(outfn, sep='\\t')\n",
    "meta.index = meta['SNP']\n",
    "_,q,_,_ = multipletests(meta['PVAL'].values, method='fdr_bh')\n",
    "meta['QVAL'] = q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>N</th>\n",
       "      <th>AF</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>VAR</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>QVAL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C*14</th>\n",
       "      <td>C*14</td>\n",
       "      <td>True</td>\n",
       "      <td>A</td>\n",
       "      <td>2519</td>\n",
       "      <td>0.021392</td>\n",
       "      <td>8.829350</td>\n",
       "      <td>10.618120</td>\n",
       "      <td>0.006736</td>\n",
       "      <td>0.547415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRB1*15</th>\n",
       "      <td>DRB1*15</td>\n",
       "      <td>True</td>\n",
       "      <td>A</td>\n",
       "      <td>2519</td>\n",
       "      <td>0.164285</td>\n",
       "      <td>19.869970</td>\n",
       "      <td>68.849480</td>\n",
       "      <td>0.016635</td>\n",
       "      <td>0.547415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*52</th>\n",
       "      <td>B*52</td>\n",
       "      <td>True</td>\n",
       "      <td>A</td>\n",
       "      <td>2519</td>\n",
       "      <td>0.012901</td>\n",
       "      <td>-4.811192</td>\n",
       "      <td>4.773922</td>\n",
       "      <td>0.027666</td>\n",
       "      <td>0.547415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B*78</th>\n",
       "      <td>B*78</td>\n",
       "      <td>True</td>\n",
       "      <td>A</td>\n",
       "      <td>921</td>\n",
       "      <td>0.029817</td>\n",
       "      <td>4.550130</td>\n",
       "      <td>4.600620</td>\n",
       "      <td>0.033891</td>\n",
       "      <td>0.547415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DQA1*01</th>\n",
       "      <td>DQA1*01</td>\n",
       "      <td>True</td>\n",
       "      <td>A</td>\n",
       "      <td>2519</td>\n",
       "      <td>0.499743</td>\n",
       "      <td>21.819600</td>\n",
       "      <td>110.190300</td>\n",
       "      <td>0.037652</td>\n",
       "      <td>0.547415</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             SNP    A1 A2     N        AF      SCORE         VAR      PVAL  \\\n",
       "SNP                                                                          \n",
       "C*14        C*14  True  A  2519  0.021392   8.829350   10.618120  0.006736   \n",
       "DRB1*15  DRB1*15  True  A  2519  0.164285  19.869970   68.849480  0.016635   \n",
       "B*52        B*52  True  A  2519  0.012901  -4.811192    4.773922  0.027666   \n",
       "B*78        B*78  True  A   921  0.029817   4.550130    4.600620  0.033891   \n",
       "DQA1*01  DQA1*01  True  A  2519  0.499743  21.819600  110.190300  0.037652   \n",
       "\n",
       "             QVAL  \n",
       "SNP                \n",
       "C*14     0.547415  \n",
       "DRB1*15  0.547415  \n",
       "B*52     0.547415  \n",
       "B*78     0.547415  \n",
       "DQA1*01  0.547415  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.sort_values(by='PVAL').head(5)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
